{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "e66803c9-d926-4417-9e40-0a0d466e3616",
   "metadata": {},
   "outputs": [],
   "source": [
    "dl20=(at6923 at6929 at7143 at8285 at9104 at9336 at9503 at9578 at9744 at9762 at9806 at9830 at9847 at9852 at9879 at9883 at9900)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0aea04a5-6e6f-471e-9e87-32cc774fc89c",
   "metadata": {},
   "outputs": [],
   "source": [
    "dl20=(at9879)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "549011d7-0a6a-40aa-86a3-78525a24ba92",
   "metadata": {},
   "outputs": [],
   "source": [
    "chmod u+w input/*\n",
    "cp -L ../assembly-and-annotation/output/04_analyses_with_fixed_at9879/prep_for_decision_making/*.scaffolds-v2.3_liftoff_polished.gff3_polished.fix.gff3 input/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "2fa34c46-60dc-4719-ae3e-7877dc2e0cc8",
   "metadata": {},
   "outputs": [],
   "source": [
    "#cp input/at9879.augustus.gff input/at9879.augustus-liftoff.gff\n",
    "blsl liftoff-gff3 input/at9879.augustus-liftoff.gff | \\\n",
    "    awk 'BEGIN{IFS=OFS=\"\\t\"}$0 ~ /^#/{print $0}$0 !~ /^#/{print $1, \"AUGUSTUS\", $3, $4, $5, $6, $7, $8, $9}' \\\n",
    "    >input/at9879.augustus.gff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e2b76eff-cfee-4d5d-916d-c2feafa59713",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "at6923\n",
      "loci at input/at6923_loci.bed\n",
      "N loci: 33672\n",
      "Added 32234 genes from input/at6923.augustus.gff (0 missing)\n",
      "Added 26966 genes from input/at6923.liftoff.protonly.gff (0 missing)\n",
      "Added 14640 genes from input/at6923.pasa.fix.gff (0 missing)\n",
      "Added 16832 genes from input/at6923.tama-v2.fix.gff (0 missing)\n",
      "Now scan over loci...\n",
      "Now write GFFs...\n",
      "DONE! Stats follow\n",
      "\n",
      "aug_liftoff_pasa_cds_agree_use_pasa\t11133\n",
      "only_aug\t6981\n",
      "aug_liftoff_cds_agree_use_liftoff\t5303\n",
      "aug_liftoff_cds_disagree_use_aug\t3313\n",
      "only_liftoff\t1926\n",
      "TOO_HARD\t1677\n",
      "pasa_liftoff_cds_agree_use_pasa\t1549\n",
      "liftoff_augustus_agree_subgene_isoseq_merges_use_liftoff\t949\n",
      "pasa_cds_different_use_pasa\t853\n",
      "aug_pasa_cds_agree_use_pasa\t534\n",
      "aug_liftoff_pasa_agree_subgene_tama_merges_use_pasa\t85\n",
      "aug_pasa_cds_disagree_but_cds_too_long_for_isoseq_use_aug\t55\n",
      "\n",
      "\n",
      "at6929\n",
      "loci at input/at6929_loci.bed\n",
      "N loci: 33410\n",
      "Added 31940 genes from input/at6929.augustus.gff (0 missing)\n",
      "Added 26842 genes from input/at6929.liftoff.protonly.gff (0 missing)\n",
      "Added 14185 genes from input/at6929.pasa.fix.gff (0 missing)\n",
      "Added 16128 genes from input/at6929.tama-v2.fix.gff (0 missing)\n",
      "Now scan over loci...\n",
      "Now write GFFs...\n",
      "DONE! Stats follow\n",
      "\n",
      "aug_liftoff_pasa_cds_agree_use_pasa\t10898\n",
      "only_aug\t6825\n",
      "aug_liftoff_cds_agree_use_liftoff\t5672\n",
      "aug_liftoff_cds_disagree_use_aug\t3434\n",
      "only_liftoff\t1945\n",
      "TOO_HARD\t1522\n",
      "pasa_liftoff_cds_agree_use_pasa\t1457\n",
      "liftoff_augustus_agree_subgene_isoseq_merges_use_liftoff\t788\n",
      "pasa_cds_different_use_pasa\t766\n",
      "aug_pasa_cds_agree_use_pasa\t557\n",
      "aug_liftoff_pasa_agree_subgene_tama_merges_use_pasa\t58\n",
      "aug_pasa_cds_disagree_but_cds_too_long_for_isoseq_use_aug\t49\n",
      "\n",
      "\n",
      "at7143\n",
      "loci at input/at7143_loci.bed\n",
      "N loci: 33570\n",
      "Added 32149 genes from input/at7143.augustus.gff (0 missing)\n",
      "Added 26927 genes from input/at7143.liftoff.protonly.gff (0 missing)\n",
      "Added 14468 genes from input/at7143.pasa.fix.gff (0 missing)\n",
      "Added 16610 genes from input/at7143.tama-v2.fix.gff (0 missing)\n",
      "Now scan over loci...\n",
      "Now write GFFs...\n",
      "DONE! Stats follow\n",
      "\n",
      "aug_liftoff_pasa_cds_agree_use_pasa\t10994\n",
      "only_aug\t7000\n",
      "aug_liftoff_cds_agree_use_liftoff\t5458\n",
      "aug_liftoff_cds_disagree_use_aug\t3270\n",
      "only_liftoff\t1935\n",
      "TOO_HARD\t1576\n",
      "pasa_liftoff_cds_agree_use_pasa\t1565\n",
      "liftoff_augustus_agree_subgene_isoseq_merges_use_liftoff\t978\n",
      "pasa_cds_different_use_pasa\t828\n",
      "aug_pasa_cds_agree_use_pasa\t542\n",
      "aug_liftoff_pasa_agree_subgene_tama_merges_use_pasa\t69\n",
      "aug_pasa_cds_disagree_but_cds_too_long_for_isoseq_use_aug\t49\n",
      "\n",
      "\n",
      "at8285\n",
      "loci at input/at8285_loci.bed\n",
      "N loci: 33719\n",
      "Added 32306 genes from input/at8285.augustus.gff (0 missing)\n",
      "Added 26881 genes from input/at8285.liftoff.protonly.gff (0 missing)\n",
      "Added 13779 genes from input/at8285.pasa.fix.gff (0 missing)\n",
      "Added 15553 genes from input/at8285.tama-v2.fix.gff (0 missing)\n",
      "Now scan over loci...\n",
      "Now write GFFs...\n",
      "DONE! Stats follow\n",
      "\n",
      "aug_liftoff_pasa_cds_agree_use_pasa\t10494\n",
      "only_aug\t7187\n",
      "aug_liftoff_cds_agree_use_liftoff\t6087\n",
      "aug_liftoff_cds_disagree_use_aug\t3528\n",
      "only_liftoff\t1912\n",
      "pasa_liftoff_cds_agree_use_pasa\t1523\n",
      "TOO_HARD\t1395\n",
      "pasa_cds_different_use_pasa\t777\n",
      "liftoff_augustus_agree_subgene_isoseq_merges_use_liftoff\t758\n",
      "aug_pasa_cds_agree_use_pasa\t491\n",
      "aug_liftoff_pasa_agree_subgene_tama_merges_use_pasa\t61\n",
      "aug_pasa_cds_disagree_but_cds_too_long_for_isoseq_use_aug\t48\n",
      "\n",
      "\n",
      "at9104\n",
      "loci at input/at9104_loci.bed\n",
      "N loci: 33698\n",
      "Added 32144 genes from input/at9104.augustus.gff (0 missing)\n",
      "Added 26968 genes from input/at9104.liftoff.protonly.gff (0 missing)\n",
      "Added 14692 genes from input/at9104.pasa.fix.gff (0 missing)\n",
      "Added 16757 genes from input/at9104.tama-v2.fix.gff (0 missing)\n",
      "Now scan over loci...\n",
      "Now write GFFs...\n",
      "DONE! Stats follow\n",
      "\n",
      "aug_liftoff_pasa_cds_agree_use_pasa\t11165\n",
      "only_aug\t6952\n",
      "aug_liftoff_cds_agree_use_liftoff\t5315\n",
      "aug_liftoff_cds_disagree_use_aug\t3405\n",
      "only_liftoff\t1935\n",
      "TOO_HARD\t1602\n",
      "pasa_liftoff_cds_agree_use_pasa\t1597\n",
      "liftoff_augustus_agree_subgene_isoseq_merges_use_liftoff\t791\n",
      "pasa_cds_different_use_pasa\t774\n",
      "aug_pasa_cds_agree_use_pasa\t619\n",
      "aug_liftoff_pasa_agree_subgene_tama_merges_use_pasa\t66\n",
      "aug_pasa_cds_disagree_but_cds_too_long_for_isoseq_use_aug\t46\n",
      "\n",
      "\n",
      "at9336\n",
      "loci at input/at9336_loci.bed\n",
      "N loci: 34308\n",
      "Added 32798 genes from input/at9336.augustus.gff (0 missing)\n",
      "Added 26893 genes from input/at9336.liftoff.protonly.gff (0 missing)\n",
      "Added 14640 genes from input/at9336.pasa.fix.gff (0 missing)\n",
      "Added 16709 genes from input/at9336.tama-v2.fix.gff (0 missing)\n",
      "Now scan over loci...\n",
      "Now write GFFs...\n",
      "DONE! Stats follow\n",
      "\n",
      "aug_liftoff_pasa_cds_agree_use_pasa\t11133\n",
      "only_aug\t7669\n",
      "aug_liftoff_cds_agree_use_liftoff\t5215\n",
      "aug_liftoff_cds_disagree_use_aug\t3398\n",
      "only_liftoff\t1992\n",
      "pasa_liftoff_cds_agree_use_pasa\t1621\n",
      "TOO_HARD\t1533\n",
      "liftoff_augustus_agree_subgene_isoseq_merges_use_liftoff\t890\n",
      "pasa_cds_different_use_pasa\t800\n",
      "aug_pasa_cds_agree_use_pasa\t562\n",
      "aug_liftoff_pasa_agree_subgene_tama_merges_use_pasa\t65\n",
      "aug_pasa_cds_disagree_but_cds_too_long_for_isoseq_use_aug\t51\n",
      "\n",
      "\n",
      "at9503\n",
      "loci at input/at9503_loci.bed\n",
      "N loci: 33506\n",
      "Added 32072 genes from input/at9503.augustus.gff (0 missing)\n",
      "Added 26974 genes from input/at9503.liftoff.protonly.gff (0 missing)\n",
      "Added 14196 genes from input/at9503.pasa.fix.gff (0 missing)\n",
      "Added 16383 genes from input/at9503.tama-v2.fix.gff (0 missing)\n",
      "Now scan over loci...\n",
      "Now write GFFs...\n",
      "DONE! Stats follow\n",
      "\n",
      "aug_liftoff_pasa_cds_agree_use_pasa\t10784\n",
      "only_aug\t6928\n",
      "aug_liftoff_cds_agree_use_liftoff\t5562\n",
      "aug_liftoff_cds_disagree_use_aug\t3341\n",
      "only_liftoff\t1957\n",
      "TOO_HARD\t1658\n",
      "pasa_liftoff_cds_agree_use_pasa\t1516\n",
      "liftoff_augustus_agree_subgene_isoseq_merges_use_liftoff\t1031\n",
      "pasa_cds_different_use_pasa\t799\n",
      "aug_pasa_cds_agree_use_pasa\t525\n",
      "aug_liftoff_pasa_agree_subgene_tama_merges_use_pasa\t71\n",
      "aug_pasa_cds_disagree_but_cds_too_long_for_isoseq_use_aug\t48\n",
      "\n",
      "\n",
      "at9578\n",
      "loci at input/at9578_loci.bed\n",
      "N loci: 33621\n",
      "Added 32110 genes from input/at9578.augustus.gff (0 missing)\n",
      "Added 26913 genes from input/at9578.liftoff.protonly.gff (0 missing)\n",
      "Added 14825 genes from input/at9578.pasa.fix.gff (0 missing)\n",
      "Added 17243 genes from input/at9578.tama-v2.fix.gff (0 missing)\n",
      "Now scan over loci...\n",
      "Now write GFFs...\n",
      "DONE! Stats follow\n",
      "\n",
      "aug_liftoff_pasa_cds_agree_use_pasa\t11202\n",
      "only_aug\t6908\n",
      "aug_liftoff_cds_agree_use_liftoff\t5027\n",
      "aug_liftoff_cds_disagree_use_aug\t3279\n",
      "only_liftoff\t1902\n",
      "TOO_HARD\t1825\n",
      "pasa_liftoff_cds_agree_use_pasa\t1599\n",
      "liftoff_augustus_agree_subgene_isoseq_merges_use_liftoff\t1015\n",
      "pasa_cds_different_use_pasa\t858\n",
      "aug_pasa_cds_agree_use_pasa\t582\n",
      "aug_liftoff_pasa_agree_subgene_tama_merges_use_pasa\t93\n",
      "aug_pasa_cds_disagree_but_cds_too_long_for_isoseq_use_aug\t54\n",
      "\n",
      "\n",
      "at9744\n",
      "loci at input/at9744_loci.bed\n",
      "N loci: 34000\n",
      "Added 32498 genes from input/at9744.augustus.gff (0 missing)\n",
      "Added 26999 genes from input/at9744.liftoff.protonly.gff (0 missing)\n",
      "Added 14827 genes from input/at9744.pasa.fix.gff (0 missing)\n",
      "Added 17087 genes from input/at9744.tama-v2.fix.gff (0 missing)\n",
      "Now scan over loci...\n",
      "Now write GFFs...\n",
      "DONE! Stats follow\n",
      "\n",
      "aug_liftoff_pasa_cds_agree_use_pasa\t11360\n",
      "only_aug\t7230\n",
      "aug_liftoff_cds_agree_use_liftoff\t5092\n",
      "aug_liftoff_cds_disagree_use_aug\t3335\n",
      "only_liftoff\t1957\n",
      "TOO_HARD\t1695\n",
      "pasa_liftoff_cds_agree_use_pasa\t1569\n",
      "liftoff_augustus_agree_subgene_isoseq_merges_use_liftoff\t944\n",
      "pasa_cds_different_use_pasa\t814\n",
      "aug_pasa_cds_agree_use_pasa\t566\n",
      "aug_pasa_cds_disagree_but_cds_too_long_for_isoseq_use_aug\t52\n",
      "aug_liftoff_pasa_agree_subgene_tama_merges_use_pasa\t50\n",
      "\n",
      "\n",
      "at9762\n",
      "loci at input/at9762_loci.bed\n",
      "N loci: 33739\n",
      "Added 32306 genes from input/at9762.augustus.gff (0 missing)\n",
      "Added 26872 genes from input/at9762.liftoff.protonly.gff (0 missing)\n",
      "Added 14085 genes from input/at9762.pasa.fix.gff (0 missing)\n",
      "Added 16157 genes from input/at9762.tama-v2.fix.gff (0 missing)\n",
      "Now scan over loci...\n",
      "Now write GFFs...\n",
      "DONE! Stats follow\n",
      "\n",
      "aug_liftoff_pasa_cds_agree_use_pasa\t10652\n",
      "only_aug\t7174\n",
      "aug_liftoff_cds_agree_use_liftoff\t5621\n",
      "aug_liftoff_cds_disagree_use_aug\t3539\n",
      "only_liftoff\t1953\n",
      "TOO_HARD\t1565\n",
      "pasa_liftoff_cds_agree_use_pasa\t1488\n",
      "liftoff_augustus_agree_subgene_isoseq_merges_use_liftoff\t820\n",
      "pasa_cds_different_use_pasa\t808\n",
      "aug_pasa_cds_agree_use_pasa\t600\n",
      "aug_liftoff_pasa_agree_subgene_tama_merges_use_pasa\t76\n",
      "aug_pasa_cds_disagree_but_cds_too_long_for_isoseq_use_aug\t48\n",
      "\n",
      "\n",
      "at9806\n",
      "loci at input/at9806_loci.bed\n",
      "N loci: 33132\n",
      "Added 31675 genes from input/at9806.augustus.gff (0 missing)\n",
      "Added 26962 genes from input/at9806.liftoff.protonly.gff (0 missing)\n",
      "Added 13998 genes from input/at9806.pasa.fix.gff (0 missing)\n",
      "Added 15780 genes from input/at9806.tama-v2.fix.gff (0 missing)\n",
      "Now scan over loci...\n",
      "Now write GFFs...\n",
      "DONE! Stats follow\n",
      "\n",
      "aug_liftoff_pasa_cds_agree_use_pasa\t10727\n",
      "only_aug\t6574\n",
      "aug_liftoff_cds_agree_use_liftoff\t5874\n",
      "aug_liftoff_cds_disagree_use_aug\t3537\n",
      "only_liftoff\t1961\n",
      "pasa_liftoff_cds_agree_use_pasa\t1530\n",
      "TOO_HARD\t1376\n",
      "pasa_cds_different_use_pasa\t766\n",
      "liftoff_augustus_agree_subgene_isoseq_merges_use_liftoff\t740\n",
      "aug_pasa_cds_agree_use_pasa\t462\n",
      "aug_liftoff_pasa_agree_subgene_tama_merges_use_pasa\t83\n",
      "aug_pasa_cds_disagree_but_cds_too_long_for_isoseq_use_aug\t50\n",
      "\n",
      "\n",
      "at9830\n",
      "loci at input/at9830_loci.bed\n",
      "N loci: 33496\n",
      "Added 31933 genes from input/at9830.augustus.gff (0 missing)\n",
      "Added 26860 genes from input/at9830.liftoff.protonly.gff (0 missing)\n",
      "Added 14338 genes from input/at9830.pasa.fix.gff (0 missing)\n",
      "Added 16203 genes from input/at9830.tama-v2.fix.gff (0 missing)\n",
      "Now scan over loci...\n",
      "Now write GFFs...\n",
      "DONE! Stats follow\n",
      "\n",
      "aug_liftoff_pasa_cds_agree_use_pasa\t11020\n",
      "only_aug\t6848\n",
      "aug_liftoff_cds_agree_use_liftoff\t5610\n",
      "aug_liftoff_cds_disagree_use_aug\t3501\n",
      "only_liftoff\t1969\n",
      "pasa_liftoff_cds_agree_use_pasa\t1535\n",
      "TOO_HARD\t1445\n",
      "pasa_cds_different_use_pasa\t737\n",
      "liftoff_augustus_agree_subgene_isoseq_merges_use_liftoff\t666\n",
      "aug_pasa_cds_agree_use_pasa\t541\n",
      "aug_liftoff_pasa_agree_subgene_tama_merges_use_pasa\t70\n",
      "aug_pasa_cds_disagree_but_cds_too_long_for_isoseq_use_aug\t48\n",
      "\n",
      "\n",
      "at9847\n",
      "loci at input/at9847_loci.bed\n",
      "N loci: 33564\n",
      "Added 32154 genes from input/at9847.augustus.gff (0 missing)\n",
      "Added 26859 genes from input/at9847.liftoff.protonly.gff (0 missing)\n",
      "Added 14728 genes from input/at9847.pasa.fix.gff (0 missing)\n",
      "Added 16943 genes from input/at9847.tama-v2.fix.gff (0 missing)\n",
      "Now scan over loci...\n",
      "Now write GFFs...\n",
      "DONE! Stats follow\n",
      "\n",
      "aug_liftoff_pasa_cds_agree_use_pasa\t11150\n",
      "only_aug\t7006\n",
      "aug_liftoff_cds_agree_use_liftoff\t5105\n",
      "aug_liftoff_cds_disagree_use_aug\t3269\n",
      "only_liftoff\t1932\n",
      "TOO_HARD\t1652\n",
      "pasa_liftoff_cds_agree_use_pasa\t1612\n",
      "liftoff_augustus_agree_subgene_isoseq_merges_use_liftoff\t1025\n",
      "pasa_cds_different_use_pasa\t852\n",
      "aug_pasa_cds_agree_use_pasa\t555\n",
      "aug_liftoff_pasa_agree_subgene_tama_merges_use_pasa\t74\n",
      "aug_pasa_cds_disagree_but_cds_too_long_for_isoseq_use_aug\t52\n",
      "\n",
      "\n",
      "at9852\n",
      "loci at input/at9852_loci.bed\n",
      "N loci: 32924\n",
      "Added 31642 genes from input/at9852.augustus.gff (0 missing)\n",
      "Added 26816 genes from input/at9852.liftoff.protonly.gff (0 missing)\n",
      "Added 13962 genes from input/at9852.pasa.fix.gff (0 missing)\n",
      "Added 15879 genes from input/at9852.tama-v2.fix.gff (0 missing)\n",
      "Now scan over loci...\n",
      "Now write GFFs...\n",
      "DONE! Stats follow\n",
      "\n",
      "aug_liftoff_pasa_cds_agree_use_pasa\t10734\n",
      "only_aug\t6524\n",
      "aug_liftoff_cds_agree_use_liftoff\t5738\n",
      "aug_liftoff_cds_disagree_use_aug\t3479\n",
      "only_liftoff\t1884\n",
      "TOO_HARD\t1490\n",
      "pasa_liftoff_cds_agree_use_pasa\t1461\n",
      "liftoff_augustus_agree_subgene_isoseq_merges_use_liftoff\t858\n",
      "pasa_cds_different_use_pasa\t745\n",
      "aug_pasa_cds_agree_use_pasa\t510\n",
      "aug_liftoff_pasa_agree_subgene_tama_merges_use_pasa\t60\n",
      "aug_pasa_cds_disagree_but_cds_too_long_for_isoseq_use_aug\t50\n",
      "\n",
      "\n",
      "at9879\n",
      "loci at input/at9879_loci.bed\n",
      "N loci: 34209\n",
      "Added 32643 genes from input/at9879.augustus.gff (0 missing)\n",
      "Added 26870 genes from input/at9879.liftoff.protonly.gff (0 missing)\n",
      "Added 32611 genes from input/at9879.pasa.fix.gff (0 missing)\n",
      "Added 16274 genes from input/at9879.tama-v2.fix.gff (0 missing)\n",
      "Now scan over loci...\n",
      "Now write GFFs...\n",
      "DONE! Stats follow\n",
      "\n",
      "aug_liftoff_pasa_cds_agree_use_pasa\t16281\n",
      "aug_pasa_cds_agree_use_pasa\t11766\n",
      "only_liftoff\t1966\n",
      "TOO_HARD\t1556\n",
      "pasa_liftoff_cds_agree_use_pasa\t1488\n",
      "pasa_cds_different_use_pasa\t848\n",
      "aug_liftoff_pasa_agree_subgene_tama_merges_use_pasa\t746\n",
      "aug_pasa_cds_disagree_but_cds_too_long_for_isoseq_use_aug\t46\n",
      "liftoff_augustus_agree_subgene_isoseq_merges_use_liftoff\t20\n",
      "\n",
      "\n",
      "at9883\n",
      "loci at input/at9883_loci.bed\n",
      "N loci: 33811\n",
      "Added 32338 genes from input/at9883.augustus.gff (0 missing)\n",
      "Added 26944 genes from input/at9883.liftoff.protonly.gff (0 missing)\n",
      "Added 13990 genes from input/at9883.pasa.fix.gff (0 missing)\n",
      "Added 15734 genes from input/at9883.tama-v2.fix.gff (0 missing)\n",
      "Now scan over loci...\n",
      "Now write GFFs...\n",
      "DONE! Stats follow\n",
      "\n",
      "aug_liftoff_pasa_cds_agree_use_pasa\t10694\n",
      "only_aug\t7186\n",
      "aug_liftoff_cds_agree_use_liftoff\t5949\n",
      "aug_liftoff_cds_disagree_use_aug\t3506\n",
      "only_liftoff\t1953\n",
      "pasa_liftoff_cds_agree_use_pasa\t1524\n",
      "TOO_HARD\t1380\n",
      "pasa_cds_different_use_pasa\t798\n",
      "liftoff_augustus_agree_subgene_isoseq_merges_use_liftoff\t728\n",
      "aug_pasa_cds_agree_use_pasa\t515\n",
      "aug_liftoff_pasa_agree_subgene_tama_merges_use_pasa\t62\n",
      "aug_pasa_cds_disagree_but_cds_too_long_for_isoseq_use_aug\t44\n",
      "\n",
      "\n",
      "at9900\n",
      "loci at input/at9900_loci.bed\n",
      "N loci: 33472\n",
      "Added 31856 genes from input/at9900.augustus.gff (0 missing)\n",
      "Added 26865 genes from input/at9900.liftoff.protonly.gff (0 missing)\n",
      "Added 14796 genes from input/at9900.pasa.fix.gff (0 missing)\n",
      "Added 16754 genes from input/at9900.tama-v2.fix.gff (0 missing)\n",
      "Now scan over loci...\n",
      "Now write GFFs...\n",
      "DONE! Stats follow\n",
      "\n",
      "aug_liftoff_pasa_cds_agree_use_pasa\t11241\n",
      "only_aug\t6785\n",
      "aug_liftoff_cds_agree_use_liftoff\t5210\n",
      "aug_liftoff_cds_disagree_use_aug\t3375\n",
      "only_liftoff\t1955\n",
      "pasa_liftoff_cds_agree_use_pasa\t1609\n",
      "TOO_HARD\t1562\n",
      "pasa_cds_different_use_pasa\t839\n",
      "liftoff_augustus_agree_subgene_isoseq_merges_use_liftoff\t765\n",
      "aug_pasa_cds_agree_use_pasa\t563\n",
      "aug_liftoff_pasa_agree_subgene_tama_merges_use_pasa\t74\n",
      "aug_pasa_cds_disagree_but_cds_too_long_for_isoseq_use_aug\t48\n"
     ]
    }
   ],
   "source": [
    "for acc in \"${dl20[@]}\"\n",
    "do\n",
    "\n",
    "    echo\n",
    "    echo\n",
    "    echo ${acc}\n",
    "    grep 'Note=protein_coding_gene' input/${acc}.scaffolds-v2.3_liftoff_polished.gff3_polished.fix.gff3  | grep -Po \"\\tID=[^;]+\" | cut -f 2 -d = >input/${acc}_tairpconly.txt \n",
    "    grep -f input/${acc}_tairpconly.txt input/${acc}.scaffolds-v2.3_liftoff_polished.gff3_polished.fix.gff3 > input/${acc}.liftoff.protonly.gff\n",
    "\n",
    "    files=( input/${acc}.augustus.gff  input/${acc}.liftoff.protonly.gff input/${acc}.pasa.fix.gff input/${acc}.tama-v2.fix.gff)\n",
    "    \n",
    "    for file in \"${files[@]}\"\n",
    "    do\n",
    "        (head -n 1 $file | grep -q '^##gff-version') || sed -i -e '1 i\\##gff-version 3' $file\n",
    "    done\n",
    "    cat \"${files[@]}\" |\n",
    "        sortBed -i - |\n",
    "        mergeBed -i - -s -c 7 -o distinct >input/${acc}_loci.bed\n",
    "        \n",
    "    \n",
    "    grep $acc ../assembly-and-annotation/output/04_analyses_with_fixed_at9879/prep_for_decision_making/All_nlrs_list.txt | awk '{printf(\"ID=%s$\\n\",  $3)}' | sort -u -o ${acc}.nlraugids.txt\n",
    "    grep -f ${acc}.nlraugids.txt input/${acc}.augustus.gff  |\n",
    "        sortBed -i - |\n",
    "        intersectBed -wa -a <(awk 'BEGIN{IFS=OFS=\"\\t\"}{print $1, $2, $3, \".\", \"0\", $4}' < input/${acc}_loci.bed) -b - -s |\n",
    "        awk 'BEGIN{IFS=OFS=\"\\t\"}{print $1, $2, $3, $6}' > input/${acc}_nlrloci.bed\n",
    "    \n",
    "    python3 gffuuuu.py \\\n",
    "        --sad output/${acc}_sad.gff3 \\\n",
    "        --decisions output/${acc}_decisions.bed \\\n",
    "        --happy output/${acc}_happy.gff3 \\\n",
    "        --loci input/${acc}_loci.bed \\\n",
    "        \"${files[@]}\"\n",
    "    \n",
    "    rm -fr ~/public_html/gffuuu/${acc}/\n",
    "    mkdir -p ~/public_html/gffuuu/${acc}/\n",
    "    blsl genigvjs --outdir ~/public_html/gffuuu/${acc}/  \\\n",
    "                --reference ~/dl20/assembly-and-annotation/output/01_assembly/01_pansn-named/${acc}.scaffolds-v2.3.fasta \\\n",
    "                \"${files[@]}\" \\\n",
    "                output/${acc}_sad.gff3 \\\n",
    "                output/${acc}_decisions.bed \\\n",
    "                output/${acc}_happy.gff3\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "b2846641-9c7a-4e2e-9d39-cd6310fb54c6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "at6923\n",
      "\n",
      "\n",
      "at6929\n",
      "\n",
      "\n",
      "at7143\n",
      "\n",
      "\n",
      "at8285\n",
      "\n",
      "\n",
      "at9104\n",
      "\n",
      "\n",
      "at9336\n",
      "\n",
      "\n",
      "at9503\n",
      "\n",
      "\n",
      "at9578\n",
      "\n",
      "\n",
      "at9744\n",
      "\n",
      "\n",
      "at9762\n",
      "\n",
      "\n",
      "at9806\n",
      "\n",
      "\n",
      "at9830\n",
      "\n",
      "\n",
      "at9847\n",
      "\n",
      "\n",
      "at9852\n",
      "\n",
      "\n",
      "at9879\n",
      "\n",
      "\n",
      "at9883\n",
      "\n",
      "\n",
      "at9900\n"
     ]
    }
   ],
   "source": [
    "for acc in \"${dl20[@]}\"\n",
    "do\n",
    "\n",
    "    echo\n",
    "    echo\n",
    "    echo ${acc}\n",
    "   \n",
    "    grep \"^$acc\" \\\n",
    "        /ebio/abt6_projects7/dliso/dlis/leon_pipeline/output/v2/collapse_tama/${acc}_augustus_transdecoder/${acc}_transcripts.fasta.transdecoder.genome.gff3 \\\n",
    "     | sed -e '1 i\\##gff-version 3' \\\n",
    "     >input/${acc}.transdecoder.gff\n",
    "    files=( input/${acc}.augustus.gff  input/${acc}.liftoff.protonly.gff input/${acc}.pasa.fix.gff input/${acc}.tama-v2.fix.gff\n",
    "    input/${acc}.transdecoder.gff)\n",
    "    \n",
    "    rm -fr ~/public_html/gffuuu/${acc}/\n",
    "    mkdir -p ~/public_html/gffuuu/${acc}/\n",
    "    blsl genigvjs --outdir ~/public_html/gffuuu/${acc}/  \\\n",
    "                --reference ~/dl20/assembly-and-annotation/output/01_assembly/01_pansn-named/${acc}.scaffolds-v2.3.fasta \\\n",
    "                \"${files[@]}\" \\\n",
    "                output/${acc}_sad.gff3 \\\n",
    "                output/${acc}_decisions.bed \\\n",
    "                output/${acc}_happy.gff3\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "ab6e189f-ef2e-4442-b11d-e778bbe7bc61",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "loci at input/at6137_loci.bed\n",
      "N loci: 33514\n",
      "Added 31972 genes from input/at6137.augustus.gff (0 missing)\n",
      "Added 26838 genes from input/at6137.liftoff.protonly.gff (0 missing)\n",
      "Now scan over loci...\n",
      "Now write GFFs...\n",
      "DONE! Stats follow\n",
      "\n",
      "aug_liftoff_cds_agree_use_liftoff\t17936\n",
      "only_aug\t7128\n",
      "aug_liftoff_cds_disagree_use_aug\t5752\n",
      "only_liftoff\t1972\n",
      "TOO_HARD\t699\n",
      "liftoff_augustus_agree_subgene_isoseq_merges_use_liftoff\t143\n"
     ]
    }
   ],
   "source": [
    "for acc in at6137\n",
    "do\n",
    "    grep 'Note=protein_coding_gene' input/${acc}.scaffolds-v2.3_liftoff_polished.gff3_polished.fix.gff3  | grep -Po \"\\tID=[^;]+\" | cut -f 2 -d = >input/${acc}_tairpconly.txt \n",
    "    grep -f input/${acc}_tairpconly.txt input/${acc}.scaffolds-v2.3_liftoff_polished.gff3_polished.fix.gff3  > input/${acc}.liftoff.protonly.gff\n",
    "    \n",
    "    annots=( input/${acc}.augustus.gff  input/${acc}.liftoff.protonly.gff )\n",
    "\n",
    "    for file in \"${annots[@]}\"\n",
    "    do\n",
    "        (head -n 1 $file | grep -q '^##gff-version') || sed -i -e '1 i\\##gff-version 3' $file\n",
    "    done\n",
    "    \n",
    "    cat \"${annots[@]}\" |\n",
    "        sortBed -i - |\n",
    "        mergeBed -i - -s -c 7 -o distinct >input/${acc}_loci.bed\n",
    "        \n",
    "    python3 gffuuuu.py \\\n",
    "        --sad output/${acc}_sad.gff3 \\\n",
    "        --decisions output/${acc}_decisions.bed \\\n",
    "        --happy output/${acc}_happy.gff3 \\\n",
    "        --loci input/${acc}_loci.bed \\\n",
    "        \"${annots[@]}\"\n",
    "    \n",
    "    \n",
    "    rm -fr ~/public_html/gffuuu/${acc}/\n",
    "    mkdir -p ~/public_html/gffuuu/${acc}/\n",
    "    blsl genigvjs --outdir ~/public_html/gffuuu/${acc}/  \\\n",
    "                --reference ~/dl20/assembly-and-annotation/output/01_assembly/01_pansn-named/${acc}.scaffolds-v2.3.fasta \\\n",
    "                \"${files[@]}\" \\\n",
    "                output/${acc}_sad.gff3 \\\n",
    "                output/${acc}_decisions.bed \\\n",
    "                output/${acc}_happy.gff3\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "ddc6c6b4-66fe-4433-a244-ff66f71a35a7",
   "metadata": {},
   "outputs": [],
   "source": [
    "sed -i -e 's/\\tNone\\t/\\t.\\t/g' output/*.gff3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "4903bded-7aeb-47be-aab8-7b8e58a8b31f",
   "metadata": {},
   "outputs": [],
   "source": [
    "for acc in \"${dl20[@]}\" at6137\n",
    "do\n",
    "    intersectBed -wa -a output/${acc}_decisions.bed \\\n",
    "        -b ../assembly-and-annotation/output/04_analyses_with_fixed_at9879/NLRs_list/${acc}.nlrs.genes.augustus.bed \\\n",
    "        -s \\\n",
    "        > output/${acc}_nlr_decisions.bed\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c4b64f62-e01b-4280-a0b3-3775997bb044",
   "metadata": {},
   "outputs": [],
   "source": [
    "d"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
