{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1ab4fefc-280b-4ef7-b655-62935ed69d09",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "conda activate dl20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "0aefed38-ae6a-4b2f-82c8-0bb3c9982315",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using standard /tmp/global2/kmurray/conda/envs/dl20/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/kmurray/conda/envs/dl20/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/kmurray/conda/envs/dl20/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/kmurray/conda/envs/dl20/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/kmurray/conda/envs/dl20/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 761055\n",
      "=> Number of comment lines: 33697\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 727358\n",
      "=> Number of feature type (3rd column): 9\n",
      "\t* Level1: 3 => pseudogene pseudogenic_region gene\n",
      "\t* level2: 2 => pseudogenic_transcript mRNA\n",
      "\t* level3: 4 => five_prime_UTR CDS three_prime_UTR exon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 87 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 12 seconds ------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "2671 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 12 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "6 UTRs created that were missing\n",
      "33 UTRs locations modified that were wrong\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "We fixed 15 wrong level1 location cases\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "Lets remove isoform at9852_G063230_pseudogenic_transcript02\n",
      "Lets remove isoform at9852_G235170_mRNA02\n",
      "Lets remove isoform at9852_G032580_mRNA03\n",
      "Lets remove isoform at9852_G197270_mRNA03\n",
      "Lets remove isoform at9852_G197270_mRNA02\n",
      "Lets remove isoform at9852_G049360_mRNA02\n",
      "Lets remove isoform at9852_G176660_mRNA02\n",
      "Lets remove isoform at9852_G114160_mRNA03\n",
      "Lets remove isoform at9852_G026030_mRNA02\n",
      "Lets remove isoform at9852_G165600_mRNA02\n",
      "Lets remove isoform at9852_G047230_mRNA02\n",
      "11 identical isoforms removed\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 53 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 140 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/kmurray/conda/envs/dl20/bin/agat_convert_sp_gxf2gxf.pl -g post-curation/at9852_hodgepodgemerged.gff3 -o tmp/9852.hp.agat.gff3\n",
      "Job done in 176 seconds\n"
     ]
    }
   ],
   "source": [
    "rm -f tmp/9852.hp.agat.gff3\n",
    "agat_convert_sp_gxf2gxf.pl \\\n",
    "    -g post-curation/at9852_hodgepodgemerged.gff3 \\\n",
    "    -o tmp/9852.hp.agat.gff3\n",
    "sort -i tmp/9852.hp.agat.gff3 -k 1,1 -k 4,4n -s -o tmp/9852.hp.agat.sort.gff3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4e3e341f-8419-443c-b857-071fa54892bd",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
