---
title: "anno_TE_intacts"
output: html_document
---



```{r}

library(patchwork)
library(dplyr)
library(ggplot2)

TE_colors <- c("DNA/CACTA" = "#d28bc6",
               "LTR/Ty1" = "#6fb445",
               "LTR/Ty3" = "#9754d1",
               "LTR/unknown" = "#cf46ac",
               "DNA/HAT" = "#b86d54",
               "RC/Helitron" = "#a94a70",
               "LINE" = "#4ea98a",
               "DNA/Mutator" = "#725b9a",
               "DNA/Harbinger" = "#657c38",
               "DNA/Mariner" = "#bf9740",
               "DNA"= "#6474d8",
               "MITE/CACTA" = "#4f8ec1",
               "MITE/Harbinger"  = "#4f8ec1",
               "MITE/Mariner"  = "#4f8ec1",
               "MITE/Mutator"  = "#4f8ec1",
               "Unassigned"  = "#f44336", 
               "SINE"  = "#666666")
```


# Load the main TE intacts file.
```{r}

pangenome_TE_intacts <- vroom("/ebio/abt6_projects7/diffLines_20/data/TE_annotation/pangenome_TEannotation_intacts.gff3", 
                                 skip = 19,
                                 col_names = FALSE)

pangenome_TE_intacts <- parse_TE_GFF(pangenome_TE_intacts)
pangenome_TE_intacts <- parse_TE_intacts_GFF(pangenome_TE_intacts)

```




# AGE TES
```{bash}
#!/bin/bash
# LTR AGE
# First I ran a script to retrieve every pair of 3' and 5' LTR from  the annotation, 
# by  genome, and then, I pair them and align them  using mafft. 

cd /tmp/global2/acontreras/difflines_annex_adri/TE_annotation/output/FINAL_TE_annotation/split_by_accession/ltr_sequences/

pangenome="/tmp/global2/acontreras/difflines_annex_adri/TE_annotation/output/FINAL_TE_annotation/pangenome_onlychrom.fasta"
# Get an ID list for each genome annotation:
 for TEanno in *_TEintact.gff3; do
      name=$(basename $TEanno _TEintact.gff3) ; cat $TEanno  |
      awk 'BEGIN{OFS="\t"}{if($3 ~ /LTR/ && $9 ~ /[r,l]LTR/) print $9}' |
      cut -f2 -d ';' |
      sed 's/Parent=//' | uniq >  ltr_sequences/${name}_ltr_IDs.txt ;
done
cd ltr_sequences/
# Extract fasta files and alignm them using mafft :
for  genome_list in *_ltr_IDs.txt ;
  do gname=$(basename $genome_list _ltr_IDs.txt) ;
  mkdir -p ${gname}_ltrs/ ;
  for ID in $( cat $genome_list) ; do
    grep -w $ID ../${gname}_TEintact.gff3 |
    awk 'BEGIN{OFS="\t"}{if($9 ~ /[r,l]LTR/) print $0}' |
    bedtools getfasta -fi ${pangenome} -bed - |
    mafft --adjustdirection --auto - > ${gname}_ltrs/ltr_${ID}.aln
  done;
done
```


```{r}
# Lad the aligments
library(ape)
library(reshape2)
library(stringr)
library(plyr)

# Mutation rate used 
mutation_rate= 9.1 * 1E-9

genome_list <- c("at6137","at6923","at6929","at7143","at8285","at9104","at9336","at9503","at9578",
                 "at9744","at9762","at9806","at9830","at9847","at9852","at9879","at9883","at9900")

ltr_age_df_list = list()
for (a in 1:length(genome_list)){
              path <- paste0("/tmp/global2/acontreras/difflines_annex_adri/TE_annotation/output/FINAL_TE_annotation/TE_analisis/ltr_sequences/", genome_list[a],"_ltrs/")
              filenames=list.files(path, pattern='.aln')
              ages=data.frame(TEname=str_split_fixed(filenames, '\\.', 3)[,1])
              ages$Accession <- genome_list[a]
              ages$k2p=NA
              ages$raw=NA
              ages$tn93=NA
              for (i in 1:length(filenames)){
                          ltr=read.FASTA(paste0(path,filenames[i]))
                          if( !is.null(ltr) && length(ltr) > 1  ){
                                d=dist.dna(ltr, model='K80', gamma=F)
                                ages$k2p[i]=d
                                raw=dist.dna(ltr, model='raw', gamma=F)
                                ages$raw[i]=raw
                                tn93=dist.dna(ltr, model='tn93', gamma=F)
                                ages$tn93[i]=tn93
                                }
                          }
              ltr_age_df_list[[a]] <- assign(paste0("ltr_age_",genome_list[a]), ages)
}

ltr_age_all = do.call(rbind, ltr_age_df_list)
ltr_age_all$mya_k2p=ltr_age_all$k2p/(2*mutation_rate)/1E6
ltr_age_all$mya_raw=ltr_age_all$raw/(2*mutation_rate)/1E6
ltr_age_all$mya_tn93=ltr_age_all$tn93/(2*mutation_rate/1E6)

# Plot 
ltr_age_all %>%  
  ggplot(aes(x=mya_k2p)) + 
  geom_density() +
  scale_x_reverse() + 
  xlab("Mya") + 
  facet_wrap(~Accession) +
  theme_light() +
  theme(axis.text.x = element_text(angle=45, vjust=0.5, hjust=0.5), 
        panel.spacing = unit(1.2, "lines"),
        strip.background = element_rect(fill="white"),
        strip.text = element_text(color="grey40"))

# Plot numnbero of LTRs
ltr_age_all %>%  
  group_by(Accession) %>%  
  tally() %>%  
  ggplot(aes(x=reorder(Accession, n), y=n)) + 
  geom_col() + 
  xlab("") +
  ylab("# intact TEs") + 
  coord_flip() + 
  theme_light()


# Transform IDs back to pangenome notation and merge this DF with pangenome_TE_intacts:
ltr_age_all <- ltr_age_all %>%  mutate(ID = str_remove(TEname, "ltr_"))
pangenome_TE_intacts <- dplyr::left_join(pangenome_TE_intacts, ltr_age_all[,c(1:3,6,9)], by=c("Accession", "ID"))
pangenome_TE_intacts <- dplyr::left_join(pangenome_TE_intacts, ltr_age_all[,c(1:3,6,9)], by=c("Accession", "ID" == "Parent"))

# LTR sizes:

pangenome_TE_intacts %>% filter(grepl(".LTR", ID))  %>%  
  ggplot(aes(x=length)) + 
  geom_histogram() + 
  facet_wrap(~Accession) +
  theme_light() +
  xlab("LTR length of TEs") + 
  theme(axis.text.x = element_text(angle=45, vjust=0.5, hjust=0.5), 
        panel.spacing = unit(1.2, "lines"),
        strip.background = element_rect(fill="white"),
        strip.text = element_text(color="grey40"))


# FUn thing:

dplyr::left_join(pangenome_TE_intacts, ltr_age_all[,c(1:3,6,9)], by=c("Accession", "Parent" = "ID")) %>%  
    filter(grepl(".LTR", ID)) %>% 
    ggplot(aes(x=length, y=mya_k2p.y, color=Accession)) + 
    geom_point(alpha=0.1) + 
    xlab("LTR size of TEs") +  
    ylab("MYA") +
    theme_light(base_size = 15) + facet_wrap(~Accession) +  
    theme(axis.text.x = element_text(angle=45, vjust=0.5, hjust=0.5), 
          panel.spacing = unit(1.2, "lines"),
          strip.background = element_rect(fill="white"),
          strip.text = element_text(color="grey40"))

```




