{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "7b807d2e-aa58-4d2b-afa4-c26fcd0fb5e2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "source ~/.bashrc\n",
    "conda activate pggb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "15649021-e3cb-41da-bc4a-9a2e2eedfbce",
   "metadata": {},
   "outputs": [],
   "source": [
    "which odgi"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a00bf65d-0e4f-4dd2-b467-a0e514ff8645",
   "metadata": {},
   "source": [
    "# Chromosome 3\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "21cbdac9-b404-4382-8643-4ef6f621b011",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "\n",
    "mkdir -p chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og\n",
    "mkdir -p chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_gfa\n",
    "mkdir -p chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_allPathCoordinates\n",
    "mkdir -p chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_viz\n",
    "mkdir -p chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_pav\n",
    "mkdir -p chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_similarity\n",
    "mkdir -p chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_heaps\n",
    "mkdir -p chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_panacus\n",
    "mkdir -p chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_genebed\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "24b60d35-cb5e-4fcf-b396-826cd39a243d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "inputOdgi=chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr3.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "819e5305-dc07-46fb-a39e-a772a0a1948b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "at6137_1_chr3:0-26194750\n",
      "at6923_1_chr3:0-28738137\n",
      "at6929_1_chr3:0-25712594\n",
      "at7143_1_chr3:0-26044309\n",
      "at8285_1_chr3:0-26796854\n",
      "at9104_1_chr3:0-27458818\n",
      "at9336_1_chr3:0-27060997\n",
      "at9503_1_chr3:0-28010412\n",
      "at9578_1_chr3:0-26348032\n",
      "at9744_1_chr3:0-27175644\n",
      "at9762_1_chr3:0-27262250\n",
      "at9806_1_chr3:0-27369393\n",
      "at9830_1_chr3:0-27041555\n",
      "at9847_1_chr3:0-28044082\n",
      "at9852_1_chr3:0-26358531\n",
      "at9879_1_chr3:0-27173682\n",
      "at9883_1_chr3:0-26028663\n",
      "at9900_1_chr3:0-27314136\n",
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "odgi paths -L \\\n",
    "-i $inputOdgi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "e0b872d9-234d-4705-a871-80357e1bcd04",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [2.91685e+07, 2.97161e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 547.583\n",
      "[odgi::viz] image width: 1000\n",
      "bash: -t: command not found\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 8.28e-01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 2.01e+00/s elapsed: 00:00:00:08 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 2.27e+00/s elapsed: 00:00:00:07 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.17e+05/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 3.72e+00/s elapsed: 00:00:00:13 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.09e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/RPP4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [2.31108e+07, 2.34623e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 351.441\n",
      "[odgi::viz] image width: 1000\n",
      "bash: -t: command not found\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 8.45e-01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 5.69e+00/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 6.18e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 3 (max 3) 100.00% @ 5.61e+00/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 6.08e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 7.97e+00/s elapsed: 00:00:00:06 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.13e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster1.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [2.47871e+07, 2.49134e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 126.33\n",
      "[odgi::viz] image width: 1000\n",
      "bash: -t: command not found\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 8.03e-01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 3.28e+00/s elapsed: 00:00:00:05 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 3.44e+00/s elapsed: 00:00:00:05 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 3 (max 3) 100.00% @ 3.41e+00/s elapsed: 00:00:00:05 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 4.00e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 7.20e+00/s elapsed: 00:00:00:07 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.14e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster2.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [2.55308e+07, 2.56331e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 102.305\n",
      "[odgi::viz] image width: 1000\n",
      "bash: -t: command not found\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.04e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 6.23e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 6.41e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 3.07e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 9.26e+00/s elapsed: 00:00:00:05 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.17e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster3.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [2.73797e+07, 2.74543e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 74.674\n",
      "[odgi::viz] image width: 1000\n",
      "bash: -t: command not found\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.02e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 6.79e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 5.92e+00/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.85e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 6.07e+00/s elapsed: 00:00:00:08 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.16e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster4.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [2.75558e+07, 2.75746e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 18.807\n",
      "[odgi::viz] image width: 1000\n",
      "bash: -t: command not found\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.02e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 5.44e+00/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 5.62e+00/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 9.07e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 8.08e+00/s elapsed: 00:00:00:06 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster5.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "/mpcdf/soft/SLE_15/packages/x86_64/anaconda/3/2021.05/lib/python3.8/site-packages/scipy/optimize/minpack.py:828: OptimizeWarning: Covariance of the parameters could not be estimated\n",
      "  warnings.warn('Covariance of the parameters could not be estimated',\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.07674e+07, 3.08562e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 88.83\n",
      "[odgi::viz] image width: 1000\n",
      "bash: -t: command not found\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.00e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 5.88e+00/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 6.39e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 2.40e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 8.73e+00/s elapsed: 00:00:00:05 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.17e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster6.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.29038e+07, 3.2931e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 27.216\n",
      "[odgi::viz] image width: 1000\n",
      "bash: -t: command not found\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 9.90e-01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 5.99e+00/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 6.03e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 4.13e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 9.05e+00/s elapsed: 00:00:00:05 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster7.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.49525e+07, 3.49732e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 20.665\n",
      "[odgi::viz] image width: 1000\n",
      "bash: -t: command not found\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 9.17e-01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 7.57e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 7.45e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 3.05e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.09e+01/s elapsed: 00:00:00:04 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster8.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.53677e+07, 3.55427e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 175.014\n",
      "[odgi::viz] image width: 1000\n",
      "bash: -t: command not found\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 7.40e-01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 5.89e+00/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 6.48e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.52e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 9.25e+00/s elapsed: 00:00:00:05 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.17e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster9.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.8365e+07, 3.83791e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 14.065\n",
      "[odgi::viz] image width: 1000\n",
      "bash: -t: command not found\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.21e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 6.97e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 7.29e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 5.08e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 6.76e+00/s elapsed: 00:00:00:07 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.17e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster10.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.95797e+07, 3.963e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 50.285\n",
      "[odgi::viz] image width: 1000\n",
      "bash: -t: command not found\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.60e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 6.16e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 4.32e+00/s elapsed: 00:00:00:04 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.34e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 7.67e+00/s elapsed: 00:00:00:06 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::paths] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "[odgi::heaps] error: the given file \"chr4_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/cluster11.region.optimized_sorted.og\" does not exist. Please specify an existing input file in ODGI format via -i=[FILE], --idx=[FILE].\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "while read a b; \n",
    "do\n",
    "\n",
    "    r=${a}\n",
    "    outprefix=${b}.region\n",
    "\n",
    "    ########################## view the pangenome graph of the region ######################\n",
    "    odgi viz \\\n",
    "    -i $inputOdgi \\\n",
    "    -o chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_viz/${outprefix}.bm.png \\\n",
    "    -x 1000 \\\n",
    "    -bm \\\n",
    "    -t 20 \\\n",
    "    -P \\\n",
    "    -r ${r}\n",
    "    ########################## Node Strandedness ############################################\n",
    "    odgi viz \\\n",
    "    -i $inputOdgi \\\n",
    "    -o chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_viz/${outprefix}.z.png \\\n",
    "    -x 1000 \\\n",
    "    -z \\\n",
    "    -t 20 \\\n",
    "    -P \\\n",
    "    -r ${r}\n",
    "    \n",
    "    ############################################ extract the region ############################################\n",
    "    odgi extract \\\n",
    "    -i $inputOdgi \\\n",
    "    -r ${r} \\\n",
    "    --threads 24 -P \\\n",
    "    -d 100000 \\\n",
    "    -c 1  \\\n",
    "    -o chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.og \n",
    "    \n",
    "    ########################### get the coordinates in all the accessions ########################################\n",
    "    odgi paths -L \\\n",
    "    -i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.og  \\\n",
    "    > chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_allPathCoordinates/${outprefix}.allPathsCoordinates.txt\n",
    "    \n",
    "    \n",
    "    ########################### for bandage plot #################################################################\n",
    "    odgi view -i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.og  \\\n",
    "    -g > chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_gfa/${outprefix}.gfa\n",
    "    \n",
    "    ########################### Individual path-based PAV comparisons ############################################\n",
    "    while read line;\n",
    "    do\n",
    "        r=${line}\n",
    "        echo $r|cut -d ':' -f1,2 >tmp1\n",
    "        echo $r|cut -d ':' -f3|sed \"s/-/\\t/g\" >tmp2\n",
    "        paste tmp1 tmp2 > \\\n",
    "        chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_allPathCoordinates/${outprefix}.${line}.bed && rm tmp1 && rm tmp2\n",
    "        \n",
    "        odgi pav \\\n",
    "        -i $inputOdgi \\\n",
    "        -t 24 \\\n",
    "        -b chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_allPathCoordinates/${outprefix}.${line}.bed > \\\n",
    "        chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_pav/${outprefix}.${line}.pav.txt\n",
    "        \n",
    "    done < chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_allPathCoordinates/${outprefix}.allPathsCoordinates.txt\n",
    "    \n",
    "    ########################### Node Jumps in Neighborhood ######################################################\n",
    "    ## for each node how many jumps it has taken\n",
    "    cat chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_gfa/${outprefix}.gfa \\\n",
    "    |grep \"^L\" \\\n",
    "    |awk '{print $1,$2,$4,$4-$2}'\\\n",
    "    |sed \"s/ /\\t/g\" \\\n",
    "    > chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_jumps/${outprefix}.nodeJumps.txt\n",
    "    \n",
    "    \n",
    "    ########################### Node Degrees ############################################################\n",
    "    odgi \\\n",
    "    degree \\\n",
    "    -i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.og \\\n",
    "    -t 24 \\\n",
    "    -d \\\n",
    "    -r ${r} > chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_jumps/${outprefix}.nodeDegrees.txt\n",
    "    \n",
    "    \n",
    "    ########################### Node Depth ###############################################################\n",
    "    odgi \\\n",
    "    depth \\\n",
    "    -i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.og \\\n",
    "    -t 24 \\\n",
    "    -d \\\n",
    "    -r ${r} > chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_jumps/${outprefix}.nodeDepth.txt\n",
    "    \n",
    "    ########################### Distance between different Paths (accessions) for the region #############\n",
    "    odgi \\\n",
    "    similarity \\\n",
    "    -i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.og \\\n",
    "    -t 24 \\\n",
    "    -d > chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_similarity/${outprefix}.distance.txt\n",
    "    \n",
    "    ###### Pangenome Graph Growth  Heaps' Law #########\n",
    "\n",
    "    odgi sort \\\n",
    "    -i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.og \\\n",
    "    -p Y \\\n",
    "    -O \\\n",
    "    -t 24 \\\n",
    "    -P \\\n",
    "    -o chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.optimized_sorted.og \n",
    "    \n",
    "    for i in {1..18}\n",
    "    do\n",
    "    seq -w 1 ${i} > chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_heaps/pathGroup${i}.groups.txt\n",
    "    done\n",
    "    \n",
    "    \n",
    "    for i in {1..18}\n",
    "    do\n",
    "    odgi paths -i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.optimized_sorted.og \\\n",
    "    -L|sort -R|head -n ${i} > pathGroup${i}.txt \n",
    "    done\n",
    "    \n",
    "    for i in {1..18}\n",
    "    do\n",
    "    paste pathGroup${i}.txt pathGroup${i}.groups.txt > tmp\n",
    "    mv tmp pathGroup${i}.txt\n",
    "    done\n",
    "    \n",
    "    for i in {1..18}\n",
    "    do\n",
    "        odgi \\\n",
    "        heaps \\\n",
    "        -i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.optimized_sorted.og \\\n",
    "        -t 24 \\\n",
    "        -S \\\n",
    "        -d 2 \\\n",
    "        -n 100 \\\n",
    "        -p pathGroup${i}.txt \\\n",
    "        > chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_heaps/${outprefix}.pathGroup${i}.optimized_sorted.heaps.txt\n",
    "    done\n",
    "    \n",
    "    \n",
    "    ########################## Panacus ##############################\n",
    "    parentDir=chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended\n",
    "    \n",
    "    grep '^P' ${parentDir}/cluster_gfa/${outprefix}.gfa|cut -f2|sed \"s/_.*//g\" > ${parentDir}/cluster_panacus/${outprefix}.groupNames.txt\n",
    "    grep '^P' ${parentDir}/cluster_gfa/${outprefix}.gfa|cut -f2> ${parentDir}/cluster_panacus/${outprefix}.pathNames.txt\n",
    "    \n",
    "    paste ${parentDir}/cluster_panacus/${outprefix}.pathNames.txt \\\n",
    "    ${parentDir}/cluster_panacus/${outprefix}.groupNames.txt \\\n",
    "    > ${parentDir}/cluster_panacus/${outprefix}.groups.txt\n",
    "    \n",
    "    sed -i \"s/_1_chr3/#1#chr3/g\" ${parentDir}/cluster_panacus/${outprefix}.pathNames.txt \n",
    "    sed -i \"s/_1_chr3/#1#chr3/g\" ${parentDir}/cluster_panacus/${outprefix}.groups.txt\n",
    "    \n",
    "    \n",
    "    cat ${parentDir}/cluster_panacus/${outprefix}.pathNames.txt |cut -f1,3 -d \":\" > \\\n",
    "    ${parentDir}/cluster_panacus/${outprefix}.newPathNames.txt\n",
    "    \n",
    "    paste ${parentDir}/cluster_panacus/${outprefix}.pathNames.txt \\\n",
    "    ${parentDir}/cluster_panacus/${outprefix}.newPathNames.txt > ${parentDir}/cluster_panacus/tmp\n",
    "    \n",
    "    while read a b ; do sed -i \"s/${a}/${b}/\" ${parentDir}/cluster_panacus/${outprefix}.pathNames.txt ; done < ${parentDir}/cluster_panacus/tmp\n",
    "    while read a b ; do sed -i \"s/${a}/${b}/\" ${parentDir}/cluster_panacus/${outprefix}.groups.txt ; done < ${parentDir}/cluster_panacus/tmp\n",
    "    cp ${parentDir}/cluster_gfa/${outprefix}.gfa ${parentDir}/cluster_panacus/${outprefix}.ed.gfa\n",
    "    sed -i \"s/_1_chr3/#1#chr3/g\" ${parentDir}/cluster_panacus/${outprefix}.ed.gfa\n",
    "    \n",
    "    while read a b ; do sed -i \"s/${a}/${b}/g\" ${parentDir}/cluster_panacus/${outprefix}.ed.gfa ; done < ${parentDir}/cluster_panacus/tmp \n",
    "    \n",
    "    panacus histgrowth -t4 -q 0,1,0.5,0.1 -g ${parentDir}/cluster_panacus/${outprefix}.groups.txt \\\n",
    "    -s ${parentDir}/cluster_panacus/${outprefix}.pathNames.txt \\\n",
    "    ${parentDir}/cluster_panacus/${outprefix}.ed.gfa > ${parentDir}/cluster_panacus/${outprefix}.histgrowth.txt\n",
    "    \n",
    "    panacus-visualize -e ${parentDir}/cluster_panacus/${outprefix}.histgrowth.txt > ${parentDir}/cluster_panacus/${outprefix}.histgrowth.node.pdf\n",
    "\n",
    "done < chr3_cluster.neighborhoods.txt\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "raw",
   "id": "49dc456f-5551-41a1-80dd-ec6b7730e39d",
   "metadata": {},
   "source": [
    "odgi kmers \\\n",
    "-i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.og \\\n",
    "-t 24 \\\n",
    "-k 7 \\\n",
    "-t 24 -c > chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_similarity/test.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "8b649ab8-2b69-4f73-a522-7bf5fe1afa08",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "at9852_1_chr4:0-24203097:15114970-15264769 RPP4\n",
      "at9852_1_chr4:0-24203097:11476201-11576200 cluster1\n",
      "at9852_1_chr4:0-24203097:12318801-12419300 cluster2\n",
      "at9852_1_chr4:0-24203097:12787951-12838750 cluster3\n",
      "at9852_1_chr4:0-24203097:13902081-13922400 cluster4\n",
      "at9852_1_chr4:0-24203097:14001856-14014880 cluster5\n",
      "at9852_1_chr4:0-24203097:16095461-16116140 cluster6\n",
      "at9852_1_chr4:0-24203097:17230521-17251120 cluster7\n",
      "at9852_1_chr4:0-24203097:18826871-18839090 cluster8\n",
      "at9852_1_chr4:0-24203097:19165561-19244000 cluster9\n",
      "at9852_1_chr4:0-24203097:21647641-21659750 cluster10\n",
      "at9852_1_chr4:0-24203097:22696841-22721060 cluster11\n",
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "while read a b \n",
    "do\n",
    "echo $a $b\n",
    "done < chr3_cluster.neighborhoods.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "b2aed3ca-1bd1-4e65-8f0f-a8f6c282bd83",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (9.53422e-10 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 4.90e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (4.73011e-09 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 6.79e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (2.60062e-08 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 4.03e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (4.34933e-08 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 3.10e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (1.30706e-07 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 1.85e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (4.28305e-07 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 1.06e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (9.0779e-08 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 1.94e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (2.38886e-06 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 4.32e+06/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (3.50685e-06 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 3.76e+06/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (7.06582e-08 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 1.83e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (1.30612e-06 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 5.92e+06/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (2.0037e-07 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 1.42e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "while read a b \n",
    "do \n",
    "    r=${a}\n",
    "    outprefix=${b}.region\n",
    "    \n",
    "    ###### Pangenome Graph Growth  #########\n",
    "    \n",
    "    odgi sort \\\n",
    "    -i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.og \\\n",
    "    -p Y \\\n",
    "    -O \\\n",
    "    -t 24 \\\n",
    "    -P \\\n",
    "    -o chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.optimized_sorted.og \n",
    "    \n",
    "    for i in {1..18}\n",
    "    do\n",
    "        seq -w 1 ${i} > chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_heaps/pathGroup${i}.groups.txt\n",
    "    done\n",
    "    \n",
    "    \n",
    "    for i in {1..18}\n",
    "    do\n",
    "    odgi paths -i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.optimized_sorted.og \\\n",
    "    -L|sort -R|head -n ${i} > pathGroup${i}.txt \n",
    "    done\n",
    "    \n",
    "    for i in {1..18}\n",
    "    do\n",
    "    paste pathGroup${i}.txt pathGroup${i}.groups.txt > tmp\n",
    "    mv tmp pathGroup${i}.txt\n",
    "    done\n",
    "    \n",
    "    for i in {1..18}\n",
    "    do\n",
    "    \n",
    "        odgi \\\n",
    "        heaps \\\n",
    "        -i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.optimized_sorted.og \\\n",
    "        -t 24 \\\n",
    "        -S \\\n",
    "        -d 2 \\\n",
    "        -n 100 \\\n",
    "        -p pathGroup${i}.txt \\\n",
    "        > chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_heaps/${outprefix}.pathGroup${i}.optimized_sorted.heaps.txt\n",
    "    done\n",
    "\n",
    "done < chr3_cluster.neighborhoods.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "6a29ada9-59f0-468a-8019-fb95fe5d7803",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "#### PAV statistics for clusters ######\n",
    "parentDir=chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended\n",
    "\n",
    "while read a b \n",
    "do\n",
    "    cat ${parentDir}/cluster_pav/${b}.region.*pav.txt|cut -f1,5,6|sed '/pav/d'|sed \"s/$/\\t${b}/g\" > ${parentDir}/cluster_pav/${b}.region.tmp\n",
    "done < chr3_cluster.neighborhoods.txt\n",
    "\n",
    "cat ${parentDir}/cluster_pav/*.region.tmp > ${parentDir}/cluster_pav/all.clusters.pavs.txt\n",
    "awk '($1!=$2)' ${parentDir}/cluster_pav/all.clusters.pavs.txt > tmp && mv tmp  ${parentDir}/cluster_pav/all.clusters.pavs.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "6c04b4f9-2551-4462-a044-e1f1854d4091",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "rm all.clusters.pavs.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "b7564c99-aa5c-4d3f-b5ee-88fb83b928b6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "parentDir=chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended\n",
    "while read a b \n",
    "do\n",
    "cat ${parentDir}/cluster_similarity/${b}.region.distance.txt|cut -f1,2,3,4,5,6,7|sed '/distance/d'|sed \"s/$/\\t${b}/g\" > ${parentDir}/cluster_similarity/${b}.region.tmp\n",
    "done < chr3_cluster.neighborhoods.txt\n",
    "\n",
    "cat ${parentDir}/cluster_similarity/*.region.tmp > ${parentDir}/cluster_similarity/all.clusters.similarity.txt\n",
    "awk '($1!=$2)' ${parentDir}/cluster_similarity/all.clusters.similarity.txt > tmp && mv tmp  ${parentDir}/cluster_similarity/all.clusters.similarity.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "81930270-ba80-4088-a5c7-57324301f3c7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.67095e+07, 4.67204e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 10.882\n",
      "[odgi::viz] image width: 1000\n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.67095e+07, 4.67204e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 10.882\n",
      "[odgi::viz] image width: 1000\n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.59e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 500000 bps - iteration 1 (max 3) 100.00% @ 5.75e+00/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 500000 bps - iteration 2 (max 3) 100.00% @ 3.00e+00/s elapsed: 00:00:00:05 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 2.27e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 5.95e+00/s elapsed: 00:00:00:08 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "at9852_1_chr3:22049921-22060460\t10539\n",
      "at6137_1_chr3:21786308-21796197\t9889\n",
      "at6923_1_chr3:24493649-24504190\t10541\n",
      "at6929_1_chr3:21468670-21479217\t10547\n",
      "at7143_1_chr3:21746960-21756850\t9890\n",
      "at8285_1_chr3:22490362-22500235\t9873\n",
      "at9104_1_chr3:23160311-23170861\t10550\n",
      "at9336_1_chr3:22750450-22760995\t10545\n",
      "at9503_1_chr3:23720756-23731301\t10545\n",
      "at9578_1_chr3:22072815-22083362\t10547\n",
      "at9744_1_chr3:22863963-22874510\t10547\n",
      "at9762_1_chr3:22949771-22960320\t10549\n",
      "at9806_1_chr3:23013060-23023607\t10547\n",
      "at9830_1_chr3:22666054-22676603\t10549\n",
      "at9847_1_chr3:23582182-23592730\t10548\n",
      "at9879_1_chr3:22842088-22852633\t10545\n",
      "at9883_1_chr3:21740000-21750545\t10545\n",
      "at9900_1_chr3:23036199-23046733\t10534\n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "(/u/gshirsekar/conda-envs/pggb) \n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 100\n",
      "[odgi::viz] image width: 108\n",
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "a=at9852_1_chr3:0-26358531:22049921-22060460\n",
    "b=nh13\n",
    "r=${a}\n",
    "outprefix=${b}.region\n",
    "\n",
    "echo -e \"at9852_1_chr3\\t21864956\\t21868075\\tNLR\\nat9852_1_chr3\\t21674775\\t21675418\\tNLR2\\nat9852_1_chr3\\t19060775\\t19063847\\tNLR3\\nat9852_1_chr3\\t19097486\\t19100449\\tNLR4\\nat9852_1_chr3\\t11396138\\t11404504\\tENSPM5\" \\\n",
    "> chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_genebed/${outprefix}.bed\n",
    "\n",
    "    ########################## view the pangenome graph of the region ######################\n",
    "    odgi viz \\\n",
    "    -i $inputOdgi \\\n",
    "    -o chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_viz/${outprefix}.bm.png \\\n",
    "    -x 1000 \\\n",
    "    -bm \\\n",
    "    -t 20 \\\n",
    "    -P \\\n",
    "    -r ${r}\n",
    "    ########################## Node Strandedness ############################################\n",
    "    odgi viz \\\n",
    "    -i $inputOdgi \\\n",
    "    -o chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_viz/${outprefix}.z.png \\\n",
    "    -x 1000 \\\n",
    "    -z \\\n",
    "    -t 40 \\\n",
    "    -P \\\n",
    "    -r ${r}\n",
    "\n",
    "odgi extract \\\n",
    "    -i $inputOdgi \\\n",
    "    -r ${r} \\\n",
    "    --threads 24 -P \\\n",
    "    -d 500000 \\\n",
    "    -c 1  \\\n",
    "    -o chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.og\n",
    "\n",
    "odgi paths -L \\\n",
    "    -i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.og|cut -d ':' -f1,2,3| \\\n",
    "sed \"s/:/\\t/g\" |sed \"s/-/\\t/g\"|awk 'OFS=\"\\t\"{print $1,\":\",$4,\"-\",$5,$5-$4}'|sed \"s/\\t:\\t/:/g\"|sed \"s/\\t-\\t/-/g\"\n",
    "\n",
    "\n",
    "#### injecting gene annotation #####\n",
    "\n",
    "\n",
    "\n",
    "odgi view -i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.og  \\\n",
    "    -g > chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_gfa/${outprefix}.gfa\n",
    "\n",
    "cat chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_gfa/${outprefix}.gfa |grep \"^P\"|cut -f2|cut -f1,2 -d ':' > oldNames.txt\n",
    "cat chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_gfa/${outprefix}.gfa |grep \"^P\"|cut -f2|cut -f1 -d \":\" > newNames.txt\n",
    "paste oldNames.txt newNames.txt  > oNewNames.txt\n",
    "\n",
    "cp chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_gfa/${outprefix}.gfa \\\n",
    "    chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_gfa/${outprefix}.ed.gfa\n",
    "\n",
    "while read a b \n",
    "do\n",
    "    sed -i \"s/${a}/${b}/g\" chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_gfa/${outprefix}.ed.gfa \n",
    "done < oNewNames.txt\n",
    "\n",
    "\n",
    "odgi procbed \\\n",
    "-i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_gfa/${outprefix}.ed.gfa \\\n",
    "-b chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_genebed/${outprefix}.bed \\\n",
    "> chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_genebed/${outprefix}.proc.bed\n",
    "\n",
    "odgi inject \\\n",
    "-i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_gfa/${outprefix}.ed.gfa \\\n",
    "-b chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_genebed/${outprefix}.proc.bed \\\n",
    "-o chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.proc.og\n",
    "\n",
    "odgi viz \\\n",
    "-i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.proc.og \\\n",
    "-o chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_viz/${outprefix}.proc.png \\\n",
    "-P -c 12 -w 100 -x 5000 -y 50 -m -B Spectral:4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "9343a672-4843-47d5-a4ba-f50945f9d55f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "at9852_1_chr3:21645148-21677047\t31899\n",
      "at6137_1_chr3:21386454-21412785\t26331\n",
      "at6923_1_chr3:24082379-24118750\t36371\n",
      "at6929_1_chr3:21079725-21098256\t18531\n",
      "at7143_1_chr3:21323346-21370319\t46973\n",
      "at8285_1_chr3:22058842-22108477\t49635\n",
      "at9104_1_chr3:22745133-22788590\t43457\n",
      "at9336_1_chr3:22357392-22373341\t15949\n",
      "at9503_1_chr3:23294121-23337438\t43317\n",
      "at9578_1_chr3:21670064-21687907\t17843\n",
      "at9744_1_chr3:22465659-22484176\t18517\n",
      "at9762_1_chr3:22551851-22571962\t20111\n",
      "at9806_1_chr3:22617522-22632340\t14818\n",
      "at9830_1_chr3:22233470-22288399\t54929\n",
      "at9847_1_chr3:23189377-23207878\t18501\n",
      "at9879_1_chr3:22417079-22464894\t47815\n",
      "at9883_1_chr3:21349314-21367852\t18538\n",
      "at9900_1_chr3:22636036-22659974\t23938\n",
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "odgi paths -L \\\n",
    "    -i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.og|cut -d ':' -f1,2,3| \\\n",
    "sed \"s/:/\\t/g\" |sed \"s/-/\\t/g\"|awk 'OFS=\"\\t\"{print $1,\":\",$4,\"-\",$5,$5-$4}'|sed \"s/\\t:\\t/:/g\"|sed \"s/\\t-\\t/-/g\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "9e897ae4-e2e0-45f5-84a8-86a4997fc681",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "odgi view -i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_og/${outprefix}.og  \\\n",
    "    -g > chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_gfa/${outprefix}.gfa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "dee12cad-2ab1-47c5-9b42-ca2c615e839a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "at9852_1_chr1:0-33329944\t19714894\t19718968\tNLR\n",
      "at9852_1_chr1:0-33329944\t19712162\t19713796\tVANDAL\n",
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cat chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_gfa/${outprefix}.region.gfa |grep \"^P\"|cut -f2|cut -f1,2 -d ':' > oldNames.txt\n",
    "cat chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_gfa/${outprefix}.region.gfa |grep \"^P\"|cut -f2|cut -f1 -d \":\" > newNames.txt\n",
    "paste oldNames.txt newNames.txt  > oNewNames.txt\n",
    "\n",
    "while read a b \n",
    "do\n",
    "    cp chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_gfa/${outprefix}.region.gfa \\\n",
    "    chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_gfa/${outprefix}.region.ed.gfa\n",
    "    \n",
    "    sed -i \"s/${a}/${b}/g\" chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_gfa/${outprefix}.region.ed.gfa \n",
    "done < oNewNames.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "e593c190-3cbd-476d-adc8-f818a8f0957d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "odgi procbed \\\n",
    "-i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_gfa/${outprefix}.gfa \\\n",
    "-b test1.bed \\\n",
    "> test1.adj.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "3d74cf06-77ab-4d2b-ba8b-cc08c6f59bdf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "at9852_1_chr1:19709802-19723621\n",
      "at6137_1_chr1:18522278-18522734\n",
      "at6923_1_chr1:17988272-18002307\n",
      "at6929_1_chr1:18937321-18937777\n",
      "at7143_1_chr1:19934976-19935432\n",
      "at8285_1_chr1:21033986-21034442\n",
      "at9104_1_chr1:18985264-18985720\n",
      "at9336_1_chr1:20317692-20318148\n",
      "at9503_1_chr1:18491569-18492025\n",
      "at9578_1_chr1:19768406-19768862\n",
      "at9744_1_chr1:18993676-19010178\n",
      "at9762_1_chr1:21064932-21065388\n",
      "at9806_1_chr1:21013890-21014346\n",
      "at9830_1_chr1:19314896-19315352\n",
      "at9847_1_chr1:18031599-18032055\n",
      "at9879_1_chr1:20217099-20217556\n",
      "at9883_1_chr1:20132519-20132975\n",
      "at9900_1_chr1:19663633-19679969\n",
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "odgi paths -i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_gfa/${outprefix}.gfa \\\n",
    "-L "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "e71e94f1-bfb5-47f9-b1d0-93a31c4b2aef",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "at9852_1_chr1:19709802-19723621\t2360\t3994\tVANDAL\n",
      "at9852_1_chr1:19709802-19723621\t5092\t9166\tNLR\n",
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cat test1.adj.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "57d25923-7503-4112-b288-a4c298e558f6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "odgi inject -i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_gfa/${outprefix}.gfa \\\n",
    "-b test1.adj.bed -o - |odgi paths -i - -L|tail -2 > test1.gene.names.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "2a3dceb2-9a90-4d83-b1dd-ec120990f7df",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VANDAL\n",
      "NLR\n",
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cat test1.gene.names.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "88fd0ad2-8e68-4716-a03c-badbb670294a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "odgi inject \\\n",
    "-i chr3_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/cluster_gfa/${outprefix}.gfa \\\n",
    "-b test1.adj.bed \\\n",
    "-o test1.genes.og"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "11bbec22-41a6-4f21-ba03-bdf0de19cfd6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 50\n",
      "[odgi::viz] image width: 349\n",
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "odgi viz -i test1.genes.og -o test1.genes.png -P -c 12 -w 50 -x 2000 -y 50 -m -B Spectral:4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6ccada7d-db91-4aa2-8a7c-32cdbc3fd643",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
