{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0d1d63ac-1ff1-47cb-b6bf-3fb78d7179e0",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-23T06:22:17.487059Z",
     "iopub.status.busy": "2023-08-23T06:22:17.486025Z",
     "iopub.status.idle": "2023-08-23T06:22:17.901504Z",
     "shell.execute_reply": "2023-08-23T06:22:17.900260Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "conda activate pggb_fatica\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4cd277c5-7d42-4852-a3ad-b961337284b4",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-23T06:22:17.907766Z",
     "iopub.status.busy": "2023-08-23T06:22:17.907112Z",
     "iopub.status.idle": "2023-08-23T06:22:21.431777Z",
     "shell.execute_reply": "2023-08-23T06:22:21.430506Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "chromosomeName=$(odgi paths -L -i chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og|grep \"9852\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "616ee04e-eeac-42de-82e5-e9036b72d6c9",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-23T06:22:21.438025Z",
     "iopub.status.busy": "2023-08-23T06:22:21.437331Z",
     "iopub.status.idle": "2023-08-23T06:22:21.566732Z",
     "shell.execute_reply": "2023-08-23T06:22:21.565470Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cat neighborhoods/final/final_all_neighborhoods.chr1.bed|grep \"at9852\"|sed \"s/at9852_1_chr1/${chromosomeName}/g\" > neighborhoods/all_neighborhoods_at9852.chr1.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "dff2a97c-3be1-4439-afc6-ca8ea19c5077",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-23T06:22:21.572768Z",
     "iopub.status.busy": "2023-08-23T06:22:21.572122Z",
     "iopub.status.idle": "2023-08-23T06:22:21.693674Z",
     "shell.execute_reply": "2023-08-23T06:22:21.692446Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "at9852_1_chr1:0-33329944\t3148661\t3153437\tchr1_nh01\n",
      "at9852_1_chr1:0-33329944\t3658437\t3664598\tchr1_nh02\n",
      "at9852_1_chr1:0-33329944\t4146752\t4195752\tchr1_nh03\n",
      "at9852_1_chr1:0-33329944\t5486321\t5520209\tchr1_nh04\n",
      "at9852_1_chr1:0-33329944\t6067975\t6081285\tchr1_nh05\n",
      "at9852_1_chr1:0-33329944\t9335621\t9353580\tchr1_nh06\n",
      "at9852_1_chr1:0-33329944\t11132927\t11183727\tchr1_nh07\n",
      "at9852_1_chr1:0-33329944\t11671769\t11685367\tchr1_e_r5_nh01\n",
      "at9852_1_chr1:0-33329944\t11726461\t11735037\tchr1_e_r3_nh01\n",
      "at9852_1_chr1:0-33329944\t12039366\t12066573\tchr1_nh08\n",
      "at9852_1_chr1:0-33329944\t12215382\t12286922\tchr1_e_r4_nh01\n",
      "at9852_1_chr1:0-33329944\t12290158\t12377654\tchr1_nh09\n",
      "at9852_1_chr1:0-33329944\t19185564\t19235773\tchr1_e_nh01\n",
      "at9852_1_chr1:0-33329944\t19451716\t19518693\tchr1_e_r2_nh01\n",
      "at9852_1_chr1:0-33329944\t19621896\t19725905\tchr1_nh10\n",
      "at9852_1_chr1:0-33329944\t20328985\t20357580\tchr1_nh11\n",
      "at9852_1_chr1:0-33329944\t21569880\t21578981\tchr1_nh12\n",
      "at9852_1_chr1:0-33329944\t21980955\t22010123\tchr1_e_nh02\n",
      "at9852_1_chr1:0-33329944\t22066885\t22076210\tchr1_nh13\n",
      "at9852_1_chr1:0-33329944\t22569629\t22575930\tchr1_nh14\n",
      "at9852_1_chr1:0-33329944\t22656847\t22662183\tchr1_nh15\n",
      "at9852_1_chr1:0-33329944\t22850725\t22860978\tchr1_nh16\n",
      "at9852_1_chr1:0-33329944\t24032348\t24034285\tchr1_e_r6_nh01\n",
      "at9852_1_chr1:0-33329944\t24105498\t24118100\tchr1_e_r2_nh02\n",
      "at9852_1_chr1:0-33329944\t24119759\t24165629\tchr1_nh17\n",
      "at9852_1_chr1:0-33329944\t24304282\t24389167\tchr1_nh18\n",
      "at9852_1_chr1:0-33329944\t24611456\t24657983\tchr1_nh19\n",
      "at9852_1_chr1:0-33329944\t24665855\t24887162\tchr1_nh20\n",
      "at9852_1_chr1:0-33329944\t24948195\t24959469\tchr1_e_r2_nh03\n",
      "at9852_1_chr1:0-33329944\t25098667\t25133188\tchr1_e_nh03-chr1_nh21\n",
      "at9852_1_chr1:0-33329944\t25388900\t25404594\tchr1_nh22\n",
      "at9852_1_chr1:0-33329944\t25416640\t25503173\tchr1_nh23\n",
      "at9852_1_chr1:0-33329944\t25786195\t25798492\tchr1_e_nh04\n",
      "at9852_1_chr1:0-33329944\t25991706\t26011075\tchr1_e_r4_nh02\n",
      "at9852_1_chr1:0-33329944\t26049762\t26366009\tchr1_nh24\n",
      "at9852_1_chr1:0-33329944\t26496530\t26603880\tchr1_nh25\n",
      "at9852_1_chr1:0-33329944\t26652207\t26673792\tchr1_nh26\n",
      "at9852_1_chr1:0-33329944\t26741177\t26753413\tchr1_nh27\n",
      "at9852_1_chr1:0-33329944\t27176537\t27194539\tchr1_nh28\n",
      "at9852_1_chr1:0-33329944\t27384841\t27392448\tchr1_nh29\n",
      "at9852_1_chr1:0-33329944\t27487655\t27497097\tchr1_nh30\n",
      "at9852_1_chr1:0-33329944\t29014727\t29041233\tchr1_nh31\n",
      "at9852_1_chr1:0-33329944\t30286495\t30325736\tchr1_nh32\n",
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cat neighborhoods/all_neighborhoods_at9852.chr1.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6b5a53b1-d908-45d1-bb1d-5476f533c907",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-23T06:22:21.697320Z",
     "iopub.status.busy": "2023-08-23T06:22:21.696681Z",
     "iopub.status.idle": "2023-08-23T06:22:22.257787Z",
     "shell.execute_reply": "2023-08-23T06:22:22.256510Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(pggb_fatica) \n",
      "(pggb_fatica) \n",
      "(pggb_fatica) \n",
      "(pggb_fatica) \n",
      "(pggb_fatica) \n",
      "(pggb_fatica) \n",
      "(pggb_fatica) \n",
      "(pggb_fatica) \n",
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "mkdir -p neighborhoods/chr1/cluster_og\n",
    "mkdir -p neighborhoods/chr1/cluster_gfa\n",
    "mkdir -p neighborhoods/chr1/cluster_allPathCoordinates\n",
    "mkdir -p neighborhoods/chr1/cluster_viz\n",
    "mkdir -p neighborhoods/chr1/cluster_pav\n",
    "mkdir -p neighborhoods/chr1/cluster_similarity\n",
    "mkdir -p neighborhoods/chr1/cluster_jumps\n",
    "mkdir -p neighborhoods/chr1/cluster_heaps\n",
    "mkdir -p neighborhoods/chr1/cluster_panacus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "32304c64-ede9-4274-b4bb-cfefff7f345e",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-23T06:22:22.264334Z",
     "iopub.status.busy": "2023-08-23T06:22:22.263665Z",
     "iopub.status.idle": "2023-08-23T08:39:37.496175Z",
     "shell.execute_reply": "2023-08-23T08:39:37.494418Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(pggb_fatica) \n",
      "(pggb_fatica) \n",
      "(pggb_fatica) \n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.60054e+06, 3.60686e+06]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 6.321\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.60054e+06, 3.60686e+06]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 6.321\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 7.13e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 9.66e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.22e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.62e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.85e+01/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (1.45679e-05 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 1.83e+06/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.20332e+06, 4.20957e+06]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 6.244\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.20332e+06, 4.20957e+06]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 6.244\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 7.38e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 9.56e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.10e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 8.04e+02/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.76e+01/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 2.11e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (5.32793e-05 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 9.67e+05/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.82777e+06, 4.90664e+06]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 78.872\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.82777e+06, 4.90664e+06]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 78.872\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 6.65e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.23e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.10e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.57e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.65e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.16e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (1.50698e-07 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 1.77e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [6.43704e+06, 6.47465e+06]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 37.617\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [6.43704e+06, 6.47465e+06]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 37.617\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 6.48e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.02e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.08e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 8.34e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.64e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.16e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (4.42672e-07 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 9.66e+06/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [7.12533e+06, 7.14525e+06]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 19.922\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [7.12533e+06, 7.14525e+06]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 19.922\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 6.66e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.04e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.08e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 2.09e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.81e+01/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (7.8025e-06 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 2.46e+06/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.12927e+07, 1.13226e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 29.952\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.12927e+07, 1.13226e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 29.952\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 5.73e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.04e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.19e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 4.43e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.79e+01/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (2.05253e-06 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 4.43e+06/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.38509e+07, 1.39719e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 121.022\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.38509e+07, 1.39719e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 121.022\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 4.87e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.01e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.05e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 3.94e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.76e+01/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.17e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (2.16136e-08 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 3.78e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.49909e+07, 1.50261e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 35.204\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.49909e+07, 1.50261e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 35.204\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 5.37e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 9.83e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.30e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 9.58e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.61e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.17e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (4.82253e-07 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 8.96e+06/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.50935e+07, 1.51123e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 18.747\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.50935e+07, 1.51123e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 18.747\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 5.49e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.03e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.03e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 5.09e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.69e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (2.20785e-06 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 4.42e+06/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.56826e+07, 1.57196e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 36.918\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.56826e+07, 1.57196e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 36.918\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 5.05e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 9.34e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.11e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.36e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.68e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (2.20415e-07 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 1.42e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.60603e+07, 1.66126e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 552.316\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.60603e+07, 1.66126e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 552.316\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 4.97e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.05e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.03e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 3.56e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.68e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 9.55e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (2.28459e-08 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 3.75e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.65418e+07, 1.67321e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 190.367\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.65418e+07, 1.67321e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 190.367\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 5.17e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.12e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.20e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 5.92e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.79e+01/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.17e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (1.61168e-08 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 4.76e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.19369e+07, 3.22019e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 264.979\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.19369e+07, 3.22019e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 264.979\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.84e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.06e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.04e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 4.61e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.67e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 8.77e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (8.60842e-09 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 5.03e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.28687e+07, 3.30024e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 133.642\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.28687e+07, 3.30024e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 133.642\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.55e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.07e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.07e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 4.56e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.61e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.16e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (1.43873e-08 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 5.16e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.31458e+07, 3.33565e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 210.698\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.31458e+07, 3.33565e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 210.698\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.46e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.07e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.12e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 6.41e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.65e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.16e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (1.10431e-08 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 5.57e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.45284e+07, 3.46194e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 90.997\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.45284e+07, 3.46194e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 90.997\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 2.12e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.10e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 9.78e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 2.09e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.58e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (1.22598e-07 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 1.82e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.64246e+07, 3.64421e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 17.563\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.64246e+07, 3.64421e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 17.563\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.28e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.04e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.13e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 4.63e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.71e+01/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (6.75179e-07 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 5.70e+06/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.70038e+07, 3.71877e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 183.867\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.70038e+07, 3.71877e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 183.867\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.62e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.05e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.13e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 2.20e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.67e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (1.32039e-07 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 1.77e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.71665e+07, 3.72079e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 41.445\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.71665e+07, 3.72079e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 41.445\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.40e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.11e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 9.62e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 3 (max 3) 100.00% @ 1.14e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 7.07e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.64e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (2.45074e-07 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 9.31e+06/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.80948e+07, 3.81099e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 15.089\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.80948e+07, 3.81099e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 15.089\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.26e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.14e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.09e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.48e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.69e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 3.50e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (1.47929e-05 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 1.64e+06/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.82386e+07, 3.82442e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 5.577\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.82386e+07, 3.82442e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 5.577\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.42e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.00e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.07e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 2.46e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.65e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.19e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (5.33365e-06 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 3.04e+06/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.85178e+07, 3.85402e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 22.382\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.85178e+07, 3.85402e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 22.382\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.70e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 9.99e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.12e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 4.49e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.53e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (1.76833e-06 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 5.26e+06/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.01773e+07, 4.01914e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 14.056\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.01773e+07, 4.01914e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 14.056\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.28e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.27e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.28e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 2.89e+02/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.73e+01/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 3.99e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (0.000416493 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 3.28e+05/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.02952e+07, 4.03153e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 20.064\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.02952e+07, 4.03153e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 20.064\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.65e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.04e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.08e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 8.21e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.72e+01/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (5.14606e-07 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 8.92e+06/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.03171e+07, 4.04588e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 141.699\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.03171e+07, 4.04588e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 141.699\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.61e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 9.59e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.11e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 2.82e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.64e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (5.28229e-08 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 2.29e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.06198e+07, 4.07988e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 179.036\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.06198e+07, 4.07988e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 179.036\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.32e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.10e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.12e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 5.50e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.53e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.16e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (1.13366e-08 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 2.94e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.11744e+07, 4.12804e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 105.984\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.11744e+07, 4.12804e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 105.984\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.35e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.11e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.01e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 4.28e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.61e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.17e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (2.56986e-08 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 3.98e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.12949e+07, 4.1821e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 526.058\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.12949e+07, 4.1821e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 526.058\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.28e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 9.57e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 9.11e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.71e+05/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.69e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.12e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (1.48157e-09 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 5.48e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.19251e+07, 4.19458e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 20.737\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.19251e+07, 4.19458e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 20.737\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.23e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.08e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.01e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 6.79e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.63e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.16e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (8.34011e-07 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 7.47e+06/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.21878e+07, 4.22291e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 41.284\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.21878e+07, 4.22291e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 41.284\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.83e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.05e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.07e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.28e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.54e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.15e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (1.68103e-07 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 1.56e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.26426e+07, 4.2695e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 52.391\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.26426e+07, 4.2695e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 52.391\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.63e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 9.52e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.09e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.21e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.58e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (2.87503e-07 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 1.25e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.27122e+07, 4.29761e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 263.914\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.27122e+07, 4.29761e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 263.914\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.58e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 9.22e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 9.80e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 9.58e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.55e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.14e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (3.75871e-09 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 5.18e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.34983e+07, 4.35303e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 31.981\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.34983e+07, 4.35303e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 31.981\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.57e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.04e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.06e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 6.83e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.62e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (9.48111e-07 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 6.98e+06/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.38254e+07, 4.39046e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 79.181\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.38254e+07, 4.39046e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 79.181\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.62e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.19e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.15e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.53e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.66e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.17e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (1.62198e-07 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 1.74e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.39383e+07, 4.45797e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 641.329\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.39383e+07, 4.45797e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 641.329\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.53e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.10e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.04e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 3 (max 3) 100.00% @ 1.07e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.96e+05/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.61e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.04e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (7.90996e-10 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 5.64e+07/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.47437e+07, 4.4936e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 192.311\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.47437e+07, 4.4936e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 192.311\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.53e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.14e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 9.60e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 5.12e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.48e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.15e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (1.62776e-08 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 5.22e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.49985e+07, 4.50329e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 34.408\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.49985e+07, 4.50329e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 34.408\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.33e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 9.93e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.03e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 8.72e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.49e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.17e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (5.19064e-07 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 9.66e+06/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.51358e+07, 4.51615e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 25.695\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.51358e+07, 4.51615e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 25.695\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.30e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.00e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.11e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 5.21e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.52e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (1.26531e-06 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 5.90e+06/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.58044e+07, 4.58295e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 25.145\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.58044e+07, 4.58295e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 25.145\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.55e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 9.94e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.02e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.04e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.59e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (3.54731e-07 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 1.16e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/mpcdf/soft/SLE_15/packages/x86_64/anaconda/3/2021.05/lib/python3.8/site-packages/scipy/optimize/minpack.py:828: OptimizeWarning: Covariance of the parameters could not be estimated\n",
      "  warnings.warn('Covariance of the parameters could not be estimated',\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.60804e+07, 4.60886e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 8.183\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.60804e+07, 4.60886e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 8.183\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.34e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.21e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.03e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 2.35e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.65e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (6.09663e-06 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 2.82e+06/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.62495e+07, 4.62674e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 17.891\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.62495e+07, 4.62674e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 17.891\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.32e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 9.85e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.01e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 6.47e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.67e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (1.42401e-06 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 5.03e+06/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.8277e+07, 4.83329e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 55.898\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.8277e+07, 4.83329e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 55.898\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.59e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.02e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.08e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 2.66e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.67e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.17e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (6.58475e-08 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 2.64e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.98148e+07, 4.99165e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 101.748\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.98148e+07, 4.99165e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 101.748\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.73e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.06e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.04e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 2.57e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.52e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.16e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "path_sgd_zipf_space_max: 100\n",
      "path_sgd_zipf_max_number_of_distributions: 101\n",
      "[odgi::path_linear_sgd] calculating linear SGD schedule (5.77015e-08 1 100 0 0.01)\n",
      "[odgi::path_linear_sgd] calculating zetas for 102 zipf distributions\n",
      "[odgi::path_linear_sgd] 1D path-guided SGD: 100.00% @ 2.42e+07/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:107: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[0].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "/u/gshirsekar/conda-envs/pggb/bin/panacus-visualize:128: UserWarning: FixedFormatter should only be used together with FixedLocator\n",
      "  axs[1].set_yticklabels(calibrate_yticks_text(yticks))\n",
      "(/u/gshirsekar/conda-envs/pggb) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "inputOdgi=chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og\n",
    "outdir=neighborhoods/chr1\n",
    "\n",
    "while read a b c d; \n",
    "do   \n",
    "    r_path=$(echo \"${a}:${b}-${c}\")\n",
    "    r=${r_path}\n",
    "    nh=${d}\n",
    "    outprefix=${nh}.region\n",
    "\n",
    "    ########################## view the pangenome graph of the region ######################\n",
    "    odgi viz \\\n",
    "    -i $inputOdgi \\\n",
    "    -o $outdir/${outprefix}.bm.png \\\n",
    "    -x 1000 \\\n",
    "    -bm \\\n",
    "    -t 20 \\\n",
    "    -P \\\n",
    "    -r ${r}\n",
    "    ########################## Node Strandedness ############################################\n",
    "    odgi viz \\\n",
    "    -i $inputOdgi \\\n",
    "    -o $outdir/${outprefix}.z.png \\\n",
    "    -x 1000 \\\n",
    "    -z \\\n",
    "    -t 20 \\\n",
    "    -P \\\n",
    "    -r ${r}\n",
    "    \n",
    "    ############################################ extract the region ############################################\n",
    "    odgi extract \\\n",
    "    -i $inputOdgi \\\n",
    "    -r ${r} \\\n",
    "    --threads 24 -P \\\n",
    "    -d 100000 \\\n",
    "    -c 1  \\\n",
    "    -o $outdir/cluster_og/${outprefix}.og \n",
    "    \n",
    "    ########################### get the coordinates in all the accessions ########################################\n",
    "    odgi paths -L \\\n",
    "    -i $outdir/cluster_og/${outprefix}.og  \\\n",
    "    > $outdir/cluster_allPathCoordinates/${outprefix}.allPathsCoordinates.txt\n",
    "    \n",
    "    ########################### for bandage plot #################################################################\n",
    "    odgi view -i $outdir/cluster_og/${outprefix}.og  \\\n",
    "    -g > $outdir/cluster_gfa/${outprefix}.gfa\n",
    "    \n",
    "    ########################### Individual path-based PAV comparisons ############################################\n",
    "    while read line;\n",
    "    do\n",
    "        r=${line}\n",
    "        echo $r|cut -d ':' -f1,2 >tmp1\n",
    "        echo $r|cut -d ':' -f3|sed \"s/-/\\t/g\" >tmp2\n",
    "        paste tmp1 tmp2 > \\\n",
    "        $outdir/cluster_allPathCoordinates/${outprefix}.${line}.bed && rm tmp1 && rm tmp2\n",
    "        \n",
    "        odgi pav \\\n",
    "        -i $inputOdgi \\\n",
    "        -t 24 \\\n",
    "        -b $outdir/cluster_allPathCoordinates/${outprefix}.${line}.bed > \\\n",
    "        $outdir/cluster_pav/${outprefix}.${line}.pav.txt\n",
    "        \n",
    "    done < $outdir/cluster_allPathCoordinates/${outprefix}.allPathsCoordinates.txt\n",
    "    \n",
    "    ########################### Node Jumps in Neighborhood ######################################################\n",
    "    ## for each node how many jumps it has taken\n",
    "    cat $outdir/cluster_gfa/${outprefix}.gfa \\\n",
    "    |grep \"^L\" \\\n",
    "    |awk '{print $1,$2,$4,$4-$2}'\\\n",
    "    |sed \"s/ /\\t/g\" \\\n",
    "    > $outdir/cluster_jumps/${outprefix}.nodeJumps.txt\n",
    "    \n",
    "    \n",
    "    ########################### Node Degrees ############################################################\n",
    "    odgi \\\n",
    "    degree \\\n",
    "    -i $outdir/cluster_og/${outprefix}.og \\\n",
    "    -t 24 \\\n",
    "    -d \\\n",
    "    -r ${r} > $outdir/cluster_jumps/${outprefix}.nodeDegrees.txt\n",
    "    \n",
    "    \n",
    "    ########################### Node Depth ###############################################################\n",
    "    odgi \\\n",
    "    depth \\\n",
    "    -i $outdir/cluster_og/${outprefix}.og \\\n",
    "    -t 24 \\\n",
    "    -d \\\n",
    "    -r ${r} > $outdir/cluster_jumps/${outprefix}.nodeDepth.txt\n",
    "    \n",
    "    ########################### Distance between different Paths (accessions) for the region #############\n",
    "    odgi \\\n",
    "    similarity \\\n",
    "    -i $outdir/cluster_og/${outprefix}.og \\\n",
    "    -t 24 \\\n",
    "    -d > $outdir/cluster_similarity/${outprefix}.distance.txt\n",
    "    \n",
    "    ###### Pangenome Graph Growth  Heaps' Law #########\n",
    "\n",
    "    odgi sort \\\n",
    "    -i $outdir/cluster_og/${outprefix}.og \\\n",
    "    -p Y \\\n",
    "    -O \\\n",
    "    -t 24 \\\n",
    "    -P \\\n",
    "    -o $outdir/cluster_og/${outprefix}.optimized_sorted.og \n",
    "    \n",
    "    for i in {1..18}\n",
    "    do\n",
    "    seq -w 1 ${i} > $outdir/cluster_heaps/pathGroup${i}.groups.txt\n",
    "    done\n",
    "    \n",
    "    \n",
    "    for i in {1..18}\n",
    "    do\n",
    "    odgi paths -i $outdir/cluster_og/${outprefix}.optimized_sorted.og \\\n",
    "    -L|sort -R|head -n ${i} > $outdir/cluster_heaps/pathGroup${i}.txt \n",
    "    done\n",
    "    \n",
    "    for i in {1..18}\n",
    "    do\n",
    "    paste $outdir/cluster_heaps/pathGroup${i}.txt $outdir/cluster_heaps/pathGroup${i}.groups.txt > tmp\n",
    "    mv tmp $outdir/cluster_heaps/pathGroup${i}.txt\n",
    "    done\n",
    "    \n",
    "    for i in {1..18}\n",
    "    do\n",
    "        odgi \\\n",
    "        heaps \\\n",
    "        -i $outdir/cluster_og/${outprefix}.optimized_sorted.og \\\n",
    "        -t 24 \\\n",
    "        -S \\\n",
    "        -d 2 \\\n",
    "        -n 100 \\\n",
    "        -p $outdir/cluster_heaps/pathGroup${i}.txt \\\n",
    "        > $outdir/cluster_heaps/${outprefix}.pathGroup${i}.optimized_sorted.heaps.txt\n",
    "    done\n",
    "    \n",
    "    \n",
    "    ########################## Panacus ##############################\n",
    "    parentDir=$outdir\n",
    "    \n",
    "    conda activate /u/gshirsekar/conda-envs/pggb #panacus is installed here\n",
    "\n",
    "    grep '^P' ${parentDir}/cluster_gfa/${outprefix}.gfa|cut -f2|sed \"s/_.*//g\" > ${parentDir}/cluster_panacus/${outprefix}.groupNames.txt\n",
    "    grep '^P' ${parentDir}/cluster_gfa/${outprefix}.gfa|cut -f2> ${parentDir}/cluster_panacus/${outprefix}.pathNames.txt\n",
    "    \n",
    "    paste ${parentDir}/cluster_panacus/${outprefix}.pathNames.txt \\\n",
    "    ${parentDir}/cluster_panacus/${outprefix}.groupNames.txt \\\n",
    "    > ${parentDir}/cluster_panacus/${outprefix}.groups.txt\n",
    "    \n",
    "    sed -i \"s/_1_chr1/#1#chr1/g\" ${parentDir}/cluster_panacus/${outprefix}.pathNames.txt \n",
    "    sed -i \"s/_1_chr1/#1#chr1/g\" ${parentDir}/cluster_panacus/${outprefix}.groups.txt\n",
    "    \n",
    "    \n",
    "    cat ${parentDir}/cluster_panacus/${outprefix}.pathNames.txt |cut -f1,3 -d \":\" > \\\n",
    "    ${parentDir}/cluster_panacus/${outprefix}.newPathNames.txt\n",
    "    \n",
    "    paste ${parentDir}/cluster_panacus/${outprefix}.pathNames.txt \\\n",
    "    ${parentDir}/cluster_panacus/${outprefix}.newPathNames.txt > ${parentDir}/cluster_panacus/tmp\n",
    "    \n",
    "    while read a b ; do sed -i \"s/${a}/${b}/\" ${parentDir}/cluster_panacus/${outprefix}.pathNames.txt ; done < ${parentDir}/cluster_panacus/tmp\n",
    "    while read a b ; do sed -i \"s/${a}/${b}/\" ${parentDir}/cluster_panacus/${outprefix}.groups.txt ; done < ${parentDir}/cluster_panacus/tmp\n",
    "    cp ${parentDir}/cluster_gfa/${outprefix}.gfa ${parentDir}/cluster_panacus/${outprefix}.ed.gfa\n",
    "    sed -i \"s/_1_chr1/#1#chr1/g\" ${parentDir}/cluster_panacus/${outprefix}.ed.gfa\n",
    "    \n",
    "    while read a b ; do sed -i \"s/${a}/${b}/g\" ${parentDir}/cluster_panacus/${outprefix}.ed.gfa ; done < ${parentDir}/cluster_panacus/tmp \n",
    "    \n",
    "    panacus histgrowth -t24 -q 0,0.1,0.5,1 -l 1,2,3,17 -g ${parentDir}/cluster_panacus/${outprefix}.groups.txt \\\n",
    "    -s ${parentDir}/cluster_panacus/${outprefix}.pathNames.txt \\\n",
    "    ${parentDir}/cluster_panacus/${outprefix}.ed.gfa > ${parentDir}/cluster_panacus/${outprefix}.histgrowth.txt\n",
    "    \n",
    "    panacus-visualize -e ${parentDir}/cluster_panacus/${outprefix}.histgrowth.txt > ${parentDir}/cluster_panacus/${outprefix}.histgrowth.node.pdf\n",
    "\n",
    "done < neighborhoods/all_neighborhoods_at9852.chr1.txt\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ac1bd101-dd00-4858-892f-7adf2ebd5faa",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
