{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0d1d63ac-1ff1-47cb-b6bf-3fb78d7179e0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "conda activate pggb_fatica"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4cd277c5-7d42-4852-a3ad-b961337284b4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "at6137_1_chr1:0-32271270\n",
      "at6923_1_chr1:0-31749057\n",
      "at6929_1_chr1:0-32573738\n",
      "at7143_1_chr1:0-33668152\n",
      "at8285_1_chr1:0-34650672\n",
      "at9104_1_chr1:0-32656548\n",
      "at9336_1_chr1:0-34071237\n",
      "at9503_1_chr1:0-32192683\n",
      "at9578_1_chr1:0-33401629\n",
      "at9744_1_chr1:0-33001351\n",
      "at9762_1_chr1:0-34863657\n",
      "at9806_1_chr1:0-34678793\n",
      "at9830_1_chr1:0-33113414\n",
      "at9847_1_chr1:0-31771136\n",
      "at9852_1_chr1:0-33329944\n",
      "at9879_1_chr1:0-34155544\n",
      "at9883_1_chr1:0-33747718\n",
      "at9900_1_chr1:0-33376575\n",
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "odgi paths -L \\\n",
    "-i chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c145199f-e2f1-4942-ac63-d0f3e5862216",
   "metadata": {},
   "source": [
    "## get length of each node\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f64ffd2d-30d5-44cb-8d7a-fb20e6b22bf7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(pggb_fatica) \n",
      "(pggb_fatica) \n",
      "(pggb_fatica) \n",
      "(pggb_fatica) \n",
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "odgi view \\\n",
    "-i chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og \\\n",
    "-g \\\n",
    "-t 24 \\\n",
    "| grep '^S' \\\n",
    "| awk -v OFS='\\t' '{print($2,length($3))}' \\\n",
    "> chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og.nodeLength.txt\n",
    "\n",
    "echo -e \"#node.id\\tlength\" > tmpHeader\n",
    "cat tmpHeader chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og.nodeLength.txt > tmp \n",
    "mv tmp chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og.nodeLength.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "97267ce7-0f02-467d-ae6f-5ec437dbb560",
   "metadata": {},
   "source": [
    "## get depth of each node"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ab4a9538-73c4-4e88-882c-18bb111348bd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "odgi \\\n",
    "    depth \\\n",
    "    -i chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og \\\n",
    "    -t 24 \\\n",
    "    -d \\\n",
    "    > chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og.nodeDepth.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e7e12f2b-8e3d-4e69-b88d-26a1741bb9f8",
   "metadata": {},
   "source": [
    "## merge depth and length"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d0682dc3-8350-4bdb-b306-514932ee2bac",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input length :chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og.nodeLength.txt\n",
      "input depth : chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og.nodeDepth.txt\n",
      "output file : chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og.nodeDepthLength.txt\n",
      "\n",
      " time for processing input: 4.293874263763428\n",
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "python mergeNodeLengthDepth.py \\\n",
    "-l chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og.nodeLength.txt \\\n",
    "-d chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og.nodeDepth.txt \\\n",
    "-o chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og.nodeDepthLength.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9fc6d345-4c39-4408-93e6-67a48b3bfd8b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "fa2a1ef7-570f-45c9-97d9-005a0334cd65",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "echo -e \"at9852_1_chr1:0-33329944\\t12055116\\t12058236\" > adr1.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "c63db910-e256-4a8e-8ee9-3d3e7951563f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "896555\n",
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "0e5cb2c8-d254-41bc-b96c-be5da5a78998",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "## here input should be a bed file of leftmost NLRs start and rightmost NLRs stop for the cluster (group of NLRs)\n",
    "\n",
    "while read chromosome start stop\n",
    "do\n",
    "\n",
    "    odgi \\\n",
    "    position \\\n",
    "    -i chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og \\\n",
    "    -t 24 \\\n",
    "    -b <(echo -e \"${chromosome}\\t${start}\\t${stop}\") \\\n",
    "    -v \n",
    "\n",
    "done < adr1.bed |cut -f1,4,5|cut -d ',' -f1,3|sed 's/+//g'|sed 's/,//g' > adr1.graph.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "afe4596b-1d46-4973-8d58-48e726d7de62",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(pggb_fatica) \n",
      "(pggb_fatica) \n",
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "## here input should be a bed file of leftmost NLRs start and rightmost NLRs stop for the cluster (group of NLRs)\n",
    "\n",
    "while read chromosome start stop compoundName\n",
    "do\n",
    "\n",
    "    odgi \\\n",
    "    position \\\n",
    "    -i chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og \\\n",
    "    -t 24 \\\n",
    "    -b <(echo -e \"${chromosome}\\t${start}\\t${stop}\") \\\n",
    "    -v \n",
    "\n",
    "done < neighborhoods/chr1_compound.bed |cut -f1,4,5|cut -d ',' -f1,3|sed 's/+//g'|sed 's/,//g' > neighborhoods/chr1_compound.graph.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "9939af1b-9563-45a5-aa0e-edacb0458a2a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cat: tmp1: No such file or directory\n",
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "4efedd5b-5f53-4c96-ab9c-d9beca1c310e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(pggb_fatica) \n",
      "(pggb_fatica) \n",
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "while read chromosome gstart gstop\n",
    "do\n",
    "    echo ${chromosome} >> tmp3\n",
    "    \n",
    "    # take fifth leftmost node with depth.uniq=18 and length greater than 100 as left anchor for the neighborhood\n",
    "    awk '($2>100)&&($4==18)' chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og.nodeDepthLength.txt|awk -v \"start=${gstart}\" '($1<start)'|tail -n 5|head -n1|cut -f1 >> tmp1\n",
    "    \n",
    "    # take fifth rightmost node with depth.uniq=18 and length greater than 100 as right anchor for the neighborhood\n",
    "    awk '($2>100)&&($4==18)' chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og.nodeDepthLength.txt|awk -v \"stop=${gstop}\" '($1>stop)'|head -n 5|tail -n1|cut -f1 >> tmp2\n",
    "\n",
    "done < adr1.graph.bed \n",
    "\n",
    "paste tmp3 tmp1 tmp2 > adr1.neighborhood.graph.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c95ffac3-88ae-408d-84d2-a9297afcd692",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(pggb_fatica) \n",
      "(pggb_fatica) \n",
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "while read chromosome gstart gstop\n",
    "do\n",
    "    echo ${chromosome} >> tmp3\n",
    "    \n",
    "    # take fifth leftmost node with depth.uniq=18 and length greater than 100 as left anchor for the neighborhood\n",
    "    awk '($2>100)&&($4==18)' chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og.nodeDepthLength.txt|awk -v \"start=${gstart}\" '($1<start)'|tail -n 5|head -n1|cut -f1 >> tmp1\n",
    "    \n",
    "    # take fifth rightmost node with depth.uniq=18 and length greater than 100 as right anchor for the neighborhood\n",
    "    awk '($2>100)&&($4==18)' chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og.nodeDepthLength.txt|awk -v \"stop=${gstop}\" '($1>stop)'|head -n 5|tail -n1|cut -f1 >> tmp2\n",
    "\n",
    "done < neighborhoods/chr1_compound.graph.bed\n",
    "\n",
    "paste tmp3 tmp1 tmp2 > neighborhoods/chr1_neighborhood.graph.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "aa49a5d1-d55f-4b6b-9774-d0e1aae875e0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "at9852_1_chr1:0-33329944\t894427\t897859\n",
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cat adr1.neighborhood.graph.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "6d1f17b0-eb7d-45e3-8af8-ed936f4f69f3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(pggb_fatica) \n",
      "(pggb_fatica) \n",
      "(pggb_fatica) \n",
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "rm tmp*\n",
    "while read chromosome nStart nStop\n",
    "do\n",
    "    echo ${chromosome} >> tmp3\n",
    "\n",
    "    odgi position \\\n",
    "    -i chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og \\\n",
    "    -t 24 \\\n",
    "    -g ${nStart} \\\n",
    "    -r ${chromosome} |tail -n +2|cut -d ',' -f4,5|sed \"s/,/\\t/g\"|cut -f1 >> tmp1\n",
    "\n",
    "    odgi position \\\n",
    "    -i chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og \\\n",
    "    -t 24 \\\n",
    "    -g ${nStop} \\\n",
    "    -r ${chromosome} |tail -n +2|cut -d ',' -f4,5|sed \"s/,/\\t/g\"|cut -f1 >> tmp2\n",
    "done < adr1.neighborhood.graph.bed\n",
    "\n",
    "paste tmp3 tmp1 tmp2 > adr1.neighborhood.coordinates.bed\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "a63efe28-4a47-4c97-a262-5c72c8c8b9e0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(pggb_fatica) \n",
      "(pggb_fatica) \n",
      "(pggb_fatica) \n",
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "rm tmp*\n",
    "while read chromosome nStart nStop\n",
    "do\n",
    "    echo ${chromosome} >> tmp3\n",
    "\n",
    "    odgi position \\\n",
    "    -i chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og \\\n",
    "    -t 24 \\\n",
    "    -g ${nStart} \\\n",
    "    -r ${chromosome} |tail -n +2|cut -d ',' -f4,5|sed \"s/,/\\t/g\"|cut -f1 >> tmp1\n",
    "\n",
    "    odgi position \\\n",
    "    -i chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og \\\n",
    "    -t 24 \\\n",
    "    -g ${nStop} \\\n",
    "    -r ${chromosome} |tail -n +2|cut -d ',' -f4,5|sed \"s/,/\\t/g\"|cut -f1 >> tmp2\n",
    "done < neighborhoods/chr1_neighborhood.graph.bed\n",
    "\n",
    "paste tmp3 tmp1 tmp2 > neighborhoods/chr1_neighborhood.coordinates.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "4eb0ba45-3d3a-43c9-938e-25d4c172f099",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "at9852_1_chr1:0-33329944\t12039366\t12066573\n",
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cat adr1.neighborhood.coordinates.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "5466b5e4-b968-46bc-9389-aeef8d3d60cd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "at9852_1_chr1:0-33329944\t3148661\t3153437\n",
      "at9852_1_chr1:0-33329944\t3658437\t3664598\n",
      "at9852_1_chr1:0-33329944\t4146752\t4195752\n",
      "at9852_1_chr1:0-33329944\t5486321\t5520209\n",
      "at9852_1_chr1:0-33329944\t6067975\t6081285\n",
      "at9852_1_chr1:0-33329944\t9335621\t9353580\n",
      "at9852_1_chr1:0-33329944\t11132927\t11183727\n",
      "at9852_1_chr1:0-33329944\t12039366\t12066573\n",
      "at9852_1_chr1:0-33329944\t12290158\t12377654\n",
      "at9852_1_chr1:0-33329944\t19621896\t19725905\n",
      "at9852_1_chr1:0-33329944\t20328985\t20357580\n",
      "at9852_1_chr1:0-33329944\t21569880\t21578981\n",
      "at9852_1_chr1:0-33329944\t22066885\t22076210\n",
      "at9852_1_chr1:0-33329944\t22569629\t22575930\n",
      "at9852_1_chr1:0-33329944\t22656847\t22662183\n",
      "at9852_1_chr1:0-33329944\t22850725\t22860978\n",
      "at9852_1_chr1:0-33329944\t24119759\t24165629\n",
      "at9852_1_chr1:0-33329944\t24304282\t24389167\n",
      "at9852_1_chr1:0-33329944\t24611456\t24657983\n",
      "at9852_1_chr1:0-33329944\t24665855\t24887162\n",
      "at9852_1_chr1:0-33329944\t25112270\t25126404\n",
      "at9852_1_chr1:0-33329944\t25388900\t25404594\n",
      "at9852_1_chr1:0-33329944\t25416640\t25503173\n",
      "at9852_1_chr1:0-33329944\t26049762\t26366009\n",
      "at9852_1_chr1:0-33329944\t26496530\t26603880\n",
      "at9852_1_chr1:0-33329944\t26652207\t26673792\n",
      "at9852_1_chr1:0-33329944\t26741177\t26753413\n",
      "at9852_1_chr1:0-33329944\t27176537\t27194539\n",
      "at9852_1_chr1:0-33329944\t27384841\t27392448\n",
      "at9852_1_chr1:0-33329944\t27487655\t27497097\n",
      "at9852_1_chr1:0-33329944\t29014727\t29041233\n",
      "at9852_1_chr1:0-33329944\t30286495\t30325736\n",
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cat neighborhoods/chr1_neighborhood.coordinates.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "f9668334-58eb-4156-a9d9-7371b44abaf6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "at9852_1_chr1:0-33329944:12039366-12066573\n",
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "while read a b c; \n",
    "do\n",
    "   r_path=$(echo \"${a}:${b}-${c}\")\n",
    "    echo ${r_path}\n",
    "done < adr1.neighborhood.coordinates.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "99f2b569-b33d-4079-ab35-8e89bb533bb7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "32304c64-ede9-4274-b4bb-cfefff7f345e",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(pggb_fatica) \n",
      "(pggb_fatica) \n",
      "(pggb_fatica) \n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.60054e+06, 3.60686e+06]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 6.321\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.60054e+06, 3.60686e+06]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 6.321\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.66e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.02e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 7.47e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.62e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.95e+01/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.20332e+06, 4.20957e+06]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 6.244\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.20332e+06, 4.20957e+06]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 6.244\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.36e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 8.49e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.20e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 8.04e+02/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.23e+01/s elapsed: 00:00:00:04 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.82777e+06, 4.90664e+06]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 78.872\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.82777e+06, 4.90664e+06]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 78.872\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.12e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.08e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.15e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.57e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 8.85e+00/s elapsed: 00:00:00:05 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.16e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [6.43704e+06, 6.47465e+06]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 37.617\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [6.43704e+06, 6.47465e+06]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 37.617\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.10e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.16e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.10e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 8.31e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.43e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [7.12533e+06, 7.14525e+06]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 19.922\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [7.12533e+06, 7.14525e+06]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 19.922\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 9.76e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.01e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.10e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 2.09e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 8.43e+00/s elapsed: 00:00:00:06 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.12927e+07, 1.13226e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 29.952\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.12927e+07, 1.13226e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 29.952\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 7.78e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 9.72e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 8.63e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 4.41e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 9.49e+00/s elapsed: 00:00:00:05 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.38509e+07, 1.39719e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 121.022\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.38509e+07, 1.39719e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 121.022\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 7.26e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.05e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 9.96e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 3.94e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.57e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.17e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.56826e+07, 1.57196e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 36.918\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.56826e+07, 1.57196e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 36.918\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 7.87e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.15e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 8.23e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.36e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.12e+01/s elapsed: 00:00:00:04 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.65418e+07, 1.67321e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 190.367\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [1.65418e+07, 1.67321e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 190.367\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 7.22e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 9.59e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.14e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 5.92e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.23e+01/s elapsed: 00:00:00:04 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.13e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.31458e+07, 3.33565e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 210.698\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.31458e+07, 3.33565e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 210.698\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.94e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 8.59e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.12e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 6.53e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.81e+01/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.16e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.45284e+07, 3.46194e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 90.997\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.45284e+07, 3.46194e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 90.997\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.97e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.06e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.03e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 2.05e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.39e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.17e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.64246e+07, 3.64421e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 17.563\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.64246e+07, 3.64421e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 17.563\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.76e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 9.73e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 8.52e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 4.73e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 9.02e+00/s elapsed: 00:00:00:05 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.71665e+07, 3.72079e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 41.445\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.71665e+07, 3.72079e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 41.445\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.35e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 8.45e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 9.29e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 3 (max 3) 100.00% @ 8.13e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 7.07e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.50e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.80948e+07, 3.81099e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 15.089\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.80948e+07, 3.81099e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 15.089\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.73e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 7.98e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 3.63e+00/s elapsed: 00:00:00:04 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.48e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 5.03e+00/s elapsed: 00:00:00:10 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 3.82e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.82386e+07, 3.82442e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 5.577\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.82386e+07, 3.82442e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 5.577\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.71e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 8.57e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.05e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 2.46e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.64e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 1.98e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.85178e+07, 3.85402e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 22.382\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [3.85178e+07, 3.85402e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 22.382\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.70e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.07e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.05e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 4.49e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.31e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.17e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.03171e+07, 4.04588e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 141.699\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.03171e+07, 4.04588e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 141.699\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 2.05e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 8.47e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.22e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 2.82e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 7.96e+00/s elapsed: 00:00:00:06 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.17e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.06198e+07, 4.07988e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 179.036\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.06198e+07, 4.07988e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 179.036\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 2.02e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 8.41e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 9.95e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 5.50e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.38e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.16e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.11744e+07, 4.12804e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 105.984\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.11744e+07, 4.12804e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 105.984\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.99e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 8.36e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 9.02e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 4.28e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.51e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.17e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.12949e+07, 4.1821e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 526.058\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.12949e+07, 4.1821e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 526.058\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.62e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 9.06e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 9.44e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.67e+05/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.45e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.12e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.21931e+07, 4.22216e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 28.558\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.21931e+07, 4.22216e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 28.558\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.66e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.09e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.17e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 3.68e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.56e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.26426e+07, 4.2695e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 52.391\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.26426e+07, 4.2695e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 52.391\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.61e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.05e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.08e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.21e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.63e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.27122e+07, 4.29761e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 263.914\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.27122e+07, 4.29761e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 263.914\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.98e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 7.01e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 4.55e+00/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 9.37e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 6.48e+00/s elapsed: 00:00:00:07 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.14e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.39383e+07, 4.45797e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 641.329\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.39383e+07, 4.45797e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 641.329\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.94e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 8.56e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 9.68e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 3 (max 3) 100.00% @ 1.15e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.91e+05/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.53e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.09e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.47437e+07, 4.4936e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 192.311\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.47437e+07, 4.4936e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 192.311\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.93e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.19e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.05e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 5.01e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.51e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.16e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.49985e+07, 4.50329e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 34.408\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.49985e+07, 4.50329e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 34.408\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.94e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.08e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.01e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 8.98e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 9.77e+00/s elapsed: 00:00:00:05 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.51358e+07, 4.51615e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 25.695\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.51358e+07, 4.51615e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 25.695\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.54e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.29e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.21e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 5.21e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.60e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.58044e+07, 4.58295e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 25.145\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.58044e+07, 4.58295e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 25.145\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.92e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 9.29e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 8.06e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 1.01e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.39e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.60804e+07, 4.60886e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 8.183\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.60804e+07, 4.60886e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 8.183\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.64e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.02e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.15e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 2.28e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.76e+01/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.62495e+07, 4.62674e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 17.891\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.62495e+07, 4.62674e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 17.891\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.65e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.19e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 7.13e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 6.47e+03/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.42e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.8277e+07, 4.83329e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 55.898\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.8277e+07, 4.83329e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 55.898\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.49e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 8.20e+00/s elapsed: 00:00:00:02 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 9.10e+00/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 2.73e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.42e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.17e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.98148e+07, 4.99165e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 101.748\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::viz] Path range to pangenomic range conversion.\n",
      "[odgi::viz] Visualizing the graph in the pangenomic range [4.98148e+07, 4.99165e+07]\n",
      "[odgi::viz] Binned mode\n",
      "[odgi::viz] bin width: 101.748\n",
      "[odgi::viz] image width: 1000\n",
      "[odgi::extract] expansion and adding connecting edges\n",
      "[odgi::extract] extracting path ranges 100.00% @ 1.88e+00/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 1 (max 3) 100.00% @ 1.40e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] merge subpaths closer than 100000 bps - iteration 2 (max 3) 100.00% @ 1.08e+01/s elapsed: 00:00:00:01 remain: 00:00:00:00\n",
      "[odgi::extract] adding connecting edges 100.00% @ 2.50e+04/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "[odgi::extract] adding subpaths 100.00% @ 1.60e+01/s elapsed: 00:00:00:03 remain: 00:00:00:00\n",
      "[odgi::extract] checking missing edges and empty subpaths 100.00% @ 7.18e+01/s elapsed: 00:00:00:00 remain: 00:00:00:00\n",
      "(pggb_fatica) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "inputOdgi=chr1_cluster.p95.s5000.n17.k47.G700-900-1100.at9852.pggbRecommended/chr1.fasta.gz.856e650.e34d4cd.a91cbe1.smooth.final.og\n",
    "outdir=neighborhoods/chr1\n",
    "count=0\n",
    "while read a b c; \n",
    "do\n",
    "    ((count+=1))   \n",
    "    r_path=$(echo \"${a}:${b}-${c}\")\n",
    "    r=${r_path}\n",
    "    nh=$(echo \"nh_${count}\")\n",
    "    outprefix=${nh}.region\n",
    "\n",
    "    ########################## view the pangenome graph of the region ######################\n",
    "    odgi viz \\\n",
    "    -i $inputOdgi \\\n",
    "    -o $outdir/${outprefix}.bm.png \\\n",
    "    -x 1000 \\\n",
    "    -bm \\\n",
    "    -t 20 \\\n",
    "    -P \\\n",
    "    -r ${r}\n",
    "    ########################## Node Strandedness ############################################\n",
    "    odgi viz \\\n",
    "    -i $inputOdgi \\\n",
    "    -o $outdir/${outprefix}.z.png \\\n",
    "    -x 1000 \\\n",
    "    -z \\\n",
    "    -t 20 \\\n",
    "    -P \\\n",
    "    -r ${r}\n",
    "    \n",
    "    ############################################ extract the region ############################################\n",
    "    odgi extract \\\n",
    "    -i $inputOdgi \\\n",
    "    -r ${r} \\\n",
    "    --threads 24 -P \\\n",
    "    -d 100000 \\\n",
    "    -c 1  \\\n",
    "    -o $outdir/${outprefix}.og \n",
    "    \n",
    "    ########################### get the coordinates in all the accessions ########################################\n",
    "    odgi paths -L \\\n",
    "    -i $outdir/${outprefix}.og  \\\n",
    "    > $outdir/${outprefix}.allPathsCoordinates.txt\n",
    "done < neighborhoods/chr1_neighborhood.coordinates.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "613d1bb6-ff84-4e07-8069-33ad6f91a337",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
