{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "c48e0c28-7a8e-46e4-9959-8d16a6f480d6",
   "metadata": {},
   "source": [
    "# Step 4. Merge the fixed annotations"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "490a0343-4b3c-4d94-a2c1-e78a716e5746",
   "metadata": {},
   "source": [
    "Required input:\n",
    "\n",
    "- Original Augustus GFFs\n",
    "- GFFs with fixed gene annotations output from Webapollo\n",
    "- text file with missing gene models"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "00729dad-a82f-4578-97f5-5625c3ad43bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "conda activate dl20-annomerge"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2bebf7c2-2de5-49ed-921d-87c4a147e52f",
   "metadata": {},
   "source": [
    "1. first merge the original GFFs with the gene models that were manually fixed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "3e036071-b8bb-4806-a459-b6377866faf7",
   "metadata": {},
   "outputs": [],
   "source": [
    "dl20=(\n",
    "at6137 at6923 at6929 at7143\n",
    "at8285 at9104 at9336 at9503\n",
    "at9578 at9744 at9762 at9806\n",
    "at9830 at9847 at9852 at9879\n",
    "at9883 at9900\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "5cf68863-9b4e-4d60-8c49-9868417e774c",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir -p output/02_annotation/02_manual-curation-updated"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1bc56758-41da-4cfd-acc7-686c59690bd2",
   "metadata": {},
   "source": [
    "Only twelve accessions had NLR gene models which could be manually fixed without RNA evidence. The fixed gffs are per chromosome."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "74948c9a-5bf5-46c7-9327-d8a8bddd0568",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "at6137 has fixes for this many genes: 3\n",
      "INFO     Checking gff file (tmp/02_annotation/02_manual-curation-updated/at6137.everything.pansn.gff3)...\n",
      "INFO     Checking genome fasta (output/01_assembly/02_masked/at6137.scaffolds.unmasked.fasta)...\n",
      "INFO     Reading gff files: (tmp/02_annotation/02_manual-curation-updated/at6137.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Checking errors in the gff files: (tmp/02_annotation/02_manual-curation-updated/at6137.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking missing attributes: (function4gff.FIX_MISSING_ATTR())...\n",
      "\n",
      "INFO     \t- Checking intra-model errors: (tmp/02_annotation/02_manual-curation-updated/at6137.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking inter-model errors: (tmp/02_annotation/02_manual-curation-updated/at6137.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Specifying prefix of output file name: (tmp)...\n",
      "INFO     Reading files: tmp/02_annotation/02_manual-curation-updated/at6137.everything.pansn.gff3, output/01_assembly/02_masked/at6137.scaffolds.unmasked.fasta...\n",
      "INFO     Extract sequences for cds...\n",
      "INFO     Print out extracted sequences: tmp_cds.fa...\n",
      "INFO     Making blast database... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/makeblastdb)\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/29/2022 15:51:39\n",
      "New DB name:   /home/luisa/work/detlef/difflines_annex/assembly-and-annotation/tmp_cds.fa\n",
      "New DB title:  tmp_cds.fa\n",
      "Sequence type: Nucleotide\n",
      "Keep Linkouts: T\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 3 sequences in 0.000423908 seconds.\n",
      "INFO     Aligning sequences... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/blastn)\n",
      "INFO     Finding mRNAs with wrongly split gene parents... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/check_gene_parent/find_wrongly_split_gene_parent.pl)\n",
      "Reading the gff file: tmp/02_annotation/02_manual-curation-updated/at6137.everything.pansn.gff3...\n",
      "Reading the blast file: blastn.out...\n",
      "INFO     Removing unnecessary files...\n",
      "INFO     \t- Checking single-feature errors: (tmp/02_annotation/02_manual-curation-updated/at6137.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Print QC report at tmp/02_annotation/02_manual-curation-updated/at6137.everything.pansn.gff3.qcreport.txt\n",
      "INFO     Print QC statistic report at tmp/02_annotation/02_manual-curation-updated/at6137.everything.pansn.gff3.statistic.txt\n",
      "INFO     Checking QC report file (tmp/02_annotation/02_manual-curation-updated/at6137.everything.pansn.gff3.qcreport.txt)...\n",
      "INFO     Checking GFF3 file (tmp/02_annotation/02_manual-curation-updated/at6137.everything.pansn.gff3)...\n",
      "INFO     Reading QC report file: (tmp/02_annotation/02_manual-curation-updated/at6137.everything.pansn.gff3.qcreport.txt)...\n",
      "\n",
      "INFO     Reading GFF3 file: (tmp/02_annotation/02_manual-curation-updated/at6137.everything.pansn.gff3)...\n",
      "\n",
      "warning: seqid \"at6137_1_chr4\" on line 2 in file \"output/02_annotation/02_manual-curation-updated/at6137/at6137.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at6137_9_u81\" on line 12 in file \"output/02_annotation/02_manual-curation-updated/at6137/at6137.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at6137_1_chr2\" on line 17 in file \"output/02_annotation/02_manual-curation-updated/at6137/at6137.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at6137_1_chr5\" on line 57 in file \"output/02_annotation/02_manual-curation-updated/at6137/at6137.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at6137_1_chr3\" on line 99 in file \"output/02_annotation/02_manual-curation-updated/at6137/at6137.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at6137_1_chr1\" on line 151 in file \"output/02_annotation/02_manual-curation-updated/at6137/at6137.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "at6923 has fixes for this many genes: 1\n",
      "INFO     Checking gff file (tmp/02_annotation/02_manual-curation-updated/at6923.everything.pansn.gff3)...\n",
      "INFO     Checking genome fasta (output/01_assembly/02_masked/at6923.scaffolds.unmasked.fasta)...\n",
      "INFO     Reading gff files: (tmp/02_annotation/02_manual-curation-updated/at6923.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Checking errors in the gff files: (tmp/02_annotation/02_manual-curation-updated/at6923.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking missing attributes: (function4gff.FIX_MISSING_ATTR())...\n",
      "\n",
      "INFO     \t- Checking intra-model errors: (tmp/02_annotation/02_manual-curation-updated/at6923.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking inter-model errors: (tmp/02_annotation/02_manual-curation-updated/at6923.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Specifying prefix of output file name: (tmp)...\n",
      "INFO     Reading files: tmp/02_annotation/02_manual-curation-updated/at6923.everything.pansn.gff3, output/01_assembly/02_masked/at6923.scaffolds.unmasked.fasta...\n",
      "INFO     Extract sequences for cds...\n",
      "INFO     Print out extracted sequences: tmp_cds.fa...\n",
      "INFO     Making blast database... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/makeblastdb)\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/29/2022 16:00:46\n",
      "New DB name:   /home/luisa/work/detlef/difflines_annex/assembly-and-annotation/tmp_cds.fa\n",
      "New DB title:  tmp_cds.fa\n",
      "Sequence type: Nucleotide\n",
      "Keep Linkouts: T\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.000187874 seconds.\n",
      "INFO     Aligning sequences... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/blastn)\n",
      "INFO     Finding mRNAs with wrongly split gene parents... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/check_gene_parent/find_wrongly_split_gene_parent.pl)\n",
      "Reading the gff file: tmp/02_annotation/02_manual-curation-updated/at6923.everything.pansn.gff3...\n",
      "Reading the blast file: blastn.out...\n",
      "INFO     Removing unnecessary files...\n",
      "INFO     \t- Checking single-feature errors: (tmp/02_annotation/02_manual-curation-updated/at6923.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Print QC report at tmp/02_annotation/02_manual-curation-updated/at6923.everything.pansn.gff3.qcreport.txt\n",
      "INFO     Print QC statistic report at tmp/02_annotation/02_manual-curation-updated/at6923.everything.pansn.gff3.statistic.txt\n",
      "INFO     Checking QC report file (tmp/02_annotation/02_manual-curation-updated/at6923.everything.pansn.gff3.qcreport.txt)...\n",
      "INFO     Checking GFF3 file (tmp/02_annotation/02_manual-curation-updated/at6923.everything.pansn.gff3)...\n",
      "INFO     Reading QC report file: (tmp/02_annotation/02_manual-curation-updated/at6923.everything.pansn.gff3.qcreport.txt)...\n",
      "\n",
      "INFO     Reading GFF3 file: (tmp/02_annotation/02_manual-curation-updated/at6923.everything.pansn.gff3)...\n",
      "\n",
      "warning: seqid \"at6923_1_chr2\" on line 2 in file \"output/02_annotation/02_manual-curation-updated/at6923/at6923.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at6923_9_u29\" on line 10 in file \"output/02_annotation/02_manual-curation-updated/at6923/at6923.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at6923_1_chr3\" on line 56 in file \"output/02_annotation/02_manual-curation-updated/at6923/at6923.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at6923_1_chr5\" on line 76 in file \"output/02_annotation/02_manual-curation-updated/at6923/at6923.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at6923_1_chr1\" on line 136 in file \"output/02_annotation/02_manual-curation-updated/at6923/at6923.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at6923_9_u94\" on line 174 in file \"output/02_annotation/02_manual-curation-updated/at6923/at6923.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at6923_1_chr4\" on line 1196 in file \"output/02_annotation/02_manual-curation-updated/at6923/at6923.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "at6929 has fixes for this many genes: 1\n",
      "INFO     Checking gff file (tmp/02_annotation/02_manual-curation-updated/at6929.everything.pansn.gff3)...\n",
      "INFO     Checking genome fasta (output/01_assembly/02_masked/at6929.scaffolds.unmasked.fasta)...\n",
      "INFO     Reading gff files: (tmp/02_annotation/02_manual-curation-updated/at6929.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Checking errors in the gff files: (tmp/02_annotation/02_manual-curation-updated/at6929.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking missing attributes: (function4gff.FIX_MISSING_ATTR())...\n",
      "\n",
      "INFO     \t- Checking intra-model errors: (tmp/02_annotation/02_manual-curation-updated/at6929.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking inter-model errors: (tmp/02_annotation/02_manual-curation-updated/at6929.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Specifying prefix of output file name: (tmp)...\n",
      "INFO     Reading files: tmp/02_annotation/02_manual-curation-updated/at6929.everything.pansn.gff3, output/01_assembly/02_masked/at6929.scaffolds.unmasked.fasta...\n",
      "INFO     Extract sequences for cds...\n",
      "INFO     Print out extracted sequences: tmp_cds.fa...\n",
      "INFO     Making blast database... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/makeblastdb)\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/29/2022 16:10:00\n",
      "New DB name:   /home/luisa/work/detlef/difflines_annex/assembly-and-annotation/tmp_cds.fa\n",
      "New DB title:  tmp_cds.fa\n",
      "Sequence type: Nucleotide\n",
      "Keep Linkouts: T\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.000154972 seconds.\n",
      "INFO     Aligning sequences... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/blastn)\n",
      "INFO     Finding mRNAs with wrongly split gene parents... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/check_gene_parent/find_wrongly_split_gene_parent.pl)\n",
      "Reading the gff file: tmp/02_annotation/02_manual-curation-updated/at6929.everything.pansn.gff3...\n",
      "Reading the blast file: blastn.out...\n",
      "INFO     Removing unnecessary files...\n",
      "INFO     \t- Checking single-feature errors: (tmp/02_annotation/02_manual-curation-updated/at6929.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Print QC report at tmp/02_annotation/02_manual-curation-updated/at6929.everything.pansn.gff3.qcreport.txt\n",
      "INFO     Print QC statistic report at tmp/02_annotation/02_manual-curation-updated/at6929.everything.pansn.gff3.statistic.txt\n",
      "INFO     Checking QC report file (tmp/02_annotation/02_manual-curation-updated/at6929.everything.pansn.gff3.qcreport.txt)...\n",
      "INFO     Checking GFF3 file (tmp/02_annotation/02_manual-curation-updated/at6929.everything.pansn.gff3)...\n",
      "INFO     Reading QC report file: (tmp/02_annotation/02_manual-curation-updated/at6929.everything.pansn.gff3.qcreport.txt)...\n",
      "\n",
      "INFO     Reading GFF3 file: (tmp/02_annotation/02_manual-curation-updated/at6929.everything.pansn.gff3)...\n",
      "\n",
      "warning: seqid \"at6929_1_chr2\" on line 2 in file \"output/02_annotation/02_manual-curation-updated/at6929/at6929.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at6929_1_chr4\" on line 14 in file \"output/02_annotation/02_manual-curation-updated/at6929/at6929.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at6929_9_u1037\" on line 24 in file \"output/02_annotation/02_manual-curation-updated/at6929/at6929.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at6929_9_u1687\" on line 29 in file \"output/02_annotation/02_manual-curation-updated/at6929/at6929.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at6929_9_u454\" on line 38 in file \"output/02_annotation/02_manual-curation-updated/at6929/at6929.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at6929_9_u732\" on line 44 in file \"output/02_annotation/02_manual-curation-updated/at6929/at6929.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at6929_9_u745\" on line 49 in file \"output/02_annotation/02_manual-curation-updated/at6929/at6929.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at6929_9_u979\" on line 56 in file \"output/02_annotation/02_manual-curation-updated/at6929/at6929.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at6929_1_chr3\" on line 289 in file \"output/02_annotation/02_manual-curation-updated/at6929/at6929.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at6929_1_chr5\" on line 341 in file \"output/02_annotation/02_manual-curation-updated/at6929/at6929.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at6929_1_chr1\" on line 407 in file \"output/02_annotation/02_manual-curation-updated/at6929/at6929.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "at7143 has fixes for this many genes: 1\n",
      "INFO     Checking gff file (tmp/02_annotation/02_manual-curation-updated/at7143.everything.pansn.gff3)...\n",
      "INFO     Checking genome fasta (output/01_assembly/02_masked/at7143.scaffolds.unmasked.fasta)...\n",
      "INFO     Reading gff files: (tmp/02_annotation/02_manual-curation-updated/at7143.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Checking errors in the gff files: (tmp/02_annotation/02_manual-curation-updated/at7143.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking missing attributes: (function4gff.FIX_MISSING_ATTR())...\n",
      "\n",
      "INFO     \t- Checking intra-model errors: (tmp/02_annotation/02_manual-curation-updated/at7143.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking inter-model errors: (tmp/02_annotation/02_manual-curation-updated/at7143.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Specifying prefix of output file name: (tmp)...\n",
      "INFO     Reading files: tmp/02_annotation/02_manual-curation-updated/at7143.everything.pansn.gff3, output/01_assembly/02_masked/at7143.scaffolds.unmasked.fasta...\n",
      "INFO     Extract sequences for cds...\n",
      "INFO     Print out extracted sequences: tmp_cds.fa...\n",
      "INFO     Making blast database... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/makeblastdb)\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/29/2022 16:19:06\n",
      "New DB name:   /home/luisa/work/detlef/difflines_annex/assembly-and-annotation/tmp_cds.fa\n",
      "New DB title:  tmp_cds.fa\n",
      "Sequence type: Nucleotide\n",
      "Keep Linkouts: T\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.00019908 seconds.\n",
      "INFO     Aligning sequences... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/blastn)\n",
      "INFO     Finding mRNAs with wrongly split gene parents... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/check_gene_parent/find_wrongly_split_gene_parent.pl)\n",
      "Reading the gff file: tmp/02_annotation/02_manual-curation-updated/at7143.everything.pansn.gff3...\n",
      "Reading the blast file: blastn.out...\n",
      "INFO     Removing unnecessary files...\n",
      "INFO     \t- Checking single-feature errors: (tmp/02_annotation/02_manual-curation-updated/at7143.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Print QC report at tmp/02_annotation/02_manual-curation-updated/at7143.everything.pansn.gff3.qcreport.txt\n",
      "INFO     Print QC statistic report at tmp/02_annotation/02_manual-curation-updated/at7143.everything.pansn.gff3.statistic.txt\n",
      "INFO     Checking QC report file (tmp/02_annotation/02_manual-curation-updated/at7143.everything.pansn.gff3.qcreport.txt)...\n",
      "INFO     Checking GFF3 file (tmp/02_annotation/02_manual-curation-updated/at7143.everything.pansn.gff3)...\n",
      "INFO     Reading QC report file: (tmp/02_annotation/02_manual-curation-updated/at7143.everything.pansn.gff3.qcreport.txt)...\n",
      "\n",
      "INFO     Reading GFF3 file: (tmp/02_annotation/02_manual-curation-updated/at7143.everything.pansn.gff3)...\n",
      "\n",
      "warning: seqid \"at7143_1_chr2\" on line 2 in file \"output/02_annotation/02_manual-curation-updated/at7143/at7143.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at7143_9_u29\" on line 20 in file \"output/02_annotation/02_manual-curation-updated/at7143/at7143.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at7143_1_chr5\" on line 64 in file \"output/02_annotation/02_manual-curation-updated/at7143/at7143.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at7143_1_chr3\" on line 98 in file \"output/02_annotation/02_manual-curation-updated/at7143/at7143.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at7143_1_chr1\" on line 156 in file \"output/02_annotation/02_manual-curation-updated/at7143/at7143.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at7143_9_u284\" on line 178 in file \"output/02_annotation/02_manual-curation-updated/at7143/at7143.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at7143_1_chr4\" on line 398 in file \"output/02_annotation/02_manual-curation-updated/at7143/at7143.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "at8285 has fixes for this many genes: 1\n",
      "INFO     Checking gff file (tmp/02_annotation/02_manual-curation-updated/at8285.everything.pansn.gff3)...\n",
      "INFO     Checking genome fasta (output/01_assembly/02_masked/at8285.scaffolds.unmasked.fasta)...\n",
      "INFO     Reading gff files: (tmp/02_annotation/02_manual-curation-updated/at8285.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Checking errors in the gff files: (tmp/02_annotation/02_manual-curation-updated/at8285.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking missing attributes: (function4gff.FIX_MISSING_ATTR())...\n",
      "\n",
      "INFO     \t- Checking intra-model errors: (tmp/02_annotation/02_manual-curation-updated/at8285.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking inter-model errors: (tmp/02_annotation/02_manual-curation-updated/at8285.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Specifying prefix of output file name: (tmp)...\n",
      "INFO     Reading files: tmp/02_annotation/02_manual-curation-updated/at8285.everything.pansn.gff3, output/01_assembly/02_masked/at8285.scaffolds.unmasked.fasta...\n",
      "INFO     Extract sequences for cds...\n",
      "INFO     Print out extracted sequences: tmp_cds.fa...\n",
      "INFO     Making blast database... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/makeblastdb)\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/29/2022 16:29:07\n",
      "New DB name:   /home/luisa/work/detlef/difflines_annex/assembly-and-annotation/tmp_cds.fa\n",
      "New DB title:  tmp_cds.fa\n",
      "Sequence type: Nucleotide\n",
      "Keep Linkouts: T\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.000236034 seconds.\n",
      "INFO     Aligning sequences... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/blastn)\n",
      "INFO     Finding mRNAs with wrongly split gene parents... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/check_gene_parent/find_wrongly_split_gene_parent.pl)\n",
      "Reading the gff file: tmp/02_annotation/02_manual-curation-updated/at8285.everything.pansn.gff3...\n",
      "Reading the blast file: blastn.out...\n",
      "INFO     Removing unnecessary files...\n",
      "INFO     \t- Checking single-feature errors: (tmp/02_annotation/02_manual-curation-updated/at8285.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Print QC report at tmp/02_annotation/02_manual-curation-updated/at8285.everything.pansn.gff3.qcreport.txt\n",
      "INFO     Print QC statistic report at tmp/02_annotation/02_manual-curation-updated/at8285.everything.pansn.gff3.statistic.txt\n",
      "INFO     Checking QC report file (tmp/02_annotation/02_manual-curation-updated/at8285.everything.pansn.gff3.qcreport.txt)...\n",
      "INFO     Checking GFF3 file (tmp/02_annotation/02_manual-curation-updated/at8285.everything.pansn.gff3)...\n",
      "INFO     Reading QC report file: (tmp/02_annotation/02_manual-curation-updated/at8285.everything.pansn.gff3.qcreport.txt)...\n",
      "\n",
      "INFO     Reading GFF3 file: (tmp/02_annotation/02_manual-curation-updated/at8285.everything.pansn.gff3)...\n",
      "\n",
      "warning: seqid \"at8285_1_chr3\" on line 2 in file \"output/02_annotation/02_manual-curation-updated/at8285/at8285.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at8285_1_chr5\" on line 22 in file \"output/02_annotation/02_manual-curation-updated/at8285/at8285.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at8285_1_chr1\" on line 56 in file \"output/02_annotation/02_manual-curation-updated/at8285/at8285.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at8285_1_chr2\" on line 674 in file \"output/02_annotation/02_manual-curation-updated/at8285/at8285.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at8285_1_chr4\" on line 1114 in file \"output/02_annotation/02_manual-curation-updated/at8285/at8285.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "at9104 has fixes for this many genes: 2\n",
      "INFO     Checking gff file (tmp/02_annotation/02_manual-curation-updated/at9104.everything.pansn.gff3)...\n",
      "INFO     Checking genome fasta (output/01_assembly/02_masked/at9104.scaffolds.unmasked.fasta)...\n",
      "INFO     Reading gff files: (tmp/02_annotation/02_manual-curation-updated/at9104.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Checking errors in the gff files: (tmp/02_annotation/02_manual-curation-updated/at9104.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking missing attributes: (function4gff.FIX_MISSING_ATTR())...\n",
      "\n",
      "INFO     \t- Checking intra-model errors: (tmp/02_annotation/02_manual-curation-updated/at9104.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking inter-model errors: (tmp/02_annotation/02_manual-curation-updated/at9104.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Specifying prefix of output file name: (tmp)...\n",
      "INFO     Reading files: tmp/02_annotation/02_manual-curation-updated/at9104.everything.pansn.gff3, output/01_assembly/02_masked/at9104.scaffolds.unmasked.fasta...\n",
      "INFO     Extract sequences for cds...\n",
      "INFO     Print out extracted sequences: tmp_cds.fa...\n",
      "INFO     Making blast database... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/makeblastdb)\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/29/2022 16:39:07\n",
      "New DB name:   /home/luisa/work/detlef/difflines_annex/assembly-and-annotation/tmp_cds.fa\n",
      "New DB title:  tmp_cds.fa\n",
      "Sequence type: Nucleotide\n",
      "Keep Linkouts: T\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 2 sequences in 0.00060606 seconds.\n",
      "INFO     Aligning sequences... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/blastn)\n",
      "INFO     Finding mRNAs with wrongly split gene parents... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/check_gene_parent/find_wrongly_split_gene_parent.pl)\n",
      "Reading the gff file: tmp/02_annotation/02_manual-curation-updated/at9104.everything.pansn.gff3...\n",
      "Reading the blast file: blastn.out...\n",
      "INFO     Removing unnecessary files...\n",
      "INFO     \t- Checking single-feature errors: (tmp/02_annotation/02_manual-curation-updated/at9104.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Print QC report at tmp/02_annotation/02_manual-curation-updated/at9104.everything.pansn.gff3.qcreport.txt\n",
      "INFO     Print QC statistic report at tmp/02_annotation/02_manual-curation-updated/at9104.everything.pansn.gff3.statistic.txt\n",
      "INFO     Checking QC report file (tmp/02_annotation/02_manual-curation-updated/at9104.everything.pansn.gff3.qcreport.txt)...\n",
      "INFO     Checking GFF3 file (tmp/02_annotation/02_manual-curation-updated/at9104.everything.pansn.gff3)...\n",
      "INFO     Reading QC report file: (tmp/02_annotation/02_manual-curation-updated/at9104.everything.pansn.gff3.qcreport.txt)...\n",
      "\n",
      "INFO     Reading GFF3 file: (tmp/02_annotation/02_manual-curation-updated/at9104.everything.pansn.gff3)...\n",
      "\n",
      "warning: seqid \"at9104_1_chr2\" on line 2 in file \"output/02_annotation/02_manual-curation-updated/at9104/at9104.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9104_1_chr4\" on line 9 in file \"output/02_annotation/02_manual-curation-updated/at9104/at9104.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9104_1_chr3\" on line 105 in file \"output/02_annotation/02_manual-curation-updated/at9104/at9104.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9104_1_chr5\" on line 125 in file \"output/02_annotation/02_manual-curation-updated/at9104/at9104.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9104_1_chr1\" on line 183 in file \"output/02_annotation/02_manual-curation-updated/at9104/at9104.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9104_9_u52\" on line 471 in file \"output/02_annotation/02_manual-curation-updated/at9104/at9104.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "at9336 has fixes for this many genes: 1\n",
      "INFO     Checking gff file (tmp/02_annotation/02_manual-curation-updated/at9336.everything.pansn.gff3)...\n",
      "INFO     Checking genome fasta (output/01_assembly/02_masked/at9336.scaffolds.unmasked.fasta)...\n",
      "INFO     Reading gff files: (tmp/02_annotation/02_manual-curation-updated/at9336.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Checking errors in the gff files: (tmp/02_annotation/02_manual-curation-updated/at9336.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking missing attributes: (function4gff.FIX_MISSING_ATTR())...\n",
      "\n",
      "INFO     \t- Checking intra-model errors: (tmp/02_annotation/02_manual-curation-updated/at9336.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking inter-model errors: (tmp/02_annotation/02_manual-curation-updated/at9336.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Specifying prefix of output file name: (tmp)...\n",
      "INFO     Reading files: tmp/02_annotation/02_manual-curation-updated/at9336.everything.pansn.gff3, output/01_assembly/02_masked/at9336.scaffolds.unmasked.fasta...\n",
      "INFO     Extract sequences for cds...\n",
      "INFO     Print out extracted sequences: tmp_cds.fa...\n",
      "INFO     Making blast database... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/makeblastdb)\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/29/2022 16:48:11\n",
      "New DB name:   /home/luisa/work/detlef/difflines_annex/assembly-and-annotation/tmp_cds.fa\n",
      "New DB title:  tmp_cds.fa\n",
      "Sequence type: Nucleotide\n",
      "Keep Linkouts: T\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.000221968 seconds.\n",
      "INFO     Aligning sequences... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/blastn)\n",
      "INFO     Finding mRNAs with wrongly split gene parents... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/check_gene_parent/find_wrongly_split_gene_parent.pl)\n",
      "Reading the gff file: tmp/02_annotation/02_manual-curation-updated/at9336.everything.pansn.gff3...\n",
      "Reading the blast file: blastn.out...\n",
      "INFO     Removing unnecessary files...\n",
      "INFO     \t- Checking single-feature errors: (tmp/02_annotation/02_manual-curation-updated/at9336.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Print QC report at tmp/02_annotation/02_manual-curation-updated/at9336.everything.pansn.gff3.qcreport.txt\n",
      "INFO     Print QC statistic report at tmp/02_annotation/02_manual-curation-updated/at9336.everything.pansn.gff3.statistic.txt\n",
      "INFO     Checking QC report file (tmp/02_annotation/02_manual-curation-updated/at9336.everything.pansn.gff3.qcreport.txt)...\n",
      "INFO     Checking GFF3 file (tmp/02_annotation/02_manual-curation-updated/at9336.everything.pansn.gff3)...\n",
      "INFO     Reading QC report file: (tmp/02_annotation/02_manual-curation-updated/at9336.everything.pansn.gff3.qcreport.txt)...\n",
      "\n",
      "INFO     Reading GFF3 file: (tmp/02_annotation/02_manual-curation-updated/at9336.everything.pansn.gff3)...\n",
      "\n",
      "warning: seqid \"at9336_1_chr4\" on line 2 in file \"output/02_annotation/02_manual-curation-updated/at9336/at9336.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9336_9_u116\" on line 8 in file \"output/02_annotation/02_manual-curation-updated/at9336/at9336.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9336_9_u143\" on line 15 in file \"output/02_annotation/02_manual-curation-updated/at9336/at9336.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9336_9_u153\" on line 21 in file \"output/02_annotation/02_manual-curation-updated/at9336/at9336.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9336_9_u166\" on line 26 in file \"output/02_annotation/02_manual-curation-updated/at9336/at9336.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9336_9_u244\" on line 34 in file \"output/02_annotation/02_manual-curation-updated/at9336/at9336.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9336_9_u258\" on line 39 in file \"output/02_annotation/02_manual-curation-updated/at9336/at9336.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9336_9_u420\" on line 44 in file \"output/02_annotation/02_manual-curation-updated/at9336/at9336.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9336_9_u476\" on line 49 in file \"output/02_annotation/02_manual-curation-updated/at9336/at9336.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9336_9_u97\" on line 54 in file \"output/02_annotation/02_manual-curation-updated/at9336/at9336.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9336_9_u45\" on line 59 in file \"output/02_annotation/02_manual-curation-updated/at9336/at9336.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9336_9_u63\" on line 65 in file \"output/02_annotation/02_manual-curation-updated/at9336/at9336.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9336_1_chr2\" on line 73 in file \"output/02_annotation/02_manual-curation-updated/at9336/at9336.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9336_1_chr3\" on line 457 in file \"output/02_annotation/02_manual-curation-updated/at9336/at9336.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9336_1_chr5\" on line 513 in file \"output/02_annotation/02_manual-curation-updated/at9336/at9336.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9336_1_chr1\" on line 693 in file \"output/02_annotation/02_manual-curation-updated/at9336/at9336.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9336_9_u62\" on line 927 in file \"output/02_annotation/02_manual-curation-updated/at9336/at9336.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9336_9_u202\" on line 1693 in file \"output/02_annotation/02_manual-curation-updated/at9336/at9336.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "at9806 has fixes for this many genes: 2\n",
      "INFO     Checking gff file (tmp/02_annotation/02_manual-curation-updated/at9806.everything.pansn.gff3)...\n",
      "INFO     Checking genome fasta (output/01_assembly/02_masked/at9806.scaffolds.unmasked.fasta)...\n",
      "INFO     Reading gff files: (tmp/02_annotation/02_manual-curation-updated/at9806.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Checking errors in the gff files: (tmp/02_annotation/02_manual-curation-updated/at9806.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking missing attributes: (function4gff.FIX_MISSING_ATTR())...\n",
      "\n",
      "INFO     \t- Checking intra-model errors: (tmp/02_annotation/02_manual-curation-updated/at9806.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking inter-model errors: (tmp/02_annotation/02_manual-curation-updated/at9806.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Specifying prefix of output file name: (tmp)...\n",
      "INFO     Reading files: tmp/02_annotation/02_manual-curation-updated/at9806.everything.pansn.gff3, output/01_assembly/02_masked/at9806.scaffolds.unmasked.fasta...\n",
      "INFO     Extract sequences for cds...\n",
      "INFO     Print out extracted sequences: tmp_cds.fa...\n",
      "INFO     Making blast database... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/makeblastdb)\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/29/2022 16:57:38\n",
      "New DB name:   /home/luisa/work/detlef/difflines_annex/assembly-and-annotation/tmp_cds.fa\n",
      "New DB title:  tmp_cds.fa\n",
      "Sequence type: Nucleotide\n",
      "Keep Linkouts: T\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 2 sequences in 0.000350952 seconds.\n",
      "INFO     Aligning sequences... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/blastn)\n",
      "INFO     Finding mRNAs with wrongly split gene parents... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/check_gene_parent/find_wrongly_split_gene_parent.pl)\n",
      "Reading the gff file: tmp/02_annotation/02_manual-curation-updated/at9806.everything.pansn.gff3...\n",
      "Reading the blast file: blastn.out...\n",
      "INFO     Removing unnecessary files...\n",
      "INFO     \t- Checking single-feature errors: (tmp/02_annotation/02_manual-curation-updated/at9806.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Print QC report at tmp/02_annotation/02_manual-curation-updated/at9806.everything.pansn.gff3.qcreport.txt\n",
      "INFO     Print QC statistic report at tmp/02_annotation/02_manual-curation-updated/at9806.everything.pansn.gff3.statistic.txt\n",
      "INFO     Checking QC report file (tmp/02_annotation/02_manual-curation-updated/at9806.everything.pansn.gff3.qcreport.txt)...\n",
      "INFO     Checking GFF3 file (tmp/02_annotation/02_manual-curation-updated/at9806.everything.pansn.gff3)...\n",
      "INFO     Reading QC report file: (tmp/02_annotation/02_manual-curation-updated/at9806.everything.pansn.gff3.qcreport.txt)...\n",
      "\n",
      "INFO     Reading GFF3 file: (tmp/02_annotation/02_manual-curation-updated/at9806.everything.pansn.gff3)...\n",
      "\n",
      "warning: seqid \"at9806_9_u28\" on line 2 in file \"output/02_annotation/02_manual-curation-updated/at9806/at9806.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9806_9_u30\" on line 18 in file \"output/02_annotation/02_manual-curation-updated/at9806/at9806.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9806_1_chr4\" on line 34 in file \"output/02_annotation/02_manual-curation-updated/at9806/at9806.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9806_1_chr2\" on line 44 in file \"output/02_annotation/02_manual-curation-updated/at9806/at9806.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9806_1_chr3\" on line 168 in file \"output/02_annotation/02_manual-curation-updated/at9806/at9806.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9806_1_chr5\" on line 188 in file \"output/02_annotation/02_manual-curation-updated/at9806/at9806.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9806_1_chr1\" on line 248 in file \"output/02_annotation/02_manual-curation-updated/at9806/at9806.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "at9830 has fixes for this many genes: 1\n",
      "INFO     Checking gff file (tmp/02_annotation/02_manual-curation-updated/at9830.everything.pansn.gff3)...\n",
      "INFO     Checking genome fasta (output/01_assembly/02_masked/at9830.scaffolds.unmasked.fasta)...\n",
      "INFO     Reading gff files: (tmp/02_annotation/02_manual-curation-updated/at9830.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Checking errors in the gff files: (tmp/02_annotation/02_manual-curation-updated/at9830.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking missing attributes: (function4gff.FIX_MISSING_ATTR())...\n",
      "\n",
      "INFO     \t- Checking intra-model errors: (tmp/02_annotation/02_manual-curation-updated/at9830.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking inter-model errors: (tmp/02_annotation/02_manual-curation-updated/at9830.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Specifying prefix of output file name: (tmp)...\n",
      "INFO     Reading files: tmp/02_annotation/02_manual-curation-updated/at9830.everything.pansn.gff3, output/01_assembly/02_masked/at9830.scaffolds.unmasked.fasta...\n",
      "INFO     Extract sequences for cds...\n",
      "INFO     Print out extracted sequences: tmp_cds.fa...\n",
      "INFO     Making blast database... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/makeblastdb)\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/29/2022 17:06:44\n",
      "New DB name:   /home/luisa/work/detlef/difflines_annex/assembly-and-annotation/tmp_cds.fa\n",
      "New DB title:  tmp_cds.fa\n",
      "Sequence type: Nucleotide\n",
      "Keep Linkouts: T\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.000149965 seconds.\n",
      "INFO     Aligning sequences... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/blastn)\n",
      "INFO     Finding mRNAs with wrongly split gene parents... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/check_gene_parent/find_wrongly_split_gene_parent.pl)\n",
      "Reading the gff file: tmp/02_annotation/02_manual-curation-updated/at9830.everything.pansn.gff3...\n",
      "Reading the blast file: blastn.out...\n",
      "INFO     Removing unnecessary files...\n",
      "INFO     \t- Checking single-feature errors: (tmp/02_annotation/02_manual-curation-updated/at9830.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Print QC report at tmp/02_annotation/02_manual-curation-updated/at9830.everything.pansn.gff3.qcreport.txt\n",
      "INFO     Print QC statistic report at tmp/02_annotation/02_manual-curation-updated/at9830.everything.pansn.gff3.statistic.txt\n",
      "INFO     Checking QC report file (tmp/02_annotation/02_manual-curation-updated/at9830.everything.pansn.gff3.qcreport.txt)...\n",
      "INFO     Checking GFF3 file (tmp/02_annotation/02_manual-curation-updated/at9830.everything.pansn.gff3)...\n",
      "INFO     Reading QC report file: (tmp/02_annotation/02_manual-curation-updated/at9830.everything.pansn.gff3.qcreport.txt)...\n",
      "\n",
      "INFO     Reading GFF3 file: (tmp/02_annotation/02_manual-curation-updated/at9830.everything.pansn.gff3)...\n",
      "\n",
      "warning: seqid \"at9830_9_u1136\" on line 2 in file \"output/02_annotation/02_manual-curation-updated/at9830/at9830.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9830_1_chr4\" on line 8 in file \"output/02_annotation/02_manual-curation-updated/at9830/at9830.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9830_9_u470\" on line 42 in file \"output/02_annotation/02_manual-curation-updated/at9830/at9830.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9830_1_chr3\" on line 108 in file \"output/02_annotation/02_manual-curation-updated/at9830/at9830.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9830_9_u393\" on line 128 in file \"output/02_annotation/02_manual-curation-updated/at9830/at9830.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9830_1_chr5\" on line 148 in file \"output/02_annotation/02_manual-curation-updated/at9830/at9830.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9830_1_chr1\" on line 194 in file \"output/02_annotation/02_manual-curation-updated/at9830/at9830.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9830_1_chr2\" on line 1014 in file \"output/02_annotation/02_manual-curation-updated/at9830/at9830.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "at9847 has fixes for this many genes: 1\n",
      "INFO     Checking gff file (tmp/02_annotation/02_manual-curation-updated/at9847.everything.pansn.gff3)...\n",
      "INFO     Checking genome fasta (output/01_assembly/02_masked/at9847.scaffolds.unmasked.fasta)...\n",
      "INFO     Reading gff files: (tmp/02_annotation/02_manual-curation-updated/at9847.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Checking errors in the gff files: (tmp/02_annotation/02_manual-curation-updated/at9847.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking missing attributes: (function4gff.FIX_MISSING_ATTR())...\n",
      "\n",
      "INFO     \t- Checking intra-model errors: (tmp/02_annotation/02_manual-curation-updated/at9847.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking inter-model errors: (tmp/02_annotation/02_manual-curation-updated/at9847.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Specifying prefix of output file name: (tmp)...\n",
      "INFO     Reading files: tmp/02_annotation/02_manual-curation-updated/at9847.everything.pansn.gff3, output/01_assembly/02_masked/at9847.scaffolds.unmasked.fasta...\n",
      "INFO     Extract sequences for cds...\n",
      "INFO     Print out extracted sequences: tmp_cds.fa...\n",
      "INFO     Making blast database... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/makeblastdb)\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/29/2022 17:46:23\n",
      "New DB name:   /home/luisa/work/detlef/difflines_annex/assembly-and-annotation/tmp_cds.fa\n",
      "New DB title:  tmp_cds.fa\n",
      "Sequence type: Nucleotide\n",
      "Keep Linkouts: T\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.000194073 seconds.\n",
      "INFO     Aligning sequences... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/blastn)\n",
      "INFO     Finding mRNAs with wrongly split gene parents... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/check_gene_parent/find_wrongly_split_gene_parent.pl)\n",
      "Reading the gff file: tmp/02_annotation/02_manual-curation-updated/at9847.everything.pansn.gff3...\n",
      "Reading the blast file: blastn.out...\n",
      "INFO     Removing unnecessary files...\n",
      "INFO     \t- Checking single-feature errors: (tmp/02_annotation/02_manual-curation-updated/at9847.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Print QC report at tmp/02_annotation/02_manual-curation-updated/at9847.everything.pansn.gff3.qcreport.txt\n",
      "INFO     Print QC statistic report at tmp/02_annotation/02_manual-curation-updated/at9847.everything.pansn.gff3.statistic.txt\n",
      "INFO     Checking QC report file (tmp/02_annotation/02_manual-curation-updated/at9847.everything.pansn.gff3.qcreport.txt)...\n",
      "INFO     Checking GFF3 file (tmp/02_annotation/02_manual-curation-updated/at9847.everything.pansn.gff3)...\n",
      "INFO     Reading QC report file: (tmp/02_annotation/02_manual-curation-updated/at9847.everything.pansn.gff3.qcreport.txt)...\n",
      "\n",
      "INFO     Reading GFF3 file: (tmp/02_annotation/02_manual-curation-updated/at9847.everything.pansn.gff3)...\n",
      "\n",
      "warning: seqid \"at9847_1_chr4\" on line 2 in file \"output/02_annotation/02_manual-curation-updated/at9847/at9847.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9847_9_u875\" on line 11 in file \"output/02_annotation/02_manual-curation-updated/at9847/at9847.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9847_1_chr2\" on line 17 in file \"output/02_annotation/02_manual-curation-updated/at9847/at9847.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9847_1_chr1\" on line 253 in file \"output/02_annotation/02_manual-curation-updated/at9847/at9847.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9847_1_chr5\" on line 269 in file \"output/02_annotation/02_manual-curation-updated/at9847/at9847.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9847_1_chr3\" on line 323 in file \"output/02_annotation/02_manual-curation-updated/at9847/at9847.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "at9852 has fixes for this many genes: 2\n",
      "INFO     Checking gff file (tmp/02_annotation/02_manual-curation-updated/at9852.everything.pansn.gff3)...\n",
      "INFO     Checking genome fasta (output/01_assembly/02_masked/at9852.scaffolds.unmasked.fasta)...\n",
      "INFO     Reading gff files: (tmp/02_annotation/02_manual-curation-updated/at9852.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Checking errors in the gff files: (tmp/02_annotation/02_manual-curation-updated/at9852.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking missing attributes: (function4gff.FIX_MISSING_ATTR())...\n",
      "\n",
      "INFO     \t- Checking intra-model errors: (tmp/02_annotation/02_manual-curation-updated/at9852.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking inter-model errors: (tmp/02_annotation/02_manual-curation-updated/at9852.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Specifying prefix of output file name: (tmp)...\n",
      "INFO     Reading files: tmp/02_annotation/02_manual-curation-updated/at9852.everything.pansn.gff3, output/01_assembly/02_masked/at9852.scaffolds.unmasked.fasta...\n",
      "INFO     Extract sequences for cds...\n",
      "INFO     Print out extracted sequences: tmp_cds.fa...\n",
      "INFO     Making blast database... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/makeblastdb)\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/29/2022 17:55:11\n",
      "New DB name:   /home/luisa/work/detlef/difflines_annex/assembly-and-annotation/tmp_cds.fa\n",
      "New DB title:  tmp_cds.fa\n",
      "Sequence type: Nucleotide\n",
      "Keep Linkouts: T\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 2 sequences in 0.000406981 seconds.\n",
      "INFO     Aligning sequences... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/blastn)\n",
      "INFO     Finding mRNAs with wrongly split gene parents... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/check_gene_parent/find_wrongly_split_gene_parent.pl)\n",
      "Reading the gff file: tmp/02_annotation/02_manual-curation-updated/at9852.everything.pansn.gff3...\n",
      "Reading the blast file: blastn.out...\n",
      "INFO     Removing unnecessary files...\n",
      "INFO     \t- Checking single-feature errors: (tmp/02_annotation/02_manual-curation-updated/at9852.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Print QC report at tmp/02_annotation/02_manual-curation-updated/at9852.everything.pansn.gff3.qcreport.txt\n",
      "INFO     Print QC statistic report at tmp/02_annotation/02_manual-curation-updated/at9852.everything.pansn.gff3.statistic.txt\n",
      "INFO     Checking QC report file (tmp/02_annotation/02_manual-curation-updated/at9852.everything.pansn.gff3.qcreport.txt)...\n",
      "INFO     Checking GFF3 file (tmp/02_annotation/02_manual-curation-updated/at9852.everything.pansn.gff3)...\n",
      "INFO     Reading QC report file: (tmp/02_annotation/02_manual-curation-updated/at9852.everything.pansn.gff3.qcreport.txt)...\n",
      "\n",
      "INFO     Reading GFF3 file: (tmp/02_annotation/02_manual-curation-updated/at9852.everything.pansn.gff3)...\n",
      "\n",
      "warning: seqid \"at9852_1_chr2\" on line 2 in file \"output/02_annotation/02_manual-curation-updated/at9852/at9852.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9852_1_chr5\" on line 41 in file \"output/02_annotation/02_manual-curation-updated/at9852/at9852.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9852_1_chr3\" on line 85 in file \"output/02_annotation/02_manual-curation-updated/at9852/at9852.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9852_1_chr1\" on line 119 in file \"output/02_annotation/02_manual-curation-updated/at9852/at9852.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9852_1_chr4\" on line 495 in file \"output/02_annotation/02_manual-curation-updated/at9852/at9852.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "at9883 has fixes for this many genes: 5\n",
      "INFO     Checking gff file (tmp/02_annotation/02_manual-curation-updated/at9883.everything.pansn.gff3)...\n",
      "INFO     Checking genome fasta (output/01_assembly/02_masked/at9883.scaffolds.unmasked.fasta)...\n",
      "INFO     Reading gff files: (tmp/02_annotation/02_manual-curation-updated/at9883.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Checking errors in the gff files: (tmp/02_annotation/02_manual-curation-updated/at9883.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking missing attributes: (function4gff.FIX_MISSING_ATTR())...\n",
      "\n",
      "INFO     \t- Checking intra-model errors: (tmp/02_annotation/02_manual-curation-updated/at9883.everything.pansn.gff3)...\n",
      "\n",
      "INFO     \t- Checking inter-model errors: (tmp/02_annotation/02_manual-curation-updated/at9883.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Specifying prefix of output file name: (tmp)...\n",
      "INFO     Reading files: tmp/02_annotation/02_manual-curation-updated/at9883.everything.pansn.gff3, output/01_assembly/02_masked/at9883.scaffolds.unmasked.fasta...\n",
      "INFO     Extract sequences for cds...\n",
      "INFO     Print out extracted sequences: tmp_cds.fa...\n",
      "INFO     Making blast database... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/makeblastdb)\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/29/2022 18:03:38\n",
      "New DB name:   /home/luisa/work/detlef/difflines_annex/assembly-and-annotation/tmp_cds.fa\n",
      "New DB title:  tmp_cds.fa\n",
      "Sequence type: Nucleotide\n",
      "Keep Linkouts: T\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 5 sequences in 0.000602007 seconds.\n",
      "INFO     Aligning sequences... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/ncbi-blast+/bin/blastn)\n",
      "INFO     Finding mRNAs with wrongly split gene parents... (/home/luisa/mambaforge/envs/dl20-annomerge/lib/python3.10/site-packages/gff3tool/lib/check_gene_parent/find_wrongly_split_gene_parent.pl)\n",
      "Reading the gff file: tmp/02_annotation/02_manual-curation-updated/at9883.everything.pansn.gff3...\n",
      "Reading the blast file: blastn.out...\n",
      "INFO     Removing unnecessary files...\n",
      "INFO     \t- Checking single-feature errors: (tmp/02_annotation/02_manual-curation-updated/at9883.everything.pansn.gff3)...\n",
      "\n",
      "INFO     Print QC report at tmp/02_annotation/02_manual-curation-updated/at9883.everything.pansn.gff3.qcreport.txt\n",
      "INFO     Print QC statistic report at tmp/02_annotation/02_manual-curation-updated/at9883.everything.pansn.gff3.statistic.txt\n",
      "INFO     Checking QC report file (tmp/02_annotation/02_manual-curation-updated/at9883.everything.pansn.gff3.qcreport.txt)...\n",
      "INFO     Checking GFF3 file (tmp/02_annotation/02_manual-curation-updated/at9883.everything.pansn.gff3)...\n",
      "INFO     Reading QC report file: (tmp/02_annotation/02_manual-curation-updated/at9883.everything.pansn.gff3.qcreport.txt)...\n",
      "\n",
      "INFO     Reading GFF3 file: (tmp/02_annotation/02_manual-curation-updated/at9883.everything.pansn.gff3)...\n",
      "\n",
      "warning: seqid \"at9883_1_chr2\" on line 2 in file \"output/02_annotation/02_manual-curation-updated/at9883/at9883.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9883_1_chr4\" on line 8 in file \"output/02_annotation/02_manual-curation-updated/at9883/at9883.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9883_9_u26\" on line 34 in file \"output/02_annotation/02_manual-curation-updated/at9883/at9883.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9883_9_u121\" on line 48 in file \"output/02_annotation/02_manual-curation-updated/at9883/at9883.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9883_1_chr5\" on line 82 in file \"output/02_annotation/02_manual-curation-updated/at9883/at9883.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9883_1_chr3\" on line 128 in file \"output/02_annotation/02_manual-curation-updated/at9883/at9883.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n",
      "warning: seqid \"at9883_1_chr1\" on line 172 in file \"output/02_annotation/02_manual-curation-updated/at9883/at9883.manualcuration.gff3\" has not been previously introduced with a \"##sequence-region\" line, create such a line automatically\n"
     ]
    }
   ],
   "source": [
    "for accession in \"${dl20[@]}\"\n",
    "do \n",
    "    mkdir -p \"output/02_annotation/02_manual-curation-updated/${accession}\"\n",
    "    mkdir -p \"tmp/02_annotation/02_manual-curation-updated/\"\n",
    "\n",
    "    if [ ! -d \"input/luisa_manual_annotation_updates/2022-07-28/${accession}\" ]\n",
    "    then\n",
    "        cp \"output/02_annotation/01_original-annotation-merged/${accession}.augustus-v2.3.gff3\" \\\n",
    "            \"output/02_annotation/02_manual-curation-updated/${accession}/${accession}.manualcuration.gff3\"\n",
    "        continue\n",
    "    fi\n",
    "\n",
    "    zcat \"input/luisa_manual_annotation_updates/2022-07-29/${accession}\"*.gff3.gz \\\n",
    "        >\"tmp/02_annotation/02_manual-curation-updated/${accession}.everything.gff3\"\n",
    "\n",
    "    gffsed=( sed\n",
    "        -e \"s/\\bChr1_RagTag_RagTag\\b/${accession}_1_chr1/g\"\n",
    "        -e \"s/\\bChr2_RagTag_RagTag\\b/${accession}_1_chr2/g\"\n",
    "        -e \"s/\\bChr3_RagTag_RagTag\\b/${accession}_1_chr3/g\"\n",
    "        -e \"s/\\bChr4_RagTag_RagTag\\b/${accession}_1_chr4/g\"\n",
    "        -e \"s/\\bChr5_RagTag_RagTag\\b/${accession}_1_chr5/g\"\n",
    "        -e \"s/\\bptg0\\+\\([0-9]\\+\\)[^\\t0-9]*/${accession}_9_u\\1/g\"\n",
    "    )\n",
    "\n",
    "    \"${gffsed[@]}\" \\\n",
    "        < \"tmp/02_annotation/02_manual-curation-updated/${accession}.everything.gff3\" \\\n",
    "        > \"tmp/02_annotation/02_manual-curation-updated/${accession}.everything.pansn.gff3\"\n",
    "\n",
    "\n",
    "    echo -n \"$accession has fixes for this many genes: \"\n",
    "    awk '$3 == \"gene\"{print}' \\\n",
    "        \"tmp/02_annotation/02_manual-curation-updated/${accession}.everything.pansn.gff3\" \\\n",
    "        | wc -l\n",
    "\n",
    "    gff3_QC \\\n",
    "        -f \"output/01_assembly/02_masked/${accession}.scaffolds.unmasked.fasta\" \\\n",
    "        -g \"tmp/02_annotation/02_manual-curation-updated/${accession}.everything.pansn.gff3\" \\\n",
    "        -o \"tmp/02_annotation/02_manual-curation-updated/${accession}.everything.pansn.gff3.qcreport.txt\" \\\n",
    "        -s \"tmp/02_annotation/02_manual-curation-updated/${accession}.everything.pansn.gff3.statistic.txt\"\n",
    "\n",
    "\n",
    "    gff3_fix \\\n",
    "        -g \"tmp/02_annotation/02_manual-curation-updated/${accession}.everything.pansn.gff3\" \\\n",
    "        -og \"tmp/02_annotation/02_manual-curation-updated/${accession}.everything.pansn.fixed.gff3\" \\\n",
    "        -qc_r \"tmp/02_annotation/02_manual-curation-updated/${accession}.everything.pansn.gff3.qcreport.txt\"\n",
    "\n",
    "    gff3_QC \\\n",
    "        -f \"output/01_assembly/02_masked/${accession}.scaffolds.unmasked.fasta\" \\\n",
    "        -g \"output/02_annotation/01_original-annotation-merged/${accession}.augustus-v2.3.gff3\"  \\\n",
    "        -o \"tmp/02_annotation/02_manual-curation-updated/${accession}.augustus.gff3.qcreport.txt\" \\\n",
    "        -s \"tmp/02_annotation/02_manual-curation-updated/${accession}.augustus.gff3.statistic.txt\" \\\n",
    "        > \"tmp/02_annotation/02_manual-curation-updated/${accession}.augustus.gff3.check.log\" 2>&1\n",
    "\n",
    "    gff3_fix \\\n",
    "        -g \"output/02_annotation/01_original-annotation-merged/${accession}.augustus-v2.3.gff3\" \\\n",
    "        -og \"tmp/02_annotation/02_manual-curation-updated/${accession}.augustus.fixed.gff3\" \\\n",
    "        -qc_r \"tmp/02_annotation/02_manual-curation-updated/${accession}.augustus.gff3.qcreport.txt\" \\\n",
    "        > \"tmp/02_annotation/02_manual-curation-updated/${accession}.augustus.gff3.fix.log\" 2>&1\n",
    "\n",
    "    gff3_merge \\\n",
    "        -f \"output/01_assembly/02_masked/${accession}.scaffolds.unmasked.fasta\" \\\n",
    "        -g1 \"tmp/02_annotation/02_manual-curation-updated/${accession}.everything.pansn.fixed.gff3\" \\\n",
    "        -g2 \"output/02_annotation/01_original-annotation-merged/${accession}.augustus-v2.3.gff3\" \\\n",
    "        -og \"output/02_annotation/02_manual-curation-updated/${accession}/${accession}.manualcuration.gff3\" \\\n",
    "        -r \"output/02_annotation/02_manual-curation-updated/${accession}/${accession}.manualcuration.mergereport.txt\" \\\n",
    "        > \"tmp/02_annotation/02_manual-curation-updated/${accession}.augustus.gff3.merge.log\" 2>&1\n",
    "\n",
    "    gt gff3 -sort -tidy -retainids -force \\\n",
    "        -o \"output/02_annotation/02_manual-curation-updated/${accession}/${accession}.manualcuration.fixed.gff3\"  \\\n",
    "        \"output/02_annotation/02_manual-curation-updated/${accession}/${accession}.manualcuration.gff3\"\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a4aa61f3-c57d-4039-a76f-6c0bf1125b8a",
   "metadata": {},
   "outputs": [],
   "source": [
    "for sample in $(cat fixed_accession_names.txt)\n",
    "do\n",
    "    gff3_QC -f input/scaffolds/${sample}.scaffolds.bionano.final.fasta -g output/GFFs_with_only_the_fixed_annotations/${sample}/${sample}_combined.gff -o output/GFFs_with_only_the_fixed_annotations/${sample}/${sample}_report.txt\n",
    "    gff3_fix -qc_r output/GFFs_with_only_the_fixed_annotations/${sample}/${sample}_report.txt -g output/GFFs_with_only_the_fixed_annotations/${sample}/${sample}_combined.gff -og output/GFFs_with_only_the_fixed_annotations/${sample}/${sample}_combined_fixed.gff\n",
    "\n",
    "    gff3_QC -f input/scaffolds/${sample}.scaffolds.bionano.final.fasta -g input/augustus_gff/${sample}.softmasked.final.gff3 -o tmp/${sample}.softmasked.final.report\n",
    "    gff3_fix -qc_r tmp/${sample}.softmasked.final.report -g input/augustus_gff/${sample}.softmasked.final.gff3 -og tmp/${sample}.softmasked.final_fixed.gff3\n",
    "\n",
    "    gff3_merge -g1 output/GFFs_with_only_the_fixed_annotations/${sample}/${sample}_combined_fixed.gff -g2 tmp/${sample}.softmasked.final_fixed.gff3 -f input/scaffolds/${sample}.scaffolds.bionano.final.fasta -og output/fixed_annotations_merged_with_augustus_annotations/${sample}.softmasked.final_fixed_20220726.gff3 -r output/fixed_annotations_merged_with_augustus_annotations/${sample}_merge_report.txt\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "728592b8-d3b1-40b5-9f3d-9d7781649ad0",
   "metadata": {},
   "outputs": [],
   "source": [
    "head input/augustus_gff/${sample}.softmasked.final.gff3 -n 100"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6d9e74be-8e7c-4f29-bd18-eeba0f6cbc86",
   "metadata": {},
   "source": [
    "2. Add the gene models that were orginally left out of the NLR orthogroups but were identified during the manual annotation."
   ]
  }
 ],
 "metadata": {
  "jupytext": {
   "formats": "ipynb,auto:light"
  },
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
