{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "25d28501-3845-4547-a467-79d461c84c97",
   "metadata": {},
   "source": [
    "## 12. Notebook for preprocessing input into the annotation descision making pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "eaf2b102-d89a-43ef-b743-eab9058a0de5",
   "metadata": {},
   "outputs": [],
   "source": [
    "conda activate dl20-orthogrouping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "e0422f03-628a-4d89-98ee-f63c03b16d3d",
   "metadata": {},
   "outputs": [],
   "source": [
    "dl20=(\n",
    "at6137 at6923 at6929 at7143\n",
    "at8285 at9104 at9336 at9503\n",
    "at9578 at9744 at9762 at9806\n",
    "at9830 at9847 at9852 at9879\n",
    "at9883 at9900\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d0b353a7-795f-4575-8128-4b491788008b",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir -p output/04_analyses_with_fixed_at9879/prep_for_decision_making"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "54b32e02-e1ef-4c6e-a51a-4f7267ba3992",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/tmp/global2/lteasdale/difflines-Luisa/assembly-and-annotation\n",
      "(dl20-orthogrouping) \n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "60b5abbe-ef88-43f9-980e-3858e091b880",
   "metadata": {},
   "source": [
    "### First we need to run the augustus annotations through AGAT to clean the gff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "3b02b8b9-02f9-4421-99e2-8127637a8f5f",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 985491\n",
      "=> Number of comment lines: 576782\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 408709\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 4 => CDS stop_codon intron start_codon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "=> Number of line in file: 994813\n",
      "=> Number of comment lines: 582764\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 412049\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 4 => intron CDS start_codon stop_codon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "=> Number of line in file: 986528\n",
      "=> Number of comment lines: 577186\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 409342\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 4 => intron CDS start_codon stop_codon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "=> Number of line in file: 986622\n",
      "=> Number of comment lines: 578262\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 408360\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 4 => stop_codon intron start_codon CDS\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "=> Number of line in file: 977069\n",
      "=> Number of comment lines: 571289\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 405780\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 4 => CDS stop_codon start_codon intron\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "=> Number of line in file: 980450\n",
      "=> Number of comment lines: 573643\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 406807\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 4 => CDS start_codon stop_codon intron\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "=> Number of line in file: 978821\n",
      "=> Number of comment lines: 572971\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 405850\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 4 => start_codon intron stop_codon CDS\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "=> Number of line in file: 987599\n",
      "=> Number of comment lines: 578241\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 409358\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 4 => start_codon intron CDS stop_codon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "=> Number of line in file: 983201\n",
      "=> Number of comment lines: 575133\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 408068\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 4 => stop_codon start_codon intron CDS\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "=> Number of line in file: 992944\n",
      "=> Number of comment lines: 581671\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 411273\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 4 => intron stop_codon CDS start_codon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "=> Number of line in file: 988545\n",
      "=> Number of comment lines: 578609\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 409936\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 4 => stop_codon start_codon CDS intron\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "=> Number of line in file: 973083\n",
      "=> Number of comment lines: 568634\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 404449\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 4 => start_codon intron stop_codon CDS\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "=> Number of line in file: 978715\n",
      "=> Number of comment lines: 572201\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 406514\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 4 => stop_codon start_codon CDS intron\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "=> Number of line in file: 1000430\n",
      "=> Number of comment lines: 586650\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 413780\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 4 => intron CDS start_codon stop_codon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "=> Number of line in file: 973248\n",
      "=> Number of comment lines: 568455\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 404793\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 4 => CDS stop_codon start_codon intron\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "=> Number of line in file: 985065\n",
      "=> Number of comment lines: 576342\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 408723\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 4 => start_codon intron CDS stop_codon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "=> Number of line in file: 982532\n",
      "=> Number of comment lines: 575181\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 407351\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 4 => intron stop_codon CDS start_codon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "=> Number of line in file: 983075\n",
      "=> Number of comment lines: 575321\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 407754\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 4 => start_codon CDS stop_codon intron\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 43 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 43 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 43 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 43 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 43 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 42 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 43 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 43 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 43 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 43 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 42 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 42 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 42 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 42 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 43 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 42 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 43 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 44 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 4 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "------------------------------ done in 4 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "------------------------------ done in 4 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "------------------------------ done in 4 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "------------------------------ done in 4 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "------------------------------ done in 4 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "------------------------------ done in 4 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "162776 exons created that were missing\n",
      "2 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "162869 exons created that were missing\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "164446 exons created that were missing\n",
      "2 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "165139 exons created that were missing\n",
      "1 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "164062 exons created that were missing\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "163396 exons created that were missing\n",
      "2 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "163290 exons created that were missing\n",
      "2 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 14 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "163579 exons created that were missing\n",
      "1 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "163879 exons created that were missing\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "164458 exons created that were missing\n",
      "2 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "163986 exons created that were missing\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 14 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "164290 exons created that were missing\n",
      "1 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 14 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "165388 exons created that were missing\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 14 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "164718 exons created that were missing\n",
      "1 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 14 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "163153 exons created that were missing\n",
      "4 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 14 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "164363 exons created that were missing\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 14 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "163699 exons created that were missing\n",
      "1 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 14 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "166181 exons created that were missing\n",
      "8 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 14 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "2 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 4 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "1 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 4 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "1 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 4 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "2 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 4 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "2 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 4 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "2 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "1 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "1 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "4 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "1 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "1 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 6 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "3 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 31 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 73 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "No problem found\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 31 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 74 seconds\n",
      "No problem found\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 31 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 74 seconds\n",
      "No problem found\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "No problem found\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 32 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 75 seconds\n",
      "No problem found\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 32 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 75 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "No problem found\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 32 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 74 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "No problem found\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 32 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 75 seconds\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 33 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 75 seconds\n",
      "No problem found\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 33 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 76 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 33 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 76 seconds\n",
      "No problem found\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 33 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 76 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "No problem found\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 34 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 77 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "No problem found\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 34 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 77 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 34 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 78 seconds\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 36 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 78 seconds\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 36 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 78 seconds\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 36 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 78 seconds\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 35 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 78 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/01_assembly/01_pansn-named/at9806.augustus-v2.3.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9806.augustus.gff\n",
      "Job done in 92 seconds\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/01_assembly/01_pansn-named/at9900.augustus-v2.3.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9900.augustus.gff\n",
      "Job done in 94 seconds\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/01_assembly/01_pansn-named/at9578.augustus-v2.3.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9578.augustus.gff\n",
      "Job done in 94 seconds\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/01_assembly/01_pansn-named/at9852.augustus-v2.3.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9852.augustus.gff\n",
      "Job done in 95 seconds\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/01_assembly/01_pansn-named/at9744.augustus-v2.3.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9744.augustus.gff\n",
      "Job done in 96 seconds\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/01_assembly/01_pansn-named/at8285.augustus-v2.3.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at8285.augustus.gff\n",
      "Job done in 97 seconds\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/01_assembly/01_pansn-named/at9830.augustus-v2.3.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9830.augustus.gff\n",
      "Job done in 96 seconds\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/01_assembly/01_pansn-named/at9883.augustus-v2.3.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9883.augustus.gff\n",
      "Job done in 96 seconds\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/01_assembly/01_pansn-named/at9503.augustus-v2.3.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9503.augustus.gff\n",
      "Job done in 96 seconds\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/01_assembly/01_pansn-named/at6923.augustus-v2.3.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at6923.augustus.gff\n",
      "Job done in 97 seconds\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/01_assembly/01_pansn-named/at6137.augustus-v2.3.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at6137.augustus.gff\n",
      "Job done in 97 seconds\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/01_assembly/01_pansn-named/at9762.augustus-v2.3.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9762.augustus.gff\n",
      "Job done in 98 seconds\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/01_assembly/01_pansn-named/at9879.augustus-v2.3.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9879.augustus.gff\n",
      "Job done in 99 seconds\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/01_assembly/01_pansn-named/at9336.augustus-v2.3.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9336.augustus.gff\n",
      "Job done in 99 seconds\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/01_assembly/01_pansn-named/at6929.augustus-v2.3.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at6929.augustus.gff\n",
      "Job done in 100 seconds\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/01_assembly/01_pansn-named/at9104.augustus-v2.3.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9104.augustus.gff\n",
      "Job done in 100 seconds\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/01_assembly/01_pansn-named/at7143.augustus-v2.3.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at7143.augustus.gff\n",
      "Job done in 100 seconds\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/01_assembly/01_pansn-named/at9847.augustus-v2.3.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9847.augustus.gff\n",
      "Job done in 101 seconds\n",
      "(dl20-orthogrouping) \n"
     ]
    }
   ],
   "source": [
    "parallel -j20 \\\n",
    "    rm output/04_analyses_with_fixed_at9879/prep_for_decision_making/{}.augustus.gff \\\n",
    "    \\&\\& \\\n",
    "    agat_convert_sp_gxf2gxf.pl --gff output/01_assembly/01_pansn-named/{}.augustus-v2.3.gff3 \\\n",
    "        --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/{}.augustus.gff \\\n",
    "    ::: \"${dl20[@]}\"\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "06ceb01e-6c8d-4eae-a29f-c50b86788d88",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(dl20-orthogrouping) \n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 287635\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 287632\n",
      "=> Number of feature type (3rd column): 5\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 3 => stop_codon start_codon CDS\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 28 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 3 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "165388 exons created that were missing\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 3 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 6 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 27 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 55 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/03_new_augustus_liftoff_for_at9879/at9879.scaffolds-v2.3_liftoff_polished_fixed.gff3_polished.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9879.augustus.gff\n",
      "Job done in 72 seconds\n",
      "(dl20-orthogrouping) \n"
     ]
    }
   ],
   "source": [
    "rm output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9879.augustus.gff\n",
    "agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/03_new_augustus_liftoff_for_at9879/at9879.scaffolds-v2.3_liftoff_polished_fixed.gff3_polished.gff3 \\\n",
    "    --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9879.augustus.gff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "a21875ce-1835-43a5-9611-b0080174eced",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "##gff-version 3\n",
      "# Liftoff v1.6.3\n",
      "# /home/luisa/mambaforge/envs/dl20-orthogrouping/bin/liftoff -g /home/luisa/work/detlef/difflines_annex/assembly-and-annotation/output/01_assembly/01_pansn-named/at9879.augustus-v2.3.gff3 -o at9879.scaffolds-v2.3_liftoff_polished_fixed.gff3 -polish -chroms at9879_chroms.txt /home/luisa/work/detlef/difflines_annex/assembly-and-annotation/output/01_assembly/01_pansn-named/at9879.scaffolds-v2.3.fasta /home/luisa/work/detlef/difflines_annex/assembly-and-annotation/output/01_assembly/01_pansn-named/at9879_chr2_fixed/at9879.scaffolds-v2.3_chr2_fixed.fasta\n",
      "at9879_1_chr1\tLiftoff\tgene\t6504\t8357\t.\t+\t.\tID=g1;copy_num_ID=g1_0;coverage=1.0;extra_copy_number=0;sequence_ID=1.0;valid_ORFs=1\n",
      "at9879_1_chr1\tLiftoff\ttranscript\t6504\t8357\t.\t+\t.\tID=g1.t1;Parent=g1;extra_copy_number=0;matches_ref_protein=True;valid_ORF=True\n",
      "at9879_1_chr1\tLiftoff\texon\t6504\t6506\t.\t+\t.\tID=nbis-exon-1;Parent=g1.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t6741\t7021\t.\t+\t.\tID=nbis-exon-2;Parent=g1.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t7213\t7332\t.\t+\t.\tID=nbis-exon-3;Parent=g1.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t7433\t7822\t.\t+\t.\tID=nbis-exon-4;Parent=g1.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t7901\t8053\t.\t+\t.\tID=nbis-exon-5;Parent=g1.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t8166\t8357\t.\t+\t.\tID=nbis-exon-6;Parent=g1.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t6504\t6657\t.\t+\t.\tID=g1.t1.cds;Parent=g1.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t6741\t7021\t.\t+\t.\tID=g1.t1.cds;Parent=g1.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t7213\t7332\t.\t+\t.\tID=g1.t1.cds;Parent=g1.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t7433\t7822\t.\t+\t.\tID=g1.t1.cds;Parent=g1.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t7901\t8053\t.\t+\t.\tID=g1.t1.cds;Parent=g1.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t8166\t8357\t.\t+\t.\tID=g1.t1.cds;Parent=g1.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tstart_codon\t6504\t6506\t.\t+\t.\tID=start_codon_1;Parent=g1.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tstop_codon\t8355\t8357\t.\t+\t.\tID=stop_codon_1;Parent=g1.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tgene\t9151\t11421\t.\t-\t.\tID=g2;copy_num_ID=g2_0;coverage=1.0;extra_copy_number=0;sequence_ID=1.0;valid_ORFs=1\n",
      "at9879_1_chr1\tLiftoff\ttranscript\t9151\t11421\t.\t-\t.\tID=g2.t1;Parent=g2;extra_copy_number=0;matches_ref_protein=True;valid_ORF=True\n",
      "at9879_1_chr1\tLiftoff\texon\t9151\t9153\t.\t-\t.\tID=nbis-exon-57626;Parent=g2.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t9913\t9988\t.\t-\t.\tID=nbis-exon-57627;Parent=g2.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t10140\t10206\t.\t-\t.\tID=nbis-exon-57628;Parent=g2.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t10320\t10405\t.\t-\t.\tID=nbis-exon-57629;Parent=g2.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t10518\t10591\t.\t-\t.\tID=nbis-exon-57630;Parent=g2.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t10698\t10743\t.\t-\t.\tID=nbis-exon-57631;Parent=g2.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t10991\t11080\t.\t-\t.\tID=nbis-exon-57632;Parent=g2.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t11172\t11219\t.\t-\t.\tID=nbis-exon-57633;Parent=g2.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t11326\t11421\t.\t-\t.\tID=nbis-exon-57634;Parent=g2.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t9151\t9317\t.\t-\t.\tID=g2.t1.cds;Parent=g2.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t9913\t9988\t.\t-\t.\tID=g2.t1.cds;Parent=g2.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t10140\t10206\t.\t-\t.\tID=g2.t1.cds;Parent=g2.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t10320\t10405\t.\t-\t.\tID=g2.t1.cds;Parent=g2.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t10518\t10591\t.\t-\t.\tID=g2.t1.cds;Parent=g2.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t10698\t10743\t.\t-\t.\tID=g2.t1.cds;Parent=g2.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t10991\t11080\t.\t-\t.\tID=g2.t1.cds;Parent=g2.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t11172\t11219\t.\t-\t.\tID=g2.t1.cds;Parent=g2.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t11326\t11421\t.\t-\t.\tID=g2.t1.cds;Parent=g2.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tstart_codon\t11419\t11421\t.\t-\t.\tID=start_codon_2;Parent=g2.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tstop_codon\t9151\t9153\t.\t-\t.\tID=stop_codon_2;Parent=g2.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tgene\t14593\t15666\t.\t-\t.\tID=g3;copy_num_ID=g3_0;coverage=1.0;extra_copy_number=0;sequence_ID=1.0;valid_ORFs=1\n",
      "at9879_1_chr1\tLiftoff\ttranscript\t14593\t15666\t.\t-\t.\tID=g3.t1;Parent=g3;extra_copy_number=0;matches_ref_protein=True;valid_ORF=True\n",
      "at9879_1_chr1\tLiftoff\texon\t14593\t14595\t.\t-\t.\tID=nbis-exon-113333;Parent=g3.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t15664\t15666\t.\t-\t.\tID=nbis-exon-113334;Parent=g3.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t14593\t15666\t.\t-\t.\tID=g3.t1.cds;Parent=g3.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tstart_codon\t15664\t15666\t.\t-\t.\tID=start_codon_3;Parent=g3.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tstop_codon\t14593\t14595\t.\t-\t.\tID=stop_codon_3;Parent=g3.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tgene\t24506\t32053\t.\t+\t.\tID=g4;copy_num_ID=g4_0;coverage=1.0;extra_copy_number=0;sequence_ID=1.0;valid_ORFs=1\n",
      "at9879_1_chr1\tLiftoff\ttranscript\t24506\t32053\t.\t+\t.\tID=g4.t1;Parent=g4;extra_copy_number=0;matches_ref_protein=True;valid_ORF=True\n",
      "at9879_1_chr1\tLiftoff\texon\t24506\t24508\t.\t+\t.\tID=nbis-exon-132723;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t25519\t25632\t.\t+\t.\tID=nbis-exon-132724;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t25729\t25939\t.\t+\t.\tID=nbis-exon-132725;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t26018\t26412\t.\t+\t.\tID=nbis-exon-132726;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t26501\t26720\t.\t+\t.\tID=nbis-exon-132727;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t26802\t26974\t.\t+\t.\tID=nbis-exon-132728;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t27058\t27180\t.\t+\t.\tID=nbis-exon-132729;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t27268\t27428\t.\t+\t.\tID=nbis-exon-132730;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t27519\t27752\t.\t+\t.\tID=nbis-exon-132731;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t27838\t27988\t.\t+\t.\tID=nbis-exon-132732;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t28075\t28257\t.\t+\t.\tID=nbis-exon-132733;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t28348\t28509\t.\t+\t.\tID=nbis-exon-132734;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t28594\t28689\t.\t+\t.\tID=nbis-exon-132735;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t28779\t29407\t.\t+\t.\tID=nbis-exon-132736;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t29684\t29781\t.\t+\t.\tID=nbis-exon-132737;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t29866\t30056\t.\t+\t.\tID=nbis-exon-132738;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t30134\t31039\t.\t+\t.\tID=nbis-exon-132739;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t31121\t31285\t.\t+\t.\tID=nbis-exon-132740;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t31384\t31790\t.\t+\t.\tID=nbis-exon-132741;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t31876\t32053\t.\t+\t.\tID=nbis-exon-132742;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t24506\t25438\t.\t+\t.\tID=g4.t1.cds;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t25519\t25632\t.\t+\t.\tID=g4.t1.cds;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t25729\t25939\t.\t+\t.\tID=g4.t1.cds;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t26018\t26412\t.\t+\t.\tID=g4.t1.cds;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t26501\t26720\t.\t+\t.\tID=g4.t1.cds;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t26802\t26974\t.\t+\t.\tID=g4.t1.cds;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t27058\t27180\t.\t+\t.\tID=g4.t1.cds;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t27268\t27428\t.\t+\t.\tID=g4.t1.cds;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t27519\t27752\t.\t+\t.\tID=g4.t1.cds;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t27838\t27988\t.\t+\t.\tID=g4.t1.cds;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t28075\t28257\t.\t+\t.\tID=g4.t1.cds;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t28348\t28509\t.\t+\t.\tID=g4.t1.cds;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t28594\t28689\t.\t+\t.\tID=g4.t1.cds;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t28779\t29407\t.\t+\t.\tID=g4.t1.cds;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t29684\t29781\t.\t+\t.\tID=g4.t1.cds;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t29866\t30056\t.\t+\t.\tID=g4.t1.cds;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t30134\t31039\t.\t+\t.\tID=g4.t1.cds;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t31121\t31285\t.\t+\t.\tID=g4.t1.cds;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t31384\t31790\t.\t+\t.\tID=g4.t1.cds;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tCDS\t31876\t32053\t.\t+\t.\tID=g4.t1.cds;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tstart_codon\t24506\t24508\t.\t+\t.\tID=start_codon_4;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tstop_codon\t32051\t32053\t.\t+\t.\tID=stop_codon_4;Parent=g4.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\tgene\t32356\t34100\t.\t-\t.\tID=g5;copy_num_ID=g5_0;coverage=1.0;extra_copy_number=0;sequence_ID=1.0;valid_ORFs=0\n",
      "at9879_1_chr1\tLiftoff\ttranscript\t32356\t34100\t.\t-\t.\tID=g5.t1;Parent=g5;extra_copy_number=0;matches_ref_protein=True;missing_start_codon=True;valid_ORF=False\n",
      "at9879_1_chr1\tLiftoff\texon\t32356\t32358\t.\t-\t.\tID=nbis-exon-138222;Parent=g5.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t32495\t32576\t.\t-\t.\tID=nbis-exon-138223;Parent=g5.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t32667\t32787\t.\t-\t.\tID=nbis-exon-138224;Parent=g5.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t32906\t32971\t.\t-\t.\tID=nbis-exon-138225;Parent=g5.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t33061\t33168\t.\t-\t.\tID=nbis-exon-138226;Parent=g5.t1;extra_copy_number=0\n",
      "at9879_1_chr1\tLiftoff\texon\t33255\t33320\t.\t-\t.\tID=nbis-exon-138227;Parent=g5.t1;extra_copy_number=0\n",
      "(dl20-orthogrouping) \n"
     ]
    }
   ],
   "source": [
    "head output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9879.augustus.gff -n 100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a01169ed-f9d5-4753-948d-36d01154c068",
   "metadata": {},
   "outputs": [],
   "source": [
    "head "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f236ef45-6855-4954-8d75-4e3816f7e388",
   "metadata": {},
   "source": [
    "### Clean the polished liftoffs using agat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "e603bd7c-360b-41de-98da-b12c4bce24cc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(dl20-orthogrouping) \n"
     ]
    }
   ],
   "source": [
    "rm output/04_analyses_with_fixed_at9879/prep_for_decision_making/*.tama-v2.fix.gff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "6e9bc6bd-843f-46b1-9bd7-4a86921495f9",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 530547\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 530544\n",
      "=> Number of feature type (3rd column): 12\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 7 => mRNA snRNA snoRNA miRNA rRNA tRNA ncRNA\n",
      "\t* level3: 4 => three_prime_UTR exon CDS five_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 63 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "34764 cases fixed where L3 features have parent feature(s) missing\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "194837 exons created that were missing\n",
      "4 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 21 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "9 UTRs created that were missing\n",
      "66 UTRs locations modified that were wrong\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "Lets remove isoform AT2G24755.3\n",
      "Lets remove isoform AT4G02130.3\n",
      "Lets remove isoform AT5G36930.2\n",
      "Lets remove isoform AT2G07806.2\n",
      "Lets remove isoform AT1G51520.2\n",
      "Lets remove isoform AT1G03710.2\n",
      "Lets remove isoform AT1G63860.2\n",
      "Lets remove isoform AT5G22608.2\n",
      "Lets remove isoform AT1G06630.3\n",
      "Lets remove isoform AT2G41440.2\n",
      "Lets remove isoform AT1G31600.3\n",
      "Lets remove isoform AT2G27720.3\n",
      "Lets remove isoform AT2G07827.2\n",
      "13 identical isoforms removed\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 60 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 123 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/01_polished_araport_liftoff_TAIR10/at9830.scaffolds-v2.3_liftoff_polished.gff3_polished --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9830.liftoff.gff\n",
      "Job done in 162 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 531235\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 531232\n",
      "=> Number of feature type (3rd column): 12\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 7 => ncRNA miRNA snRNA mRNA tRNA rRNA snoRNA\n",
      "\t* level3: 4 => exon five_prime_UTR CDS three_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 64 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "34865 cases fixed where L3 features have parent feature(s) missing\n",
      "------------------------------ done in 6 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "195034 exons created that were missing\n",
      "8 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 21 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "5 UTRs created that were missing\n",
      "55 UTRs locations modified that were wrong\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "Lets remove isoform AT1G25210.2\n",
      "Lets remove isoform AT3G57470.3\n",
      "Lets remove isoform AT5G66530.2\n",
      "Lets remove isoform AT1G06630.3\n",
      "Lets remove isoform AT5G36930.2\n",
      "Lets remove isoform AT2G41440.2\n",
      "Lets remove isoform AT2G27720.3\n",
      "Lets remove isoform AT5G50570.2\n",
      "Lets remove isoform AT2G24755.3\n",
      "9 identical isoforms removed\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 59 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 123 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/01_polished_araport_liftoff_TAIR10/at6923.scaffolds-v2.3_liftoff_polished.gff3_polished --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at6923.liftoff.gff\n",
      "Job done in 162 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 529823\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 529820\n",
      "=> Number of feature type (3rd column): 12\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 7 => miRNA snoRNA ncRNA rRNA mRNA snRNA tRNA\n",
      "\t* level3: 4 => three_prime_UTR five_prime_UTR exon CDS\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 63 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "34752 cases fixed where L3 features have parent feature(s) missing\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "194540 exons created that were missing\n",
      "4 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 21 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "12 UTRs created that were missing\n",
      "81 UTRs locations modified that were wrong\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 12 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "Lets remove isoform AT2G07806.2\n",
      "Lets remove isoform AT1G06630.3\n",
      "Lets remove isoform AT4G02130.3\n",
      "Lets remove isoform AT2G27720.3\n",
      "Lets remove isoform AT3G66652.2\n",
      "Lets remove isoform AT4G12020.3\n",
      "Lets remove isoform AT4G12020.2\n",
      "Lets remove isoform AT4G16162.2\n",
      "Lets remove isoform AT3G54730.4\n",
      "Lets remove isoform AT3G54730.3\n",
      "Lets remove isoform AT4G07670.2\n",
      "Lets remove isoform AT5G36930.2\n",
      "Lets remove isoform AT1G10586.2\n",
      "Lets remove isoform AT2G27490.2\n",
      "Lets remove isoform AT2G24755.3\n",
      "15 identical isoforms removed\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 60 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 123 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/01_polished_araport_liftoff_TAIR10/at9879.scaffolds-v2.3_liftoff_polished.gff3_polished --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9879.liftoff.gff\n",
      "Job done in 162 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 530104\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 530101\n",
      "=> Number of feature type (3rd column): 12\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 7 => tRNA snRNA mRNA snoRNA rRNA ncRNA miRNA\n",
      "\t* level3: 4 => exon CDS five_prime_UTR three_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 63 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "34726 cases fixed where L3 features have parent feature(s) missing\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "194736 exons created that were missing\n",
      "3 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 21 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "8 UTRs created that were missing\n",
      "55 UTRs locations modified that were wrong\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 12 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "Lets remove isoform AT2G27720.3\n",
      "Lets remove isoform AT2G07806.2\n",
      "Lets remove isoform AT5G36930.2\n",
      "Lets remove isoform AT5G50570.2\n",
      "Lets remove isoform AT1G31600.3\n",
      "Lets remove isoform AT4G27050.4\n",
      "Lets remove isoform AT4G27050.3\n",
      "Lets remove isoform AT4G27050.2\n",
      "Lets remove isoform AT3G57470.3\n",
      "Lets remove isoform AT1G72840.2\n",
      "Lets remove isoform AT1G06630.3\n",
      "Lets remove isoform AT4G07670.2\n",
      "Lets remove isoform AT1G10586.2\n",
      "Lets remove isoform AT5G22608.2\n",
      "Lets remove isoform AT2G41440.2\n",
      "Lets remove isoform AT1G59171.2\n",
      "Lets remove isoform AT2G24755.3\n",
      "17 identical isoforms removed\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 59 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 122 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/01_polished_araport_liftoff_TAIR10/at6929.scaffolds-v2.3_liftoff_polished.gff3_polished --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at6929.liftoff.gff\n",
      "Job done in 162 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 531111\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 531108\n",
      "=> Number of feature type (3rd column): 12\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 7 => miRNA snRNA ncRNA tRNA rRNA snoRNA mRNA\n",
      "\t* level3: 4 => five_prime_UTR three_prime_UTR CDS exon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 64 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "34876 cases fixed where L3 features have parent feature(s) missing\n",
      "------------------------------ done in 6 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "194989 exons created that were missing\n",
      "4 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 20 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "7 UTRs created that were missing\n",
      "51 UTRs locations modified that were wrong\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 12 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "Lets remove isoform AT1G06630.3\n",
      "Lets remove isoform AT3G56210.4\n",
      "Lets remove isoform AT2G41440.2\n",
      "Lets remove isoform AT4G07670.2\n",
      "Lets remove isoform AT3G57470.3\n",
      "Lets remove isoform AT2G27490.2\n",
      "Lets remove isoform AT5G50570.2\n",
      "Lets remove isoform AT3G44670.2\n",
      "Lets remove isoform AT5G22608.2\n",
      "Lets remove isoform AT3G54730.4\n",
      "Lets remove isoform AT3G54730.3\n",
      "Lets remove isoform AT2G27720.3\n",
      "Lets remove isoform AT1G27040.2\n",
      "Lets remove isoform AT2G24755.3\n",
      "14 identical isoforms removed\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 59 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 123 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/01_polished_araport_liftoff_TAIR10/at9104.scaffolds-v2.3_liftoff_polished.gff3_polished --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9104.liftoff.gff\n",
      "Job done in 162 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 531362\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 531359\n",
      "=> Number of feature type (3rd column): 12\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 7 => rRNA snoRNA snRNA mRNA tRNA ncRNA miRNA\n",
      "\t* level3: 4 => three_prime_UTR CDS exon five_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 63 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "34880 cases fixed where L3 features have parent feature(s) missing\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "195091 exons created that were missing\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 21 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "9 UTRs created that were missing\n",
      "54 UTRs locations modified that were wrong\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "Lets remove isoform AT5G36930.2\n",
      "Lets remove isoform AT1G72840.2\n",
      "Lets remove isoform AT1G59171.2\n",
      "Lets remove isoform AT4G07670.2\n",
      "Lets remove isoform AT1G16260.2\n",
      "Lets remove isoform AT5G50570.2\n",
      "Lets remove isoform AT4G02130.3\n",
      "Lets remove isoform AT2G41440.2\n",
      "Lets remove isoform AT1G06630.3\n",
      "Lets remove isoform AT2G27720.3\n",
      "Lets remove isoform AT5G66530.2\n",
      "Lets remove isoform AT4G17590.2\n",
      "Lets remove isoform AT2G24755.3\n",
      "13 identical isoforms removed\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 60 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 123 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/01_polished_araport_liftoff_TAIR10/at9503.scaffolds-v2.3_liftoff_polished.gff3_polished --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9503.liftoff.gff\n",
      "Job done in 162 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 530703\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 530700\n",
      "=> Number of feature type (3rd column): 12\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 7 => snRNA tRNA ncRNA mRNA miRNA snoRNA rRNA\n",
      "\t* level3: 4 => exon five_prime_UTR three_prime_UTR CDS\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 64 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "34788 cases fixed where L3 features have parent feature(s) missing\n",
      "------------------------------ done in 6 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "194870 exons created that were missing\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 20 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "8 UTRs created that were missing\n",
      "53 UTRs locations modified that were wrong\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 12 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "Lets remove isoform AT2G24755.3\n",
      "Lets remove isoform AT4G23290.2\n",
      "Lets remove isoform AT4G07670.2\n",
      "Lets remove isoform AT4G16162.2\n",
      "Lets remove isoform AT5G50570.2\n",
      "Lets remove isoform AT1G31600.3\n",
      "Lets remove isoform AT2G27720.3\n",
      "Lets remove isoform AT3G57470.3\n",
      "Lets remove isoform AT3G57470.2\n",
      "Lets remove isoform AT5G36930.2\n",
      "Lets remove isoform AT1G25210.2\n",
      "Lets remove isoform AT2G07806.2\n",
      "Lets remove isoform AT5G41750.2\n",
      "Lets remove isoform AT5G65050.3\n",
      "Lets remove isoform AT4G02130.3\n",
      "Lets remove isoform AT1G72930.2\n",
      "Lets remove isoform AT1G06630.3\n",
      "17 identical isoforms removed\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 59 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 123 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/01_polished_araport_liftoff_TAIR10/at8285.scaffolds-v2.3_liftoff_polished.gff3_polished --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at8285.liftoff.gff\n",
      "Job done in 162 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 530950\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 530947\n",
      "=> Number of feature type (3rd column): 12\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 7 => rRNA miRNA snoRNA tRNA mRNA snRNA ncRNA\n",
      "\t* level3: 4 => exon three_prime_UTR five_prime_UTR CDS\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 64 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "34847 cases fixed where L3 features have parent feature(s) missing\n",
      "------------------------------ done in 6 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "194947 exons created that were missing\n",
      "5 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 21 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "7 UTRs created that were missing\n",
      "46 UTRs locations modified that were wrong\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "Lets remove isoform AT1G16260.2\n",
      "Lets remove isoform AT1G06630.3\n",
      "Lets remove isoform AT3G58290.2\n",
      "Lets remove isoform AT2G07806.2\n",
      "Lets remove isoform AT5G22608.2\n",
      "Lets remove isoform AT1G03710.2\n",
      "Lets remove isoform AT1G45015.2\n",
      "Lets remove isoform AT3G57470.3\n",
      "Lets remove isoform AT2G07827.2\n",
      "Lets remove isoform AT4G07670.2\n",
      "Lets remove isoform AT5G50570.2\n",
      "Lets remove isoform AT5G36930.2\n",
      "Lets remove isoform AT1G44940.2\n",
      "Lets remove isoform AT1G59171.2\n",
      "Lets remove isoform AT2G27720.3\n",
      "15 identical isoforms removed\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 59 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 123 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/01_polished_araport_liftoff_TAIR10/at9883.scaffolds-v2.3_liftoff_polished.gff3_polished --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9883.liftoff.gff\n",
      "Job done in 162 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 530473\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 530470\n",
      "=> Number of feature type (3rd column): 12\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 7 => ncRNA snoRNA tRNA miRNA rRNA mRNA snRNA\n",
      "\t* level3: 4 => five_prime_UTR CDS exon three_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 64 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "34719 cases fixed where L3 features have parent feature(s) missing\n",
      "------------------------------ done in 6 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "194853 exons created that were missing\n",
      "8 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 20 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "7 UTRs created that were missing\n",
      "57 UTRs locations modified that were wrong\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "Lets remove isoform AT2G39310.3\n",
      "Lets remove isoform AT2G07827.2\n",
      "Lets remove isoform AT2G07806.2\n",
      "Lets remove isoform AT1G25210.2\n",
      "Lets remove isoform AT2G25900.2\n",
      "Lets remove isoform AT4G17590.2\n",
      "Lets remove isoform AT5G22608.2\n",
      "Lets remove isoform AT1G63860.2\n",
      "Lets remove isoform AT1G06630.3\n",
      "Lets remove isoform AT4G07670.2\n",
      "Lets remove isoform AT1G16260.2\n",
      "Lets remove isoform AT1G31600.3\n",
      "Lets remove isoform AT2G27720.3\n",
      "Lets remove isoform AT3G54730.4\n",
      "Lets remove isoform AT3G54730.3\n",
      "Lets remove isoform AT1G22403.2\n",
      "Lets remove isoform AT2G24755.3\n",
      "17 identical isoforms removed\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 60 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 124 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/01_polished_araport_liftoff_TAIR10/at9852.scaffolds-v2.3_liftoff_polished.gff3_polished --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9852.liftoff.gff\n",
      "Job done in 163 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 531344\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 531341\n",
      "=> Number of feature type (3rd column): 12\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 7 => tRNA snoRNA ncRNA mRNA miRNA rRNA snRNA\n",
      "\t* level3: 4 => exon three_prime_UTR five_prime_UTR CDS\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 63 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "34837 cases fixed where L3 features have parent feature(s) missing\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "195127 exons created that were missing\n",
      "7 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 21 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "5 UTRs created that were missing\n",
      "52 UTRs locations modified that were wrong\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "Lets remove isoform AT2G24755.3\n",
      "Lets remove isoform AT3G57470.3\n",
      "Lets remove isoform AT1G06630.3\n",
      "Lets remove isoform AT1G10586.2\n",
      "Lets remove isoform AT2G27720.3\n",
      "Lets remove isoform AT1G16260.2\n",
      "Lets remove isoform AT4G07670.2\n",
      "Lets remove isoform AT5G36930.2\n",
      "Lets remove isoform AT2G41440.2\n",
      "Lets remove isoform AT1G03710.2\n",
      "Lets remove isoform AT5G22608.2\n",
      "Lets remove isoform AT2G06541.2\n",
      "12 identical isoforms removed\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 61 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 124 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/01_polished_araport_liftoff_TAIR10/at7143.scaffolds-v2.3_liftoff_polished.gff3_polished --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at7143.liftoff.gff\n",
      "Job done in 163 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 530602\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 530599\n",
      "=> Number of feature type (3rd column): 12\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 7 => snoRNA tRNA snRNA rRNA miRNA ncRNA mRNA\n",
      "\t* level3: 4 => three_prime_UTR CDS exon five_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 64 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "34747 cases fixed where L3 features have parent feature(s) missing\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "194898 exons created that were missing\n",
      "3 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 20 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "6 UTRs created that were missing\n",
      "50 UTRs locations modified that were wrong\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 12 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "Lets remove isoform AT4G16162.2\n",
      "Lets remove isoform AT2G25900.2\n",
      "Lets remove isoform AT1G06630.3\n",
      "Lets remove isoform AT5G36930.2\n",
      "Lets remove isoform AT2G27720.3\n",
      "Lets remove isoform AT2G07806.2\n",
      "Lets remove isoform AT1G10586.2\n",
      "Lets remove isoform AT4G07670.2\n",
      "Lets remove isoform AT1G22403.2\n",
      "9 identical isoforms removed\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 59 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 123 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/01_polished_araport_liftoff_TAIR10/at9847.scaffolds-v2.3_liftoff_polished.gff3_polished --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9847.liftoff.gff\n",
      "Job done in 163 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 531490\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 531487\n",
      "=> Number of feature type (3rd column): 12\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 7 => ncRNA snRNA tRNA snoRNA mRNA rRNA miRNA\n",
      "\t* level3: 4 => exon three_prime_UTR CDS five_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 64 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "34899 cases fixed where L3 features have parent feature(s) missing\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "195140 exons created that were missing\n",
      "6 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 21 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "10 UTRs created that were missing\n",
      "49 UTRs locations modified that were wrong\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 12 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "Lets remove isoform AT3G61010.2\n",
      "Lets remove isoform AT1G25210.2\n",
      "Lets remove isoform AT1G66920.2\n",
      "Lets remove isoform AT1G31600.3\n",
      "Lets remove isoform AT2G34840.2\n",
      "Lets remove isoform AT1G10586.2\n",
      "Lets remove isoform AT1G06630.3\n",
      "Lets remove isoform AT3G57470.3\n",
      "Lets remove isoform AT5G50570.2\n",
      "Lets remove isoform AT1G63860.2\n",
      "Lets remove isoform AT1G37150.4\n",
      "Lets remove isoform AT1G37150.3\n",
      "Lets remove isoform AT5G57565.5\n",
      "Lets remove isoform AT2G27720.3\n",
      "Lets remove isoform AT4G07670.2\n",
      "Lets remove isoform AT5G22608.2\n",
      "Lets remove isoform AT5G36930.2\n",
      "17 identical isoforms removed\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 60 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 124 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/01_polished_araport_liftoff_TAIR10/at9744.scaffolds-v2.3_liftoff_polished.gff3_polished --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9744.liftoff.gff\n",
      "Job done in 164 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 530660\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 530657\n",
      "=> Number of feature type (3rd column): 12\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 7 => mRNA rRNA snRNA snoRNA tRNA miRNA ncRNA\n",
      "\t* level3: 4 => three_prime_UTR CDS exon five_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 64 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "34762 cases fixed where L3 features have parent feature(s) missing\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "194890 exons created that were missing\n",
      "5 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 21 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "7 UTRs created that were missing\n",
      "54 UTRs locations modified that were wrong\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "Lets remove isoform AT2G41440.2\n",
      "Lets remove isoform AT1G06630.3\n",
      "Lets remove isoform AT2G27720.3\n",
      "Lets remove isoform AT1G44940.2\n",
      "Lets remove isoform AT1G63860.2\n",
      "Lets remove isoform AT4G16950.2\n",
      "Lets remove isoform AT5G36930.2\n",
      "Lets remove isoform AT2G25900.2\n",
      "Lets remove isoform AT5G50570.2\n",
      "9 identical isoforms removed\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 61 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 125 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/01_polished_araport_liftoff_TAIR10/at9900.scaffolds-v2.3_liftoff_polished.gff3_polished --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9900.liftoff.gff\n",
      "Job done in 164 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 530789\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 530786\n",
      "=> Number of feature type (3rd column): 12\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 7 => snRNA ncRNA mRNA snoRNA tRNA rRNA miRNA\n",
      "\t* level3: 4 => exon three_prime_UTR CDS five_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 64 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "34800 cases fixed where L3 features have parent feature(s) missing\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "194929 exons created that were missing\n",
      "5 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 21 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "7 UTRs created that were missing\n",
      "58 UTRs locations modified that were wrong\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "Lets remove isoform AT5G50570.2\n",
      "Lets remove isoform AT2G27490.2\n",
      "Lets remove isoform AT2G06541.2\n",
      "Lets remove isoform AT1G63860.2\n",
      "Lets remove isoform AT1G45015.2\n",
      "Lets remove isoform AT2G27720.3\n",
      "Lets remove isoform AT2G05185.2\n",
      "Lets remove isoform AT4G02130.3\n",
      "Lets remove isoform AT1G10586.2\n",
      "Lets remove isoform AT1G06630.3\n",
      "Lets remove isoform AT1G31600.3\n",
      "Lets remove isoform AT5G36930.2\n",
      "Lets remove isoform AT2G07806.2\n",
      "Lets remove isoform AT5G22608.2\n",
      "Lets remove isoform AT1G03710.2\n",
      "Lets remove isoform AT4G07670.2\n",
      "Lets remove isoform AT2G07827.2\n",
      "Lets remove isoform AT2G24755.3\n",
      "18 identical isoforms removed\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 61 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 125 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/01_polished_araport_liftoff_TAIR10/at9336.scaffolds-v2.3_liftoff_polished.gff3_polished --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9336.liftoff.gff\n",
      "Job done in 164 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 530566\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 530563\n",
      "=> Number of feature type (3rd column): 12\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 7 => miRNA tRNA snRNA rRNA snoRNA ncRNA mRNA\n",
      "\t* level3: 4 => exon CDS three_prime_UTR five_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 64 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "34746 cases fixed where L3 features have parent feature(s) missing\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "194886 exons created that were missing\n",
      "5 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 22 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "5 UTRs created that were missing\n",
      "48 UTRs locations modified that were wrong\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "Lets remove isoform AT2G07806.2\n",
      "Lets remove isoform AT5G22608.2\n",
      "Lets remove isoform AT4G07670.2\n",
      "Lets remove isoform AT2G07827.2\n",
      "Lets remove isoform AT1G06630.3\n",
      "Lets remove isoform AT5G36930.2\n",
      "Lets remove isoform AT5G41740.2\n",
      "Lets remove isoform AT5G22660.2\n",
      "Lets remove isoform AT2G41440.2\n",
      "Lets remove isoform AT2G27720.3\n",
      "10 identical isoforms removed\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 61 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 125 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/01_polished_araport_liftoff_TAIR10/at6137.scaffolds-v2.3_liftoff_polished.gff3_polished --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at6137.liftoff.gff\n",
      "Job done in 165 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 531567\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 531564\n",
      "=> Number of feature type (3rd column): 12\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 7 => tRNA mRNA rRNA miRNA ncRNA snRNA snoRNA\n",
      "\t* level3: 4 => exon CDS five_prime_UTR three_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 64 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "34872 cases fixed where L3 features have parent feature(s) missing\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "195177 exons created that were missing\n",
      "2 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 21 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "8 UTRs created that were missing\n",
      "51 UTRs locations modified that were wrong\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "Lets remove isoform AT4G02130.3\n",
      "Lets remove isoform AT1G31600.3\n",
      "Lets remove isoform AT3G57470.3\n",
      "Lets remove isoform AT4G07670.2\n",
      "Lets remove isoform AT2G27720.3\n",
      "Lets remove isoform AT1G10586.2\n",
      "Lets remove isoform AT5G22608.2\n",
      "Lets remove isoform AT4G03490.2\n",
      "Lets remove isoform AT4G16162.2\n",
      "Lets remove isoform AT2G41440.2\n",
      "Lets remove isoform AT1G06630.3\n",
      "Lets remove isoform AT1G45015.2\n",
      "Lets remove isoform AT4G39930.2\n",
      "13 identical isoforms removed\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 61 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 125 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/01_polished_araport_liftoff_TAIR10/at9806.scaffolds-v2.3_liftoff_polished.gff3_polished --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9806.liftoff.gff\n",
      "Job done in 165 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 530503\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 530500\n",
      "=> Number of feature type (3rd column): 12\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 7 => miRNA tRNA mRNA snRNA rRNA ncRNA snoRNA\n",
      "\t* level3: 4 => exon CDS five_prime_UTR three_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 64 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "34774 cases fixed where L3 features have parent feature(s) missing\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "194829 exons created that were missing\n",
      "10 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 21 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "6 UTRs created that were missing\n",
      "75 UTRs locations modified that were wrong\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "Lets remove isoform AT5G50570.2\n",
      "Lets remove isoform AT4G16162.2\n",
      "Lets remove isoform AT4G07670.2\n",
      "Lets remove isoform AT1G66920.2\n",
      "Lets remove isoform AT1G10586.2\n",
      "Lets remove isoform AT1G45015.2\n",
      "Lets remove isoform AT1G24485.2\n",
      "Lets remove isoform AT1G06630.3\n",
      "Lets remove isoform AT4G39380.2\n",
      "Lets remove isoform AT2G27720.3\n",
      "Lets remove isoform AT2G41440.2\n",
      "Lets remove isoform AT3G54730.4\n",
      "Lets remove isoform AT3G54730.3\n",
      "13 identical isoforms removed\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 61 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 125 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/01_polished_araport_liftoff_TAIR10/at9762.scaffolds-v2.3_liftoff_polished.gff3_polished --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9762.liftoff.gff\n",
      "Job done in 165 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 530703\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 530700\n",
      "=> Number of feature type (3rd column): 12\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 7 => rRNA tRNA ncRNA snoRNA mRNA miRNA snRNA\n",
      "\t* level3: 4 => five_prime_UTR exon CDS three_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 64 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "34809 cases fixed where L3 features have parent feature(s) missing\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "194847 exons created that were missing\n",
      "5 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 20 seconds ------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "10 UTRs created that were missing\n",
      "53 UTRs locations modified that were wrong\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "Lets remove isoform AT2G41440.2\n",
      "Lets remove isoform AT4G07670.2\n",
      "Lets remove isoform AT1G03710.2\n",
      "Lets remove isoform AT2G27720.3\n",
      "Lets remove isoform AT1G06630.3\n",
      "Lets remove isoform AT1G65483.2\n",
      "Lets remove isoform AT5G22608.2\n",
      "Lets remove isoform AT1G72930.2\n",
      "Lets remove isoform AT5G36930.2\n",
      "Lets remove isoform AT2G20170.3\n",
      "Lets remove isoform AT1G25210.2\n",
      "Lets remove isoform AT1G22403.2\n",
      "12 identical isoforms removed\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 61 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 125 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/01_polished_araport_liftoff_TAIR10/at9578.scaffolds-v2.3_liftoff_polished.gff3_polished --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9578.liftoff.gff\n",
      "Job done in 165 seconds\n",
      "(dl20-orthogrouping) \n",
      "(dl20-orthogrouping) \n",
      "(dl20-orthogrouping) \n",
      "(dl20-orthogrouping) \n",
      "(dl20-orthogrouping) \n",
      "(dl20-orthogrouping) \n",
      "(dl20-orthogrouping) \n",
      "(dl20-orthogrouping) \n",
      "(dl20-orthogrouping) \n"
     ]
    }
   ],
   "source": [
    "parallel -j20 \\\n",
    "    agat_convert_sp_gxf2gxf.pl \\\n",
    "    --gff output/04_analyses_with_fixed_at9879/01_polished_araport_liftoff_TAIR10/{}.scaffolds-v2.3_liftoff_polished.gff3_polished \\\n",
    "    --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/{}.liftoff.gff \\\n",
    "    ::: \"${dl20[@]}\"\n",
    "\n",
    "\n",
    "#for accession in \"${dl20[@]}\"\n",
    "#do\n",
    "#agat_convert_sp_gxf2gxf.pl \\\n",
    "#    --gff output/04_analyses_with_fixed_at9879/01_polished_araport/liftoff/${accession}.scaffolds-v2.3_liftoff_polished.gff3_polished \\\n",
    "#    --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/${accession}.liftoff.gff\n",
    "#done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "f30c10e9-f882-491f-b8df-e072fee8d65b",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using standard /home/luisa/mambaforge/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /home/luisa/mambaforge/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /home/luisa/mambaforge/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/home/luisa/mambaforge/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /home/luisa/mambaforge/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 645694\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 645691\n",
      "=> Number of feature type (3rd column): 16\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 11 => snoRNA snRNA miRNA_primary_transcript rRNA antisense_RNA lnc_RNA tRNA mRNA antisense_lncRNA transcript_region ncRNA\n",
      "\t* level3: 4 => five_prime_UTR three_prime_UTR exon CDS\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 64 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "INFO - Feature types not expected by the GFF3 specification:\n",
      "* lnc_rna\n",
      "The feature type (3rd column in GFF3) is constrained to be either a term from th\n",
      "e Sequence Ontology or an SO accession number. The latter alternative is disting\n",
      "uished using the syntax SO:000000. In either case, it must be sequence_feature (\n",
      "SO:0000110) or an is_a child of it.To follow rigorously the gff3 format, please \n",
      "visit this website:\n",
      "https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "71 cases fixed where L3 features have parent feature(s) missing\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "22 cases fixed where L2 features have parent features missing\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "6 cases removed where L1 features do not have children (while they are suposed to have children).\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "18 exons created that were missing\n",
      "205 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "9774 level2 locations modified\n",
      "------------------------------ done in 6 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "32 UTRs created that were missing\n",
      "127 UTRs locations modified that were wrong\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "We fixed 184 wrong level2 location cases\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "We fixed 3 wrong level1 location cases\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "Lets remove isoform AT2G00620.2\n",
      "Lets remove isoform AT1G09765.3\n",
      "Lets remove isoform AT1G09765.2\n",
      "Lets remove isoform AT1G09685.3\n",
      "Lets remove isoform AT1G08197.2\n",
      "Lets remove isoform AT2G00500.2\n",
      "Lets remove isoform AT1G25175.2\n",
      "Lets remove isoform AT4G09432.2\n",
      "Lets remove isoform AT2G24755.3\n",
      "Lets remove isoform AT5G42230.2\n",
      "Lets remove isoform AT3G54730.4\n",
      "Lets remove isoform AT1G20060.3\n",
      "Lets remove isoform AT5G09910.4\n",
      "Lets remove isoform AT2G27490.4\n",
      "Lets remove isoform AT2G27490.2\n",
      "Lets remove isoform AT3G60940.2\n",
      "Lets remove isoform AT4G11050.2\n",
      "Lets remove isoform AT4G05040.6\n",
      "Lets remove isoform AT1G53970.2\n",
      "Lets remove isoform AT1G59680.2\n",
      "Lets remove isoform AT5G65560.3\n",
      "Lets remove isoform AT4G12020.9\n",
      "Lets remove isoform AT4G12020.8\n",
      "Lets remove isoform AT4G12020.7\n",
      "Lets remove isoform AT4G12020.6\n",
      "Lets remove isoform AT4G12020.5\n",
      "Lets remove isoform AT4G12020.4\n",
      "Lets remove isoform AT4G12020.3\n",
      "Lets remove isoform AT4G12020.2\n",
      "Lets remove isoform AT3G18530.3\n",
      "Lets remove isoform AT1G32270.2\n",
      "Lets remove isoform AT5G03690.4\n",
      "Lets remove isoform AT2G22720.5\n",
      "Lets remove isoform AT5G50340.3\n",
      "Lets remove isoform AT4G20480.2\n",
      "Lets remove isoform AT3G11880.2\n",
      "Lets remove isoform AT5G36930.2\n",
      "Lets remove isoform AT1G35320.2\n",
      "Lets remove isoform AT3G22565.2\n",
      "Lets remove isoform AT1G05540.2\n",
      "Lets remove isoform AT2G18180.3\n",
      "Lets remove isoform AT1G56520.3\n",
      "Lets remove isoform AT3G24255.6\n",
      "Lets remove isoform AT4G12990.8\n",
      "Lets remove isoform AT4G12990.7\n",
      "Lets remove isoform AT4G12990.3\n",
      "Lets remove isoform AT3G60960.3\n",
      "Lets remove isoform AT3G60960.2\n",
      "Lets remove isoform AT4G31710.2\n",
      "Lets remove isoform AT4G16162.2\n",
      "Lets remove isoform AT1G51790.3\n",
      "Lets remove isoform AT4G07670.2\n",
      "52 identical isoforms removed\n",
      "------------------------------ done in 3 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 29 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 93 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /home/luisa/mambaforge/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/01_polished_araport/liftoff/at9879.scaffolds-v2.3_liftoff_polished.gff3_polished --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9879.liftoff.gff\n",
      "Job done in 117 seconds\n",
      "Using standard /home/luisa/mambaforge/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /home/luisa/mambaforge/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /home/luisa/mambaforge/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/home/luisa/mambaforge/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /home/luisa/mambaforge/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 647143\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 647140\n",
      "=> Number of feature type (3rd column): 16\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 11 => tRNA rRNA antisense_RNA lnc_RNA miRNA_primary_transcript ncRNA antisense_lncRNA transcript_region snRNA mRNA snoRNA\n",
      "\t* level3: 4 => three_prime_UTR CDS exon five_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 57 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "INFO - Feature types not expected by the GFF3 specification:\n",
      "* lnc_rna\n",
      "The feature type (3rd column in GFF3) is constrained to be either a term from th\n",
      "e Sequence Ontology or an SO accession number. The latter alternative is disting\n",
      "uished using the syntax SO:000000. In either case, it must be sequence_feature (\n",
      "SO:0000110) or an is_a child of it.To follow rigorously the gff3 format, please \n",
      "visit this website:\n",
      "https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "43 cases fixed where L3 features have parent feature(s) missing\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "6 cases fixed where L2 features have parent features missing\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "1 cases removed where L1 features do not have children (while they are suposed to have children).\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 6 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "11 exons created that were missing\n",
      "199 exons locations modified that were wrong\n",
      "No supernumerary exons removed\n",
      "9783 level2 locations modified\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "23 UTRs created that were missing\n",
      "87 UTRs locations modified that were wrong\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 4 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "We fixed 186 wrong level2 location cases\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "Lets remove isoform AT1G09685.3\n",
      "Lets remove isoform AT2G00620.2\n",
      "Lets remove isoform AT1G25175.2\n",
      "Lets remove isoform AT5G51630.6\n",
      "Lets remove isoform AT5G51630.5\n",
      "Lets remove isoform AT5G51630.4\n",
      "Lets remove isoform AT5G44980.2\n",
      "Lets remove isoform AT1G32270.2\n",
      "Lets remove isoform AT5G42230.2\n",
      "Lets remove isoform AT3G32930.2\n",
      "Lets remove isoform AT3G60940.2\n",
      "Lets remove isoform AT5G44870.2\n",
      "Lets remove isoform AT4G05450.2\n",
      "Lets remove isoform AT3G60960.3\n",
      "Lets remove isoform AT3G60960.2\n",
      "Lets remove isoform AT5G46500.3\n",
      "Lets remove isoform AT1G72840.3\n",
      "Lets remove isoform AT4G16950.6\n",
      "Lets remove isoform AT4G16950.5\n",
      "Lets remove isoform AT4G16950.3\n",
      "Lets remove isoform AT2G41440.4\n",
      "Lets remove isoform AT2G41440.3\n",
      "Lets remove isoform AT2G41440.2\n",
      "Lets remove isoform AT4G16920.9\n",
      "Lets remove isoform AT4G16920.8\n",
      "Lets remove isoform AT4G16920.7\n",
      "Lets remove isoform AT4G16920.6\n",
      "Lets remove isoform AT4G16920.5\n",
      "Lets remove isoform AT4G16920.3\n",
      "Lets remove isoform AT4G16920.2\n",
      "Lets remove isoform AT1G53970.2\n",
      "Lets remove isoform AT5G36930.2\n",
      "Lets remove isoform AT1G63860.5\n",
      "Lets remove isoform AT1G44940.2\n",
      "Lets remove isoform AT4G16990.8\n",
      "Lets remove isoform AT4G16990.7\n",
      "Lets remove isoform AT4G16990.6\n",
      "Lets remove isoform AT4G16990.5\n",
      "Lets remove isoform AT4G16990.3\n",
      "Lets remove isoform AT4G16990.2\n",
      "Lets remove isoform AT4G16990.17\n",
      "Lets remove isoform AT4G16990.16\n",
      "Lets remove isoform AT4G16990.13\n",
      "Lets remove isoform AT4G16990.12\n",
      "Lets remove isoform AT4G16990.11\n",
      "Lets remove isoform AT4G16990.10\n",
      "Lets remove isoform AT4G12990.8\n",
      "Lets remove isoform AT4G12990.7\n",
      "Lets remove isoform AT4G12990.3\n",
      "49 identical isoforms removed\n",
      "------------------------------ done in 3 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 29 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 86 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /home/luisa/mambaforge/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff output/04_analyses_with_fixed_at9879/01_polished_araport/liftoff/at9900.scaffolds-v2.3_liftoff_polished.gff3_polished --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9900.liftoff.gff\n",
      "Job done in 110 seconds\n"
     ]
    }
   ],
   "source": [
    "agat_convert_sp_gxf2gxf.pl \\\n",
    "    --gff output/04_analyses_with_fixed_at9879/01_polished_araport/liftoff/at9879.scaffolds-v2.3_liftoff_polished.gff3_polished \\\n",
    "    --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9879.liftoff.gff"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "328a171e-6285-4937-9250-259ec9272ec6",
   "metadata": {},
   "source": [
    "### Clean the pasa output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "id": "1b6c202e-d2b7-4296-b440-70005c8bf28f",
   "metadata": {},
   "outputs": [],
   "source": [
    "dl20=(\n",
    "at6923 at6929 at7143\n",
    "at8285 at9104 at9336 at9503\n",
    "at9578 at9744 at9762 at9806\n",
    "at9830 at9847 at9852 at9879\n",
    "at9883 at9900\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "b3cf54b3-2804-4c27-9563-125d5bb663a2",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
      "!input/Isoseq/Pasa_v2/at6137_mydb_pasa.gene_structures_post_PASA_updates.*.gff3!\n",
      "!            does not exist. Please verify the input file name/path            !\n",
      "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
      "\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 849209\n",
      "=> Number of comment lines: 89244\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 759965\n",
      "=> Number of feature lines with 1 fields (while 9 expected): 109232\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => mRNA\n",
      "\t* level3: 4 => five_prime_UTR CDS exon three_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 66 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "109 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 31 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 97 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/Pasa_v2/at9883_mydb_pasa.gene_structures_post_PASA_updates.210550.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9883.pasa.gff\n",
      "Job done in 119 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 853897\n",
      "=> Number of comment lines: 88774\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 765123\n",
      "=> Number of feature lines with 1 fields (while 9 expected): 107665\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => mRNA\n",
      "\t* level3: 4 => exon three_prime_UTR five_prime_UTR CDS\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 68 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "107 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 3 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 30 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 98 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/Pasa_v2/at9806_mydb_pasa.gene_structures_post_PASA_updates.172008.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9806.pasa.gff\n",
      "Job done in 120 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 862059\n",
      "=> Number of comment lines: 91002\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 771057\n",
      "=> Number of feature lines with 1 fields (while 9 expected): 110723\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => mRNA\n",
      "\t* level3: 4 => exon three_prime_UTR CDS five_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 67 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "88 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 3 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 32 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 99 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/Pasa_v2/at9879_mydb_pasa.gene_structures_post_PASA_updates.12844.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9879.pasa.gff\n",
      "Job done in 121 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 872290\n",
      "=> Number of comment lines: 91532\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 780758\n",
      "=> Number of feature lines with 1 fields (while 9 expected): 110320\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => mRNA\n",
      "\t* level3: 4 => CDS five_prime_UTR exon three_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 69 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "93 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 31 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 100 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/Pasa_v2/at8285_mydb_pasa.gene_structures_post_PASA_updates.222468.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at8285.pasa.gff\n",
      "Job done in 122 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 889010\n",
      "=> Number of comment lines: 93478\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 795532\n",
      "=> Number of feature lines with 1 fields (while 9 expected): 110947\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => mRNA\n",
      "\t* level3: 4 => three_prime_UTR CDS exon five_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 70 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 6 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "94 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 6 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 3 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 31 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 101 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/Pasa_v2/at9104_mydb_pasa.gene_structures_post_PASA_updates.10333.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9104.pasa.gff\n",
      "Job done in 124 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 868203\n",
      "=> Number of comment lines: 90546\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 777657\n",
      "=> Number of feature lines with 1 fields (while 9 expected): 109077\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => mRNA\n",
      "\t* level3: 4 => exon CDS three_prime_UTR five_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 68 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "59 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 33 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 101 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/Pasa_v2/at9830_mydb_pasa.gene_structures_post_PASA_updates.249136.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9830.pasa.gff\n",
      "Job done in 125 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 883701\n",
      "=> Number of comment lines: 90888\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 792813\n",
      "=> Number of feature lines with 1 fields (while 9 expected): 108666\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => mRNA\n",
      "\t* level3: 4 => exon CDS five_prime_UTR three_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 70 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 6 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "82 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 6 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 3 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 33 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 103 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/Pasa_v2/at9852_mydb_pasa.gene_structures_post_PASA_updates.95923.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9852.pasa.gff\n",
      "Job done in 126 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 899810\n",
      "=> Number of comment lines: 93508\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 806302\n",
      "=> Number of feature lines with 1 fields (while 9 expected): 111270\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => mRNA\n",
      "\t* level3: 4 => CDS exon three_prime_UTR five_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 71 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "104 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 6 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 4 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 34 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 105 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/Pasa_v2/at9762_mydb_pasa.gene_structures_post_PASA_updates.166120.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9762.pasa.gff\n",
      "Job done in 128 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 904551\n",
      "=> Number of comment lines: 93892\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 810659\n",
      "=> Number of feature lines with 1 fields (while 9 expected): 110758\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => mRNA\n",
      "\t* level3: 4 => five_prime_UTR CDS three_prime_UTR exon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 72 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "50 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 6 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 4 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 34 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 106 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/Pasa_v2/at6929_mydb_pasa.gene_structures_post_PASA_updates.104367.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at6929.pasa.gff\n",
      "Job done in 130 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 909750\n",
      "=> Number of comment lines: 94990\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 814760\n",
      "=> Number of feature lines with 1 fields (while 9 expected): 111121\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => mRNA\n",
      "\t* level3: 4 => three_prime_UTR exon CDS five_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 73 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "106 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 6 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 3 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 34 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 107 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/Pasa_v2/at9900_mydb_pasa.gene_structures_post_PASA_updates.35530.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9900.pasa.gff\n",
      "Job done in 131 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 929778\n",
      "=> Number of comment lines: 97458\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 832320\n",
      "=> Number of feature lines with 1 fields (while 9 expected): 114253\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => mRNA\n",
      "\t* level3: 4 => CDS exon five_prime_UTR three_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 74 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "92 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 4 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 35 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 109 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/Pasa_v2/at9336_mydb_pasa.gene_structures_post_PASA_updates.96750.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9336.pasa.gff\n",
      "Job done in 133 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 937052\n",
      "=> Number of comment lines: 97062\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 839990\n",
      "=> Number of feature lines with 1 fields (while 9 expected): 112587\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => mRNA\n",
      "\t* level3: 4 => CDS exon three_prime_UTR five_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 75 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "80 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 6 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 4 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 36 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 111 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/Pasa_v2/at9503_mydb_pasa.gene_structures_post_PASA_updates.162167.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9503.pasa.gff\n",
      "Job done in 136 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 952336\n",
      "=> Number of comment lines: 97108\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 855228\n",
      "=> Number of feature lines with 1 fields (while 9 expected): 113462\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => mRNA\n",
      "\t* level3: 4 => five_prime_UTR CDS exon three_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 78 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "94 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 6 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 37 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 115 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/Pasa_v2/at9744_mydb_pasa.gene_structures_post_PASA_updates.72435.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9744.pasa.gff\n",
      "Job done in 141 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 973107\n",
      "=> Number of comment lines: 100614\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 872493\n",
      "=> Number of feature lines with 1 fields (while 9 expected): 114523\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => mRNA\n",
      "\t* level3: 4 => exon five_prime_UTR CDS three_prime_UTR\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 78 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "88 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 6 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 37 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 115 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/Pasa_v2/at7143_mydb_pasa.gene_structures_post_PASA_updates.150634.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at7143.pasa.gff\n",
      "Job done in 141 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 983965\n",
      "=> Number of comment lines: 101182\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 882783\n",
      "=> Number of feature lines with 1 fields (while 9 expected): 114823\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => mRNA\n",
      "\t* level3: 4 => three_prime_UTR exon five_prime_UTR CDS\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 79 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "88 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 5 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 39 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 118 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/Pasa_v2/at9847_mydb_pasa.gene_structures_post_PASA_updates.24698.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9847.pasa.gff\n",
      "Job done in 144 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 1017192\n",
      "=> Number of comment lines: 104850\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 912342\n",
      "=> Number of feature lines with 1 fields (while 9 expected): 116555\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => mRNA\n",
      "\t* level3: 4 => five_prime_UTR CDS three_prime_UTR exon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 82 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "114 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 6 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 11 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 4 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 39 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 121 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/Pasa_v2/at9578_mydb_pasa.gene_structures_post_PASA_updates.80973.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9578.pasa.gff\n",
      "Job done in 148 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 1030797\n",
      "=> Number of comment lines: 104650\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 926147\n",
      "=> Number of feature lines with 1 fields (while 9 expected): 116703\n",
      "=> Number of feature type (3rd column): 6\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => mRNA\n",
      "\t* level3: 4 => exon three_prime_UTR five_prime_UTR CDS\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                              done in 83 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 9 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 8 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "95 UTRs created that were missing\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 7 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 10 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 4 seconds -------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 40 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 123 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/Pasa_v2/at6923_mydb_pasa.gene_structures_post_PASA_updates.76536.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at6923.pasa.gff\n",
      "Job done in 150 seconds\n",
      "(dl20-orthogrouping) \n",
      "(dl20-orthogrouping) \n",
      "(dl20-orthogrouping) \n",
      "(dl20-orthogrouping) \n",
      "(dl20-orthogrouping) \n",
      "(dl20-orthogrouping) \n",
      "(dl20-orthogrouping) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "parallel -j20 \\\n",
    "    agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/Pasa_v2/{}_mydb_pasa.gene_structures_post_PASA_updates.*.gff3 \\\n",
    "    --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/{}.pasa.gff \\\n",
    "    ::: \"${dl20[@]}\"\n",
    "\n",
    "#for accession in \"${dl20[@]}\"\n",
    "#do\n",
    "#agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/Pasa_v2/${accession}_mydb_pasa.gene_structures_post_PASA_updates.*.gff3 \\\n",
    "#    --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/${accession}.pasa.gff\n",
    "#done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "406816c3-84b8-467f-80b2-a52be36953f4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(dl20-orthogrouping) \n"
     ]
    }
   ],
   "source": [
    "for accession in \"${dl20[@]}\"\n",
    "do\n",
    "awk 'BEGIN{IFS=OFS=\"\\t\"}$0 !~ /^#/ && $2 != \"AUGUSTUS\"{$2 = \"PASA\"; print($1, $2, $3, $4, $5, $6, $7, $8, $9)}' \\\n",
    "    < output/04_analyses_with_fixed_at9879/prep_for_decision_making/${accession}.pasa.gff \\\n",
    "    > output/04_analyses_with_fixed_at9879/prep_for_decision_making/${accession}.pasa.fix.gff\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ae6c5cf1-cc57-4920-8c09-087bf4d05b8b",
   "metadata": {},
   "source": [
    "### Clean the tama isoforms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "8ee15862-2819-492c-82ec-f31bfa57343a",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 1332208\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 1332205\n",
      "=> Number of feature type (3rd column): 3\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 1 => exon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                             done in 140 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 15 seconds ------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 13 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 57 seconds ------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 87 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 227 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/tama_v2/at9883.collapsed.tama.v2.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9883.tama-v2.gff\n",
      "Job done in 272 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 1379477\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 1379474\n",
      "=> Number of feature type (3rd column): 3\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 1 => exon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                             done in 146 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 16 seconds ------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 14 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 62 seconds ------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 94 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 240 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/tama_v2/at9879.collapsed.tama.v2.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9879.tama-v2.gff\n",
      "Job done in 288 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 1402035\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 1402032\n",
      "=> Number of feature type (3rd column): 3\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 1 => exon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                             done in 150 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 17 seconds ------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 14 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 59 seconds ------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 91 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 241 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/tama_v2/at8285.collapsed.tama.v2.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at8285.tama-v2.gff\n",
      "Job done in 290 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 1421464\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 1421461\n",
      "=> Number of feature type (3rd column): 3\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 1 => exon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                             done in 153 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 17 seconds ------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 14 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 65 seconds ------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 97 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 250 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/tama_v2/at9806.collapsed.tama.v2.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9806.tama-v2.gff\n",
      "Job done in 299 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 1446124\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 1446121\n",
      "=> Number of feature type (3rd column): 3\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 1 => exon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                             done in 152 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 17 seconds ------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 14 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 66 seconds ------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                              done in 98 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 250 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/tama_v2/at9830.collapsed.tama.v2.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9830.tama-v2.gff\n",
      "Job done in 299 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 1534896\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 1534893\n",
      "=> Number of feature type (3rd column): 3\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 1 => exon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                             done in 166 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 18 seconds ------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 15 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 74 seconds ------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                             done in 109 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 275 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/tama_v2/at9852.collapsed.tama.v2.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9852.tama-v2.gff\n",
      "Job done in 328 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 1607026\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 1607023\n",
      "=> Number of feature type (3rd column): 3\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 1 => exon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                             done in 173 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 19 seconds ------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 16 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 88 seconds ------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                             done in 125 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 298 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/tama_v2/at9762.collapsed.tama.v2.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9762.tama-v2.gff\n",
      "Job done in 354 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 1572205\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 1572202\n",
      "=> Number of feature type (3rd column): 3\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 1 => exon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                             done in 167 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 18 seconds ------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 17 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 86 seconds ------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                             done in 123 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 290 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/tama_v2/at9104.collapsed.tama.v2.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9104.tama-v2.gff\n",
      "Job done in 350 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 1661728\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 1661725\n",
      "=> Number of feature type (3rd column): 3\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 1 => exon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                             done in 179 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 19 seconds ------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 17 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 86 seconds ------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                             done in 125 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 304 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/tama_v2/at6929.collapsed.tama.v2.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at6929.tama-v2.gff\n",
      "Job done in 362 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 1685671\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 1685668\n",
      "=> Number of feature type (3rd column): 3\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 1 => exon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                             done in 181 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 20 seconds ------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 17 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 96 seconds ------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                             done in 134 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 315 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/tama_v2/at9900.collapsed.tama.v2.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9900.tama-v2.gff\n",
      "Job done in 373 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 1778154\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 1778151\n",
      "=> Number of feature type (3rd column): 3\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 1 => exon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                             done in 190 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 21 seconds ------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 18 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "------------------------------ done in 95 seconds ------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                             done in 136 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 326 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/tama_v2/at9503.collapsed.tama.v2.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9503.tama-v2.gff\n",
      "Job done in 388 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 1775915\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 1775912\n",
      "=> Number of feature type (3rd column): 3\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 1 => exon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                             done in 189 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 21 seconds ------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 18 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "----------------------------- done in 103 seconds ------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                             done in 144 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 333 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/tama_v2/at9336.collapsed.tama.v2.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9336.tama-v2.gff\n",
      "Job done in 396 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 1939628\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 1939625\n",
      "=> Number of feature type (3rd column): 3\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 1 => exon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                             done in 210 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 23 seconds ------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 19 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "----------------------------- done in 116 seconds ------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                             done in 161 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 371 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/tama_v2/at7143.collapsed.tama.v2.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at7143.tama-v2.gff\n",
      "Job done in 438 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 1944317\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 1944314\n",
      "=> Number of feature type (3rd column): 3\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 1 => exon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                             done in 214 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 23 seconds ------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 19 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "----------------------------- done in 119 seconds ------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                             done in 162 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 376 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/tama_v2/at9744.collapsed.tama.v2.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9744.tama-v2.gff\n",
      "Job done in 442 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 2045442\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 2045439\n",
      "=> Number of feature type (3rd column): 3\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 1 => exon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                             done in 222 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 24 seconds ------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 21 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "----------------------------- done in 134 seconds ------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                             done in 181 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 403 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/tama_v2/at9847.collapsed.tama.v2.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9847.tama-v2.gff\n",
      "Job done in 475 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 2233745\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 2233742\n",
      "=> Number of feature type (3rd column): 3\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 1 => exon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                             done in 243 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 26 seconds ------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 23 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 2 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "----------------------------- done in 161 seconds ------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                             done in 213 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 456 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/tama_v2/at9578.collapsed.tama.v2.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9578.tama-v2.gff\n",
      "Job done in 534 seconds\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/config.yaml file\n",
      "********************************************************************************\n",
      "*                              - Start parsing -                               *\n",
      "********************************************************************************\n",
      "-------------------------- parse options and metadata --------------------------\n",
      "=> Accessing the feature_levels YAML file\n",
      "Using standard /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/feature_levels.yaml file\n",
      "=> Attribute used to group features when no Parent/ID relationship exists (i.e common tag):\n",
      "\t* locus_tag\n",
      "\t* gene_id\n",
      "=> merge_loci option deactivated\n",
      "=> Machine information:\n",
      "\tThis script is being run by perl v5.32.1\n",
      "\tBioperl location being used: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/Bio/\n",
      "\tOperating system being used: linux \n",
      "=> Accessing Ontology\n",
      "\tNo ontology accessible from the gff file header!\n",
      "\tWe use the SOFA ontology distributed with AGAT:\n",
      "\t\t/tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo\n",
      "\tRead ontology /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/lib/perl5/site_perl/auto/share/dist/AGAT/so.obo:\n",
      "\t\t4 root terms, and 2596 total terms, and 1516 leaf terms\n",
      "\tFiltering ontology:\n",
      "\t\tWe found 1861 terms that are sequence_feature or is_a child of it.\n",
      "--------------------------------- parsing file ---------------------------------\n",
      "=> Number of line in file: 2233930\n",
      "=> Number of comment lines: 3\n",
      "=> Fasta included: No\n",
      "=> Number of features lines: 2233927\n",
      "=> Number of feature type (3rd column): 3\n",
      "\t* Level1: 1 => gene\n",
      "\t* level2: 1 => transcript\n",
      "\t* level3: 1 => exon\n",
      "\t* unknown: 0 => \n",
      "=> Version of the Bioperl GFF parser selected by AGAT: 3\n",
      "********************************************************************************\n",
      "*                               - End parsing -                                *\n",
      "*                             done in 242 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "********************************************************************************\n",
      "*                               - Start checks -                               *\n",
      "********************************************************************************\n",
      "---------------------------- Check1: feature types -----------------------------\n",
      "----------------------------------- ontology -----------------------------------\n",
      "All feature types in agreement with the Ontology.\n",
      "------------------------------------- agat -------------------------------------\n",
      "AGAT can deal with all the encountered feature types (3rd column)\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------------ Check2: duplicates ------------------------------\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "-------------------------- Check3: sequential bucket ---------------------------\n",
      "Nothing to check as sequential bucket!\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check4: l2 linked to l3 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check5: l1 linked to l2 ----------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "--------------------------- Check6: remove orphan l1 ---------------------------\n",
      "We remove only those not supposed to be orphan\n",
      "None found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------- Check7: all level3 locations -------------------------\n",
      "------------------------------ done in 26 seconds ------------------------------\n",
      "\n",
      "------------------------------ Check8: check cds -------------------------------\n",
      "No problem found\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check9: check exons ------------------------------\n",
      "No exons created\n",
      "No exons locations modified\n",
      "No supernumerary exons removed\n",
      "No level2 locations modified\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "----------------------------- Check10: check utrs ------------------------------\n",
      "No UTRs created\n",
      "No UTRs locations modified\n",
      "No supernumerary UTRs removed\n",
      "------------------------------ done in 0 seconds -------------------------------\n",
      "\n",
      "------------------------ Check11: all level2 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 23 seconds ------------------------------\n",
      "\n",
      "------------------------ Check12: all level1 locations -------------------------\n",
      "No problem found\n",
      "------------------------------ done in 1 seconds -------------------------------\n",
      "\n",
      "---------------------- Check13: remove identical isoforms ----------------------\n",
      "None found\n",
      "----------------------------- done in 154 seconds ------------------------------\n",
      "********************************************************************************\n",
      "*                                - End checks -                                *\n",
      "*                             done in 205 seconds                              *\n",
      "********************************************************************************\n",
      "\n",
      "=> OmniscientI total time: 447 seconds\n",
      "GFF3 file parsed\n",
      "Formating output to GFF3\n",
      "usage: /tmp/global2/lteasdale/conda/envs/dl20-orthogrouping/bin/agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/tama_v2/at6923.collapsed.tama.v2.gff3 --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/at6923.tama-v2.gff\n",
      "Job done in 532 seconds\n",
      "(dl20-orthogrouping) \n",
      "(dl20-orthogrouping) \n",
      "(dl20-orthogrouping) \n",
      "(dl20-orthogrouping) \n",
      "(dl20-orthogrouping) \n",
      "(dl20-orthogrouping) \n",
      "(dl20-orthogrouping) \n"
     ]
    }
   ],
   "source": [
    "parallel -j20 \\\n",
    "    agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/tama_v2/{}.collapsed.tama.v2.gff3 \\\n",
    "    --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/{}.tama-v2.gff \\\n",
    "    ::: \"${dl20[@]}\"\n",
    "\n",
    "#for accession in \"${dl20[@]}\"\n",
    "#do\n",
    "#agat_convert_sp_gxf2gxf.pl --gff input/Isoseq/tama_v2/${accession}.collapsed.tama.v2.gff3 \\\n",
    "#    --output output/04_analyses_with_fixed_at9879/prep_for_decision_making/${accession}.tama-v2.gff\n",
    "#done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "15619c12-219a-47f9-abd9-3479fcc63bd1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(dl20-orthogrouping) \n"
     ]
    }
   ],
   "source": [
    "for accession in \"${dl20[@]}\"\n",
    "do\n",
    "awk 'BEGIN{IFS=OFS=\"\\t\"}$0 !~ /^#/{$2 = \"tama_v2\"; print($1, $2, $3, $4, $5, $6, $7, $8, $9)}' \\\n",
    "    < output/04_analyses_with_fixed_at9879/prep_for_decision_making/${accession}.tama-v2.gff \\\n",
    "    > output/04_analyses_with_fixed_at9879/prep_for_decision_making/${accession}.tama-v2.fix.gff\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "01942609-487b-4bb8-92df-d413ecdb1750",
   "metadata": {},
   "source": [
    "Next step is to merge all four gffs and run bedtools merge to define the \"loci\" -> the output will be a bedfile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "77210483-bf89-46b1-91bd-517795f4dc3f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Fixed empty attribute (;=;) 195045 times\n",
      "Fixed empty attribute (;=;) 194745 times\n",
      "Fixed empty attribute (;=;) 195136 times\n",
      "Fixed empty attribute (;=;) 194887 times\n",
      "Fixed empty attribute (;=;) 195001 times\n",
      "Fixed empty attribute (;=;) 194937 times\n",
      "Fixed empty attribute (;=;) 195109 times\n",
      "Fixed empty attribute (;=;) 194859 times\n",
      "Fixed empty attribute (;=;) 195151 times\n",
      "Fixed empty attribute (;=;) 194835 times\n",
      "Fixed empty attribute (;=;) 195189 times\n",
      "Fixed empty attribute (;=;) 194850 times\n",
      "Fixed empty attribute (;=;) 194910 times\n"
     ]
    }
   ],
   "source": [
    "for accession in \"${dl20[@]}\"\n",
    "do\n",
    "blsl  liftoff-gff3 -o output/04_analyses_with_fixed_at9879/prep_for_decision_making/${accession}.scaffolds-v2.3_liftoff_polished.gff3_polished.fix.gff3 \\\n",
    "       output/04_analyses_with_fixed_at9879/01_polished_araport_liftoff_TAIR10/${accession}.scaffolds-v2.3_liftoff_polished.gff3_polished\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2bfd59d7-8a27-4e2c-8f6b-b3075e10e269",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "feef54ff-2bc5-49d4-bbf5-0c571039081f",
   "metadata": {},
   "source": [
    "## Make NLR list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "65e08e15-7475-497f-bf33-aaa80d86fa42",
   "metadata": {},
   "outputs": [],
   "source": [
    "dl20=(\n",
    "#at6137\n",
    "at9879\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "7c343d55-75c4-4d1b-a8d5-d4a5b4819171",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: If your Wiggle data is a significant portion of available system memory, use the --max-mem and --sort-tmpdir options, or use --do-not-sort to disable post-conversion sorting. See --help for more information.\n"
     ]
    }
   ],
   "source": [
    "for accession in \"${dl20[@]}\"\n",
    "do\n",
    "    git annex unlock output/04_analyses_with_fixed_at9879/NLRs_list/${accession}.nlrs.genes.augustus.bed\n",
    "    \n",
    "\n",
    "    gff2bed < output/04_analyses_with_fixed_at9879/prep_for_decision_making/${accession}.augustus.gff | \\\n",
    "    grep \"gene\" | \\\n",
    "    grep  -f <(awk '{printf(\"ID=%s[\\\\t;]\\n\", $1)}' < output/04_analyses_with_fixed_at9879/NLRs_list/${accession}_nlrs.txt ) \\\n",
    "    > output/04_analyses_with_fixed_at9879/NLRs_list/${accession}.nlrs.genes.augustus.bed  \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "857f27ee-238c-4fcd-a683-dbb6636fd5ad",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "##gff-version 3\n",
      "# This output was generated with AUGUSTUS (version 3.3.3).\n",
      "# AUGUSTUS is a gene prediction tool written by M. Stanke (mario.stanke@uni-greifswald.de),\n",
      "# O. Keller, S. König, L. Gerischer, L. Romoth and Katharina Hoff.\n",
      "# Please cite: Mario Stanke, Mark Diekhans, Robert Baertsch, David Haussler (2008),\n",
      "# Using native and syntenically mapped cDNA alignments to improve de novo gene finding\n",
      "# Bioinformatics 24: 637-644, doi 10.1093/bioinformatics/btn013\n",
      "# Sources of extrinsic information: M RM T \n",
      "# Setting acceptor splice site local malus: 0.1\n",
      "# Setting donor splice site local malus: 0.1\n",
      "# Setting exon local malus: 0.985\n",
      "# Setting CDSpart local malus: 0.985\n",
      "# Setting UTRpart local malus: 0.985\n",
      "# reading in the file at9900.liftoff.hints ...\n",
      "# Have extrinsic information about 8 sequences (in the specified range). \n",
      "# Initializing the parameters using config directory /ebio/abt6/mcollenberg/miniconda3/envs/augustus3.3.3/config/ ...\n",
      "# BUSCO_at9900_buscoResults version. Using default transition matrix.\n",
      "# admissible start codons and their probabilities: ATG(0.999), CTG(0.000382), TTG(0.000765)\n",
      "# Looks like at9900.softmasked.final.fasta is in fasta format.\n",
      "# We have hints for 8 sequences and for 8 of the sequences in the input set.\n",
      "#\n",
      "# ----- prediction on sequence number 1 (length = 33376575, name = at9900_1_chr1) -----\n",
      "#\n",
      "# Predicted genes for sequence number 1 on both strands\n",
      "# start gene g1\n",
      "at9900_1_chr1\tAUGUSTUS\tgene\t7504\t9354\t0.92\t+\t.\tID=g1\n",
      "at9900_1_chr1\tAUGUSTUS\ttranscript\t7504\t9354\t0.92\t+\t.\tID=g1.t1;Parent=g1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t7504\t7506\t0.92\t+\t.\tID=nbis-exon-1;Parent=g1.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t7740\t8020\t0.92\t+\t.\tID=nbis-exon-2;Parent=g1.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t8210\t8329\t0.92\t+\t.\tID=nbis-exon-3;Parent=g1.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t8430\t8819\t0.92\t+\t.\tID=nbis-exon-4;Parent=g1.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t8898\t9050\t0.92\t+\t.\tID=nbis-exon-5;Parent=g1.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t9163\t9354\t0.92\t+\t.\tID=nbis-exon-6;Parent=g1.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tCDS\t7504\t7657\t0.92\t+\t0\tID=g1.t1.cds;Parent=g1.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tCDS\t7740\t8020\t1\t+\t2\tID=g1.t1.cds;Parent=g1.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tCDS\t8210\t8329\t1\t+\t0\tID=g1.t1.cds;Parent=g1.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tCDS\t8430\t8819\t1\t+\t0\tID=g1.t1.cds;Parent=g1.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tCDS\t8898\t9050\t1\t+\t0\tID=g1.t1.cds;Parent=g1.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tCDS\t9163\t9354\t1\t+\t0\tID=g1.t1.cds;Parent=g1.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tintron\t7658\t7739\t1\t+\t.\tID=intron-1;Parent=g1.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tintron\t8021\t8209\t1\t+\t.\tID=intron-2;Parent=g1.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tintron\t8330\t8429\t1\t+\t.\tID=intron-3;Parent=g1.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tintron\t8820\t8897\t1\t+\t.\tID=intron-4;Parent=g1.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tintron\t9051\t9162\t1\t+\t.\tID=intron-5;Parent=g1.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tstart_codon\t7504\t7506\t.\t+\t0\tID=start_codon-1;Parent=g1.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tstop_codon\t9352\t9354\t.\t+\t0\tID=stop_codon-1;Parent=g1.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tgene\t10153\t12423\t0.31\t-\t.\tID=g2\n",
      "at9900_1_chr1\tAUGUSTUS\ttranscript\t10153\t12423\t0.31\t-\t.\tID=g2.t1;Parent=g2\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t10153\t10155\t0.31\t-\t.\tID=nbis-exon-57945;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t10915\t10990\t0.31\t-\t.\tID=nbis-exon-57946;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t11142\t11208\t0.31\t-\t.\tID=nbis-exon-57947;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t11322\t11407\t0.31\t-\t.\tID=nbis-exon-57948;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t11520\t11593\t0.31\t-\t.\tID=nbis-exon-57949;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t11700\t11745\t0.31\t-\t.\tID=nbis-exon-57950;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t11993\t12082\t0.31\t-\t.\tID=nbis-exon-57951;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t12174\t12221\t0.31\t-\t.\tID=nbis-exon-57952;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t12328\t12423\t0.31\t-\t.\tID=nbis-exon-57953;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tCDS\t10153\t10319\t0.31\t-\t2\tID=g2.t1.cds;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tCDS\t10915\t10990\t1\t-\t0\tID=g2.t1.cds;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tCDS\t11142\t11208\t1\t-\t1\tID=g2.t1.cds;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tCDS\t11322\t11407\t1\t-\t0\tID=g2.t1.cds;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tCDS\t11520\t11593\t1\t-\t2\tID=g2.t1.cds;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tCDS\t11700\t11745\t1\t-\t0\tID=g2.t1.cds;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tCDS\t11993\t12082\t1\t-\t0\tID=g2.t1.cds;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tCDS\t12174\t12221\t1\t-\t0\tID=g2.t1.cds;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tCDS\t12328\t12423\t1\t-\t0\tID=g2.t1.cds;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tintron\t10320\t10914\t0.32\t-\t.\tID=intron-6;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tintron\t10991\t11141\t1\t-\t.\tID=intron-7;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tintron\t11209\t11321\t1\t-\t.\tID=intron-8;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tintron\t11408\t11519\t1\t-\t.\tID=intron-9;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tintron\t11594\t11699\t1\t-\t.\tID=intron-10;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tintron\t11746\t11992\t1\t-\t.\tID=intron-11;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tintron\t12083\t12173\t1\t-\t.\tID=intron-12;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tintron\t12222\t12327\t1\t-\t.\tID=intron-13;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tstart_codon\t12421\t12423\t.\t-\t0\tID=start_codon-2;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tstop_codon\t10153\t10155\t.\t-\t0\tID=stop_codon-2;Parent=g2.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tgene\t15613\t16689\t0.97\t-\t.\tID=g3\n",
      "at9900_1_chr1\tAUGUSTUS\ttranscript\t15613\t16689\t0.97\t-\t.\tID=g3.t1;Parent=g3\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t15613\t15615\t0.97\t-\t.\tID=nbis-exon-115041;Parent=g3.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t16687\t16689\t0.97\t-\t.\tID=nbis-exon-115042;Parent=g3.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tCDS\t15613\t16689\t0.97\t-\t0\tID=g3.t1.cds;Parent=g3.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tstart_codon\t16687\t16689\t.\t-\t0\tID=start_codon-3;Parent=g3.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tstop_codon\t15613\t15615\t.\t-\t0\tID=stop_codon-3;Parent=g3.t1\n",
      "at9900_1_chr1\tAUGUSTUS\tgene\t25585\t33147\t0.85\t+\t.\tID=g4\n",
      "at9900_1_chr1\tAUGUSTUS\ttranscript\t25585\t33147\t0.85\t+\t.\tID=g4.t1;Parent=g4\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t25585\t25587\t0.85\t+\t.\tID=nbis-exon-129969;Parent=g4.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t26608\t26721\t0.85\t+\t.\tID=nbis-exon-129970;Parent=g4.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t26818\t27028\t0.85\t+\t.\tID=nbis-exon-129971;Parent=g4.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t27107\t27501\t0.85\t+\t.\tID=nbis-exon-129972;Parent=g4.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t27590\t27809\t0.85\t+\t.\tID=nbis-exon-129973;Parent=g4.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t27891\t28063\t0.85\t+\t.\tID=nbis-exon-129974;Parent=g4.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t28147\t28269\t0.85\t+\t.\tID=nbis-exon-129975;Parent=g4.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t28358\t28518\t0.85\t+\t.\tID=nbis-exon-129976;Parent=g4.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t28609\t28842\t0.85\t+\t.\tID=nbis-exon-129977;Parent=g4.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t28928\t29078\t0.85\t+\t.\tID=nbis-exon-129978;Parent=g4.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t29165\t29347\t0.85\t+\t.\tID=nbis-exon-129979;Parent=g4.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t29438\t29599\t0.85\t+\t.\tID=nbis-exon-129980;Parent=g4.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t29684\t29779\t0.85\t+\t.\tID=nbis-exon-129981;Parent=g4.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t29869\t30497\t0.85\t+\t.\tID=nbis-exon-129982;Parent=g4.t1\n",
      "at9900_1_chr1\tAUGUSTUS\texon\t30774\t30871\t0.85\t+\t.\tID=nbis-exon-129983;Parent=g4.t1\n"
     ]
    }
   ],
   "source": [
    "head -n 100 output/04_analyses_with_fixed_at9879/prep_for_decision_making/at9900.augustus.gff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "e92ed357-f782-4d57-b155-fc3926654d1f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "at9900_1_chr1\t7374\t9623\t\t\t+\tat9900_1_chr1\t7503\t7506\tnbis-exon-1\t0.92\t+\tAUGUSTUS\texon\t.\tID=nbis-exon-1;Parent=g1.t1\n",
      "at9900_1_chr1\t7374\t9623\t\t\t+\tat9900_1_chr1\t7503\t7506\tstart_codon-1\t.\t+\tAUGUSTUS\tstart_codon\t0\tID=start_codon-1;Parent=g1.t1\n",
      "at9900_1_chr1\t7374\t9623\t\t\t+\tat9900_1_chr1\t7503\t7657\tg1.t1.cds\t0.92\t+\tAUGUSTUS\tCDS\t0\tID=g1.t1.cds;Parent=g1.t1\n",
      "at9900_1_chr1\t7374\t9623\t\t\t+\tat9900_1_chr1\t7503\t9354\tg1\t0.92\t+\tAUGUSTUS\tgene\t.\tID=g1\n",
      "at9900_1_chr1\t7374\t9623\t\t\t+\tat9900_1_chr1\t7503\t9354\tg1.t1\t0.92\t+\tAUGUSTUS\ttranscript\t.\tID=g1.t1;Parent=g1\n",
      "at9900_1_chr1\t7374\t9623\t\t\t+\tat9900_1_chr1\t7657\t7739\tintron-1\t1\t+\tAUGUSTUS\tintron\t.\tID=intron-1;Parent=g1.t1\n",
      "at9900_1_chr1\t7374\t9623\t\t\t+\tat9900_1_chr1\t7739\t8020\tg1.t1.cds\t1\t+\tAUGUSTUS\tCDS\t2\tID=g1.t1.cds;Parent=g1.t1\n",
      "at9900_1_chr1\t7374\t9623\t\t\t+\tat9900_1_chr1\t7739\t8020\tnbis-exon-2\t0.92\t+\tAUGUSTUS\texon\t.\tID=nbis-exon-2;Parent=g1.t1\n",
      "at9900_1_chr1\t7374\t9623\t\t\t+\tat9900_1_chr1\t8020\t8209\tintron-2\t1\t+\tAUGUSTUS\tintron\t.\tID=intron-2;Parent=g1.t1\n",
      "at9900_1_chr1\t7374\t9623\t\t\t+\tat9900_1_chr1\t8209\t8329\tg1.t1.cds\t1\t+\tAUGUSTUS\tCDS\t0\tID=g1.t1.cds;Parent=g1.t1\n"
     ]
    }
   ],
   "source": [
    "head tmp/nlr_to_loci/loci_to_augustus_mapping_v2.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "4f12ca9e-5e3f-46ea-a063-0d38a06a7863",
   "metadata": {},
   "outputs": [],
   "source": [
    "grep \"gene\" tmp/nlr_to_loci/loci_to_augustus_mapping_v2.txt > tmp/nlr_to_loci/loci_to_augustus_mapping3.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "a835477d-a0fe-48a9-b0bf-2be667da80db",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "at9900_1_chr1\t7374\t9623\t\t\t+\tat9900_1_chr1\t7503\t9354\tg1\t0.92\t+\tAUGUSTUS\tgene\t.\tID=g1\n",
      "at9900_1_chr1\t10152\t12883\t\t\t-\tat9900_1_chr1\t10152\t12423\tg2\t0.31\t-\tAUGUSTUS\tgene\t.\tID=g2\n",
      "at9900_1_chr1\t13478\t17468\t\t\t-\tat9900_1_chr1\t15612\t16689\tg3\t0.97\t-\tAUGUSTUS\tgene\t.\tID=g3\n",
      "at9900_1_chr1\t25172\t33295\t\t\t+\tat9900_1_chr1\t25584\t33147\tg4\t0.85\t+\tAUGUSTUS\tgene\t.\tID=g4\n",
      "at9900_1_chr1\t33211\t35259\t\t\t-\tat9900_1_chr1\t33449\t35117\tg5\t0.56\t-\tAUGUSTUS\tgene\t.\tID=g5\n",
      "at9900_1_chr1\t35424\t39934\t\t\t-\tat9900_1_chr1\t36060\t39653\tg6\t0.06\t-\tAUGUSTUS\tgene\t.\tID=g6\n",
      "at9900_1_chr1\t40506\t43073\t\t\t-\tat9900_1_chr1\t40952\t42933\tg7\t0.51\t-\tAUGUSTUS\tgene\t.\tID=g7\n",
      "at9900_1_chr1\t47025\t49116\t\t\t-\tat9900_1_chr1\t47559\t48846\tg8\t1\t-\tAUGUSTUS\tgene\t.\tID=g8\n",
      "at9900_1_chr1\t49290\t51360\t\t\t-\tat9900_1_chr1\t49761\t51223\tg9\t1\t-\tAUGUSTUS\tgene\t.\tID=g9\n",
      "at9900_1_chr1\t51977\t53288\t\t\t-\tat9900_1_chr1\t52352\t53032\tg10\t1\t-\tAUGUSTUS\tgene\t.\tID=g10\n"
     ]
    }
   ],
   "source": [
    "head tmp/nlr_to_loci/loci_to_augustus_mapping3.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "25abd9c5-ca27-45e7-942f-401b78cb597f",
   "metadata": {},
   "outputs": [],
   "source": [
    "for sample in $(cat tmp/nlr_to_loci/at9900_NLRs.txt) \n",
    "do\n",
    "\n",
    "grep -P \"\\t${sample}\\t\" tmp/nlr_to_loci/loci_to_augustus_mapping3.txt >> tmp/nlr_to_loci/at9900.augustus_NLRs_v2.bed\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "60808b4a-b02f-4bad-944e-8f0bd0284978",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "at9900_1_chr1\t1592405\t1593754\t\t\t+\tat9900_1_chr1\t1592668\t1593577\tg458\t0.51\t+\tAUGUSTUS\tgene\t.\tID=g458\n",
      "at9900_1_chr1\t3150877\t3151742\t\t\t+\tat9900_1_chr1\t3150960\t3151742\tg902\t0.55\t+\tAUGUSTUS\tgene\t.\tID=g902\n",
      "at9900_1_chr1\t3668200\t3671806\t\t\t-\tat9900_1_chr1\t3668434\t3671401\tg1043\t0.94\t-\tAUGUSTUS\tgene\t.\tID=g1043\n",
      "at9900_1_chr1\t4164265\t4168029\t\t\t+\tat9900_1_chr1\t4164972\t4166847\tg1180\t0.96\t+\tAUGUSTUS\tgene\t.\tID=g1180\n",
      "at9900_1_chr1\t4164265\t4168029\t\t\t+\tat9900_1_chr1\t4166929\t4167637\tg1181\t0.73\t+\tAUGUSTUS\tgene\t.\tID=g1181\n",
      "at9900_1_chr1\t4196095\t4207491\t\t\t-\tat9900_1_chr1\t4196164\t4198849\tg1189\t0.85\t-\tAUGUSTUS\tgene\t.\tID=g1189\n",
      "at9900_1_chr1\t4196095\t4207491\t\t\t-\tat9900_1_chr1\t4199895\t4207315\tg1190\t0.06\t-\tAUGUSTUS\tgene\t.\tID=g1190\n",
      "at9900_1_chr1\t5487555\t5496405\t\t\t+\tat9900_1_chr1\t5488296\t5490871\tg1572\t0.29\t+\tAUGUSTUS\tgene\t.\tID=g1572\n",
      "at9900_1_chr1\t6073371\t6078643\t\t\t-\tat9900_1_chr1\t6073554\t6077101\tg1747\t1\t-\tAUGUSTUS\tgene\t.\tID=g1747\n",
      "at9900_1_chr1\t6072921\t6078886\t\t\t+\tat9900_1_chr1\t6077423\t6078686\tg1748\t1\t+\tAUGUSTUS\tgene\t.\tID=g1748\n"
     ]
    }
   ],
   "source": [
    "head tmp/nlr_to_loci/at9900.augustus_NLRs_v2.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b8338a2e-9f26-40fd-b891-2031f6976530",
   "metadata": {},
   "source": [
    "## Fixing pfam gffs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "21fb37ed-0ce3-4168-8f58-03707f261b96",
   "metadata": {},
   "outputs": [],
   "source": [
    "dl20=(\n",
    "at6137 at6923 at6929 at7143\n",
    "at8285 at9104 at9336 at9503\n",
    "at9578 at9744 at9762 at9806\n",
    "at9830 at9847 at9852 at9879\n",
    "at9883 at9900\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "3e9c44c6-7edb-47b9-be82-a83e2243d5af",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "at6137_1_chr1\thmmscan\tPFAM\t5091\t5571\t92.6\t+\t.\tID=g1.t1.NAM.1;Name=PF02365.NAM.No_apical_meristem_(NAM)_protein\n",
      "\n",
      "at6137_1_chr1\thmmscan\tPFAM\t5091\t5232\t92.6\t+\t.\tParent=g1.t1.NAM.1;Name=PF02365.NAM.No_apical_meristem_(NAM)_protein\n",
      "at6137_1_chr1\thmmscan\tPFAM\t5315\t5571\t92.6\t+\t.\tParent=g1.t1.NAM.1;Name=PF02365.NAM.No_apical_meristem_(NAM)_protein\n",
      "at6137_1_chr1\thmmscan\tPFAM\t8473\t9959\t159.4\t-\t.\tID=g2.t1.Arv1.1;Name=PF04161.Arv1.Arv1-like_family\n",
      "\n",
      "at6137_1_chr1\thmmscan\tPFAM\t9882\t9959\t159.4\t-\t.\tParent=g2.t1.Arv1.1;Name=PF04161.Arv1.Arv1-like_family\n",
      "at6137_1_chr1\thmmscan\tPFAM\t9728\t9775\t159.4\t-\t.\tParent=g2.t1.Arv1.1;Name=PF04161.Arv1.Arv1-like_family\n",
      "at6137_1_chr1\thmmscan\tPFAM\t9547\t9636\t159.4\t-\t.\tParent=g2.t1.Arv1.1;Name=PF04161.Arv1.Arv1-like_family\n",
      "at6137_1_chr1\thmmscan\tPFAM\t9254\t9299\t159.4\t-\t.\tParent=g2.t1.Arv1.1;Name=PF04161.Arv1.Arv1-like_family\n"
     ]
    }
   ],
   "source": [
    "head ../web_apollo/v2/pfam_max_augustus/at6137_pfam_genomic-v2.3.gff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "id": "a6459ac7-d07b-420e-93a0-fd66c68da913",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "for accession in \"${dl20[@]}\"\n",
    "do\n",
    "    awk 'BEGIN{IFS=OFS=\"\\t\"; print \"##gff-version 3\"}$9 ~ /^ID/{print($1, $3$2, \"gene\", $4, $5, $6, $7, $8, $9)}' \\\n",
    "        < ../web_apollo/v2/pfam_max_augustus/${accession}_pfam_genomic-v2.3.gff  \\\n",
    "        > output/04_analyses_with_fixed_at9879/prep_for_decision_making/${accession}_pfam_acutalgff.gff3\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "423a4e4c-9c66-49b1-8fb8-d8468a24bcbd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "7695281b-242a-4d94-a69d-531dda201daa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/luisa/work/detlef/difflines_annex/assembly-and-annotation\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
